<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Maize genome assembly and annotation: Hands-on training: Scaffolding using Optical Genome Mapping</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="favicons/rcac/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicons/rcac/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicons/rcac/favicon-16x16.png"><link rel="manifest" href="favicons/rcac/site.webmanifest"><link rel="mask-icon" href="favicons/rcac/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav rcac"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="rcac-logo" alt="rcac" src="assets/images/rcac-logo.svg"></div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/7-scaffolding-assemblies.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav rcac" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="rcac" src="assets/images/rcac-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Maize genome assembly and annotation: Hands-on training
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Maize genome assembly and annotation: Hands-on training
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Maize genome assembly and annotation: Hands-on training
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 43%" class="percentage">
    43%
  </div>
  <div class="progress rcac">
    <div class="progress-bar rcac" role="progressbar" style="width: 43%" aria-valuenow="43" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/7-scaffolding-assemblies.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="1-introduction.html">1. Introduction to Genome assembly</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="2-genome-assembly-strategies.html">2. Assembly Strategies</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="3-data-quality-control.html">3. Data Quality Control</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="4-pacbio-hifi-assembly.html">4. PacBio HiFi Assembly using HiFiasm</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="5-oxford-nanopore-assembly.html">5. Oxford Nanopore Assembly using Flye</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="6-hybrid-assembly.html">6. Hybrid Long Read Assembly (optional)</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        7. Scaffolding using Optical Genome Mapping
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#introduction-to-bionano-optical-genome-mapping-ogm">Introduction to Bionano optical genome mapping (OGM)</a></li>
<li><a href="#bionano-solve-hybrid-scaffolding">Bionano Solve Hybrid Scaffolding</a></li>
<li><a href="#installation-and-setup">Installation and Setup</a></li>
<li><a href="#running-bionano-solve">Running Bionano Solve</a></li>
<li><a href="#understanding-hybrid-scaffolding-output">Understanding Hybrid Scaffolding Output</a></li>
<li><a href="#quality-assessment-of-hybrid-scaffolds">Quality Assessment of Hybrid Scaffolds</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="8-assembly-evaluation.html">8. Assembly Assessment</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="9-annotation-strategies.html">9. Annotation Strategies</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="10-gene-annotation-with-braker.html">10. Annotation using BRAKER</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="11-gene-annotation-with-helixer.html">11. Annotation using Helixer</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush13">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading13">
        <a href="12-gene-annotation-with-gemoma.html">12. Homology-based gene annotation with GeMoMa</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush14">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading14">
        <a href="13-functional-annotation-with-entap.html">13. Functional annotation using EnTAP</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush15">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading15">
        <a href="14-annotation-assesment.html">14. Annotation Assesment</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>
                      <li><a href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="6-hybrid-assembly.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="8-assembly-evaluation.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="6-hybrid-assembly.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Hybrid Long Read
        </a>
        <a class="chapter-link float-end" href="8-assembly-evaluation.html" rel="next">
          Next: Assembly Assessment...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Scaffolding using Optical Genome Mapping</h1>
        <p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/7-scaffolding-assemblies.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>What is Bionano optical genome mapping (OGM) and how does it improve
genome assembly?</li>
<li>How does Bionano Solve hybrid scaffolding integrate optical maps
with sequence assemblies?</li>
<li>What are the key steps involved in running the Bionano Solve
pipeline for hybrid scaffolding?</li>
<li>How can you assess the quality of hybrid scaffolds generated by
Bionano Solve?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Understand the principles of Bionano optical genome mapping (OGM)
and its role in genome assembly.</li>
<li>Learn how to run the Bionano Solve hybrid scaffolding pipeline to
improve genome assemblies.</li>
<li>Explore the key steps involved in scaffolding HiFiasm and Flye
assemblies using Bionano Solve.</li>
<li>Evaluate the quality of hybrid scaffolds generated by Bionano Solve
and interpret the results.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction-to-bionano-optical-genome-mapping-ogm">Introduction to Bionano optical genome mapping (OGM)<a class="anchor" aria-label="anchor" href="#introduction-to-bionano-optical-genome-mapping-ogm"></a></h2>
<hr class="half-width"><p>Bionano optical mapping is a high-resolution genome analysis
technique that generates long-range structural information by labeling
and imaging ultra-long DNA molecules. It provides genome-wide maps that
can be used to scaffold contigs from sequencing-based assemblies,
significantly improving contiguity and structural accuracy. By
integrating Bionano maps with assemblies from PacBio HiFi and Oxford
Nanopore Technologies (ONT), misassemblies can be corrected, chimeric
contigs resolved, and scaffold N50s increased by orders of magnitude.
This approach is particularly valuable for complex genomes, where
repetitive sequences and structural variations pose challenges for
traditional sequencing methods. Bionano hybrid scaffolding has become a
standard for enhancing genome assemblies, enabling researchers to
achieve high-quality, chromosome-level assemblies efficiently.</p>
<figure><img src="https://github.com/user-attachments/assets/41c7c42b-c5e0-4d0d-b899-a479e62ffaee" alt="Overview of Bionano OGM" class="figure mx-auto d-block"><div class="figcaption">Overview of Bionano OGM</div>
</figure></section><section><h2 class="section-heading" id="bionano-solve-hybrid-scaffolding">Bionano Solve Hybrid Scaffolding<a class="anchor" aria-label="anchor" href="#bionano-solve-hybrid-scaffolding"></a></h2>
<hr class="half-width"><p>Bionano Solve improves genome assembly by integrating optical genome
mapping data with sequence assemblies, generating ultra-long hybrid
scaffolds that enhance contiguity and accuracy. The pipeline identifies
and resolves assembly conflicts, orders and orients sequence contigs,
and estimates gap sizes between adjacent sequences.</p>
<p>The scaffolding workflow involves:</p>
<ol style="list-style-type: decimal"><li>In Silico Map Generation – Converting sequence assembly into a map
format for alignment.</li>
<li>Conflict Resolution – Aligning in silico maps to Bionano genome maps
and identifying misassemblies.</li>
<li>Hybrid Scaffolding – Merging high-confidence sequence and optical
maps into a refined scaffold.</li>
<li>Final Alignment – Mapping sequence contigs back to the hybrid
scaffold for consistency validation.</li>
<li>Output Generation – Producing final AGP and FASTA files with
corrected genome structures.</li>
</ol><figure><img src="https://github.com/user-attachments/assets/e568386e-348d-4b18-82ca-e304369f0fed" alt="Scaffolding using Bionano" class="figure mx-auto d-block"><div class="figcaption">Scaffolding using Bionano</div>
</figure><p><strong>What is the source of this data?</strong></p>
<ul><li>The optical genome mapping data was obtained from the project <a href="https://www.ebi.ac.uk/ena/browser/view/PRJEB50694" class="external-link">PRJEB50694</a>.<br></li>
<li>Data corresponds to <em>Arabidopsis thaliana</em> ecotype Col-0,
generated using Bionano optical genome mapping technology.<br></li>
<li>The dataset is publicly available on the European Nucleotide Archive
(ENA) and can be downloaded using:<br></li>
<li>The <code>.cmap</code> file contains high-resolution optical maps
used for scaffolding and structural validation of genome
assemblies.</li>
</ul><p>To download:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">wget</span> ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR522/ERR5229977/Col-0_2021-09-30.cmap</span></code></pre>
</div>
</section><section><h2 class="section-heading" id="installation-and-setup">Installation and Setup<a class="anchor" aria-label="anchor" href="#installation-and-setup"></a></h2>
<hr class="half-width"><p>Bionano Solve is available on Bionano.com and can be installed on
Linux-based systems. The software requires a valid license and access to
Bionano data files for processing.</p>
<p>Custom container with just the hybrid scaffolding tools can be used
to run the Bionano Solve pipeline. On Negishi, you can add it to your
PATH using the command below:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span><span class="va">$PATH</span>:/apps/biocontainers/exported-wrappers/bionano/3.8.0</span></code></pre>
</div>
</section><section><h2 class="section-heading" id="running-bionano-solve">Running Bionano Solve<a class="anchor" aria-label="anchor" href="#running-bionano-solve"></a></h2>
<hr class="half-width"><p>To scaffold a genome using Bionano Solve, you need to provide the
following input files:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span><span class="va">$PATH</span>:/apps/biocontainers/exported-wrappers/bionano/3.8.0</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">run_hybridscaffold.sh</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  <span class="ex">-c</span> /opt/Solve3.7_10192021_74_1/HybridScaffold/1.0/hybridScaffold_DLE1_config.xml<span class="dt">\</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  <span class="at">-b</span> input.cmap <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="at">-n</span> genome.fasta <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="at">-u</span> CTTAAG <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  <span class="at">-z</span> results_output.zip <span class="dt">\</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>  <span class="at">-w</span> log.txt <span class="dt">\</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>  <span class="at">-B</span> 2 <span class="dt">\</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>  <span class="at">-N</span> 2 <span class="dt">\</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>  <span class="at">-g</span> <span class="dt">\</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>  <span class="at">-f</span> <span class="dt">\</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>  <span class="at">-r</span> /opt/Solve3.7_10192021_74_1/RefAligner/1.0/sse/RefAligner <span class="dt">\</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>  <span class="at">-p</span> /opt/Solve3.7_10192021_74_1/Pipeline/1.0 <span class="dt">\</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>  <span class="at">-o</span> output_dir</span></code></pre>
</div>
<div id="options-used" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="options-used" class="callout-inner">
<h3 class="callout-title">Options used</h3>
<div class="callout-content">
<table class="table"><colgroup><col width="12%"><col width="20%"><col width="66%"></colgroup><thead><tr class="header"><th align="center"><strong>Option</strong></th>
<th align="left"><strong>Argument</strong></th>
<th><strong>Description</strong></th>
</tr></thead><tbody><tr class="odd"><td align="center"><code>-c</code></td>
<td align="left"><code>/opt/Solve3.7_10192021_74_1/HybridScaffold/1.0/hybridScaffold_DLE1_config.xml</code></td>
<td>Specifies the hybrid scaffolding configuration file required for the
pipeline.</td>
</tr><tr class="even"><td align="center"><code>-b</code></td>
<td align="left"><code>input.cmap</code></td>
<td>Input Bionano CMAP file, which contains the optical genome map
data.</td>
</tr><tr class="odd"><td align="center"><code>-n</code></td>
<td align="left"><code>genome.fasta</code></td>
<td>Input genome sequence in FASTA format from NGS assembly.</td>
</tr><tr class="even"><td align="center"><code>-u</code></td>
<td align="left"><code>CTTAAG</code></td>
<td>Specifies the sequence of the enzyme recognition site, overriding
the one in the config XML file.</td>
</tr><tr class="odd"><td align="center"><code>-z</code></td>
<td align="left"><code>results_output.zip</code></td>
<td>Generates a ZIP archive containing essential output files.</td>
</tr><tr class="even"><td align="center"><code>-w</code></td>
<td align="left"><code>log.txt</code></td>
<td>Defines the name of the status text file needed for IrysView.</td>
</tr><tr class="odd"><td align="center"><code>-B</code></td>
<td align="left"><code>2</code></td>
<td>Conflict filter level: <code>2</code> means cut the contig at
conflict points (required if not using <code>-M</code>).</td>
</tr><tr class="even"><td align="center"><code>-N</code></td>
<td align="left"><code>2</code></td>
<td>Conflict filter level: <code>2</code> means cut the contig at
conflict points (same as <code>-B</code>, applied to sequencing
contigs).</td>
</tr><tr class="odd"><td align="center"><code>-g</code></td>
<td align="left">(No argument)</td>
<td>Enables trimming of overlapping NGS sequences during AGP and FASTA
export.</td>
</tr><tr class="even"><td align="center"><code>-f</code></td>
<td align="left">(No argument)</td>
<td>Forces output generation and overwrites any existing files in the
output directory.</td>
</tr><tr class="odd"><td align="center"><code>-r</code></td>
<td align="left"><code>/opt/Solve3.7_10192021_74_1/RefAligner/1.0/sse/RefAligner</code></td>
<td>Specifies the path to the RefAligner program, which is required for
scaffolding.</td>
</tr><tr class="even"><td align="center"><code>-p</code></td>
<td align="left"><code>/opt/Solve3.7_10192021_74_1/Pipeline/1.0</code></td>
<td>Specifies the directory for the de novo assembly pipeline (optional,
required for <code>-x</code>).</td>
</tr><tr class="odd"><td align="center"><code>-o</code></td>
<td align="left"><code>output_dir</code></td>
<td>Defines the output folder where scaffolded results will be
stored.</td>
</tr></tbody></table></div>
</div>
</div>
<div class="section level3">
<h3 id="scffolding-hifiasm-assembly-with-bionano-solve">Scffolding HiFiasm assembly with Bionano Solve<a class="anchor" aria-label="anchor" href="#scffolding-hifiasm-assembly-with-bionano-solve"></a></h3>
<p>For HiFiasm assembly, you need to provide the HiFiasm assembly FASTA
file as input to the Bionano Solve pipeline. The command structure
remains the same, with the only change being the input sequence
file.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span><span class="va">$PATH</span>:/apps/biocontainers/exported-wrappers/bionano/3.8.0</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">run_hybridscaffold.sh</span> <span class="dt">\</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="at">-c</span> /opt/Solve3.7_10192021_74_1/HybridScaffold/1.0/hybridScaffold_DLE1_config.xml<span class="dt">\</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="at">-b</span> Evry.OpticalMap.Col-0.cmap <span class="dt">\</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>  <span class="at">-n</span> hifiasm_60x/athaliana_hifi.asm.bp.p_ctg.fasta <span class="dt">\</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>  <span class="at">-u</span> CTTAAG <span class="dt">\</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>  <span class="at">-z</span> results_bionano_hifiasm_scaffolding.zip <span class="dt">\</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>  <span class="at">-w</span> log.txt <span class="dt">\</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>  <span class="at">-B</span> 2 <span class="dt">\</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>  <span class="at">-N</span> 2 <span class="dt">\</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  <span class="at">-g</span> <span class="dt">\</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>  <span class="at">-f</span> <span class="dt">\</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>  <span class="at">-r</span> /opt/Solve3.7_10192021_74_1/RefAligner/1.0/sse/RefAligner <span class="dt">\</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>  <span class="at">-p</span> /opt/Solve3.7_10192021_74_1/Pipeline/1.0 <span class="dt">\</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>  <span class="at">-o</span> bionano_hifiasm_scaffolding</span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="scffolding-flye-assembly-with-bionano-solve">Scffolding Flye assembly with Bionano Solve<a class="anchor" aria-label="anchor" href="#scffolding-flye-assembly-with-bionano-solve"></a></h3>
<p>For Flye assembly, the process is similar to HiFiasm, but you need to
provide the Flye assembly FASTA file instead of the HiFiasm assembly.
The command structure remains the same, with the only change being the
input sequence file.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span><span class="va">$PATH</span>:/apps/biocontainers/exported-wrappers/bionano/3.8.0</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">run_hybridscaffold.sh</span> <span class="dt">\</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="at">-c</span> /opt/Solve3.7_10192021_74_1/HybridScaffold/1.0/hybridScaffold_DLE1_config.xml<span class="dt">\</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="at">-b</span> workshop_assembly/col-0_bionano/Evry.OpticalMap.Col-0.cmap <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  <span class="at">-n</span> flye_ont_60x/assembly.fasta <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>  <span class="at">-u</span> CTTAAG <span class="dt">\</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>  <span class="at">-z</span> results_bionano_flye_scaffolding.zip <span class="dt">\</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>  <span class="at">-w</span> log.txt <span class="dt">\</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>  <span class="at">-B</span> 2 <span class="dt">\</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>  <span class="at">-N</span> 2 <span class="dt">\</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>  <span class="at">-g</span> <span class="dt">\</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>  <span class="at">-f</span> <span class="dt">\</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>  <span class="at">-r</span> /opt/Solve3.7_10192021_74_1/RefAligner/1.0/sse/RefAligner <span class="dt">\</span></span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a>  <span class="at">-p</span> /opt/Solve3.7_10192021_74_1/Pipeline/1.0 <span class="dt">\</span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a>  <span class="at">-o</span> bionano_flye_scaffolding</span></code></pre>
</div>
</div>
</section><section><h2 class="section-heading" id="understanding-hybrid-scaffolding-output">Understanding Hybrid Scaffolding Output<a class="anchor" aria-label="anchor" href="#understanding-hybrid-scaffolding-output"></a></h2>
<hr class="half-width"><p>The output of the Bionano Solve pipeline includes scaffolded genome
assemblies in AGP and FASTA formats, along with alignment and conflict
resolution information. The hybrid scaffolds provide a more accurate
representation of the genome structure, with improved contiguity and
reduced misassemblies. The output files can be visualized using genome
browsers or alignment viewers to assess the quality and completeness of
the assembly.</p>
<table class="table"><colgroup><col width="48%"><col width="52%"></colgroup><thead><tr class="header"><th><strong>Folder</strong></th>
<th><strong>Contents</strong></th>
</tr></thead><tbody><tr class="odd"><td><strong>agp_fasta/</strong></td>
<td>Final scaffolded genome assembly in FASTA, AGP format, alignment
results, gap information, and logs.</td>
</tr><tr class="even"><td><strong>align0/</strong></td>
<td>Initial alignment of optical maps to the sequence assembly,
including XMAP, CMAP, and error logs.</td>
</tr><tr class="odd"><td><strong>align1/</strong></td>
<td>Secondary alignment refinement, similar to align0 but after
resolving initial inconsistencies.</td>
</tr><tr class="even"><td><strong>align_final/</strong></td>
<td>Final alignment results of hybrid scaffolds to optical maps,
including mapping rates and statistics.</td>
</tr><tr class="odd"><td><strong>assignAlignType/</strong></td>
<td>Tracks conflicts between NGS contigs and optical maps, includes
exclusion and trimming decisions.</td>
</tr><tr class="even"><td><strong>cut_conflicts/</strong></td>
<td>Stores files related to contig trimming and conflict resolution
between NGS and optical maps.</td>
</tr><tr class="odd"><td><strong>fa2cmap/</strong></td>
<td>Converts the sequence assembly into Bionano’s CMAP format before
integration with optical maps.</td>
</tr><tr class="even"><td><strong>hybrid_scaffolds/</strong></td>
<td>Contains final scaffolded genome with CMAP, XMAP, AGP files, and a
scaffolding report.</td>
</tr><tr class="odd"><td><strong>mergeNGS_BN/</strong></td>
<td>Stores intermediate files merging NGS contigs with Bionano maps,
including hybrid scaffold progress.</td>
</tr><tr class="even"><td><strong>results_output.zip</strong></td>
<td>Compressed archive containing essential scaffolding results for easy
sharing.</td>
</tr></tbody></table><p>Within <code>hybrid_scaffolds</code>, the files ending with
<code>HYBRID_SCAFFOLD.fasta</code> and
<code>HYBRID_SCAFFOLD_NOT_SCAFFOLDED.fasta</code> represent the final
scaffolded genome and unplaced contigs, respectively. You will need to
merge these files to obtain your final scaffolded genome assembly.</p>
</section><section><h2 class="section-heading" id="quality-assessment-of-hybrid-scaffolds">Quality Assessment of Hybrid Scaffolds<a class="anchor" aria-label="anchor" href="#quality-assessment-of-hybrid-scaffolds"></a></h2>
<hr class="half-width"><p>The hybird scaffolds report file will be in the
<code>hybrid_scaffolds</code> directory and will provide a summary of
the scaffolding process, including alignment statistics, conflict
resolution, and scaffold N50 values. This report is essential for
evaluating the quality and completeness of the hybrid scaffolds and
identifying any potential issues that need further investigation.</p>
<table class="table"><colgroup><col width="61%"><col width="19%"><col width="19%"></colgroup><thead><tr class="header"><th align="left"><strong>Category</strong></th>
<th align="right"><strong>PacBio HiFi (hifiasm)</strong></th>
<th align="right"><strong>ONT (Flye)</strong></th>
</tr></thead><tbody><tr class="odd"><td align="left"><strong>Original BioNano Genome Map</strong></td>
<td align="right"></td>
<td align="right"></td>
</tr><tr class="even"><td align="left">Count</td>
<td align="right">18</td>
<td align="right">18</td>
</tr><tr class="odd"><td align="left">Min length (Mbp)</td>
<td align="right">0.342</td>
<td align="right">0.342</td>
</tr><tr class="even"><td align="left">Median length (Mbp)</td>
<td align="right">3.956</td>
<td align="right">3.956</td>
</tr><tr class="odd"><td align="left">Mean length (Mbp)</td>
<td align="right">7.396</td>
<td align="right">7.396</td>
</tr><tr class="even"><td align="left">N50 length (Mbp)</td>
<td align="right">15.529</td>
<td align="right">15.529</td>
</tr><tr class="odd"><td align="left">Max length (Mbp)</td>
<td align="right">17.518</td>
<td align="right">17.518</td>
</tr><tr class="even"><td align="left">Total length (Mbp)</td>
<td align="right">133.124</td>
<td align="right">133.124</td>
</tr><tr class="odd"><td align="left"><strong>Original NGS Sequences</strong></td>
<td align="right"></td>
<td align="right"></td>
</tr><tr class="even"><td align="left">Count</td>
<td align="right">152</td>
<td align="right">43</td>
</tr><tr class="odd"><td align="left">Min length (Mbp)</td>
<td align="right">0.027</td>
<td align="right">0.008</td>
</tr><tr class="even"><td align="left">Median length (Mbp)</td>
<td align="right">0.050</td>
<td align="right">0.276</td>
</tr><tr class="odd"><td align="left">Mean length (Mbp)</td>
<td align="right">0.896</td>
<td align="right">2.797</td>
</tr><tr class="even"><td align="left">N50 length (Mbp)</td>
<td align="right">7.981</td>
<td align="right">9.261</td>
</tr><tr class="odd"><td align="left">Max length (Mbp)</td>
<td align="right">13.758</td>
<td align="right">14.609</td>
</tr><tr class="even"><td align="left">Total length (Mbp)</td>
<td align="right">136.156</td>
<td align="right">120.259</td>
</tr><tr class="odd"><td align="left"><strong>Conflict Resolution (BNG-NGS
Alignment)</strong></td>
<td align="right"></td>
<td align="right"></td>
</tr><tr class="even"><td align="left">Conflict cuts made to Bionano maps</td>
<td align="right">2</td>
<td align="right">0</td>
</tr><tr class="odd"><td align="left">Conflict cuts made to NGS sequences</td>
<td align="right">30</td>
<td align="right">0</td>
</tr><tr class="even"><td align="left">Bionano maps to be cut</td>
<td align="right">2</td>
<td align="right">0</td>
</tr><tr class="odd"><td align="left">NGS sequences to be cut</td>
<td align="right">18</td>
<td align="right">0</td>
</tr><tr class="even"><td align="left"><strong>NGS FASTA Sequence in Hybrid
Scaffold</strong></td>
<td align="right"></td>
<td align="right"></td>
</tr><tr class="odd"><td align="left">Count</td>
<td align="right">40</td>
<td align="right">26</td>
</tr><tr class="even"><td align="left">Min length (Mbp)</td>
<td align="right">0.033</td>
<td align="right">0.065</td>
</tr><tr class="odd"><td align="left">Median length (Mbp)</td>
<td align="right">0.945</td>
<td align="right">1.681</td>
</tr><tr class="even"><td align="left">Mean length (Mbp)</td>
<td align="right">2.689</td>
<td align="right">4.558</td>
</tr><tr class="odd"><td align="left">N50 length (Mbp)</td>
<td align="right">8.437</td>
<td align="right">9.261</td>
</tr><tr class="even"><td align="left">Max length (Mbp)</td>
<td align="right">13.484</td>
<td align="right">14.609</td>
</tr><tr class="odd"><td align="left">Total length (Mbp)</td>
<td align="right">107.558</td>
<td align="right">118.508</td>
</tr><tr class="even"><td align="left"><strong>Hybrid Scaffold FASTA</strong></td>
<td align="right"></td>
<td align="right"></td>
</tr><tr class="odd"><td align="left">Count</td>
<td align="right">11</td>
<td align="right">12</td>
</tr><tr class="even"><td align="left">Min length (Mbp)</td>
<td align="right">0.104</td>
<td align="right">0.524</td>
</tr><tr class="odd"><td align="left">Median length (Mbp)</td>
<td align="right">11.824</td>
<td align="right">12.426</td>
</tr><tr class="even"><td align="left">Mean length (Mbp)</td>
<td align="right">10.518</td>
<td align="right">9.891</td>
</tr><tr class="odd"><td align="left">N50 length (Mbp)</td>
<td align="right">14.479</td>
<td align="right">14.886</td>
</tr><tr class="even"><td align="left">Max length (Mbp)</td>
<td align="right">15.227</td>
<td align="right">16.188</td>
</tr><tr class="odd"><td align="left">Total length (Mbp)</td>
<td align="right">115.698</td>
<td align="right">118.689</td>
</tr><tr class="even"><td align="left"><strong>Hybrid Scaffold FASTA + Not Scaffolded
NGS</strong></td>
<td align="right"></td>
<td align="right"></td>
</tr><tr class="odd"><td align="left">Count</td>
<td align="right">161</td>
<td align="right">33</td>
</tr><tr class="even"><td align="left">Min length (Mbp)</td>
<td align="right">0.024</td>
<td align="right">0.006</td>
</tr><tr class="odd"><td align="left">Median length (Mbp)</td>
<td align="right">0.051</td>
<td align="right">0.159</td>
</tr><tr class="even"><td align="left">Mean length (Mbp)</td>
<td align="right">0.896</td>
<td align="right">3.650</td>
</tr><tr class="odd"><td align="left">N50 length (Mbp)</td>
<td align="right">14.083</td>
<td align="right">14.886</td>
</tr><tr class="even"><td align="left">Max length (Mbp)</td>
<td align="right">15.227</td>
<td align="right">16.188</td>
</tr><tr class="odd"><td align="left">Total length (Mbp)</td>
<td align="right">144.295</td>
<td align="right">120.440</td>
</tr></tbody></table><div id="which-assembler-and-data-performed-better" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="which-assembler-and-data-performed-better" class="callout-inner">
<h3 class="callout-title">Which assembler and data performed better?</h3>
<div class="callout-content">
<ul><li>The HiFiasm assembly with PacBio HiFi data resulted in a higher N50
length and total length in the hybrid scaffold compared to the Flye
assembly with ONT data.</li>
<li>The conflict resolution process involved more cuts in the NGS
sequences for the HiFiasm assembly, indicating a higher level of
alignment discrepancies.</li>
<li>The final hybrid scaffold from the HiFiasm assembly had a higher N50
length and total length, suggesting better contiguity and completeness
compared to the Flye assembly.</li>
</ul></div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>Bionano optical genome mapping (OGM) provides long-range structural
information for scaffolding genome assemblies.</li>
<li>Bionano Solve hybrid scaffolding integrates optical maps with
sequence assemblies to improve contiguity and accuracy.</li>
<li>The Bionano Solve pipeline involves in silico map generation,
conflict resolution, hybrid scaffolding, and final alignment.</li>
<li>The output of Bionano Solve includes scaffolded genome assemblies in
AGP and FASTA formats, alignment results, and conflict resolution
information.</li>
<li>Quality assessment of hybrid scaffolds involves evaluating alignment
statistics, conflict resolution, scaffold N50 values, and completeness
of the assembly.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="6-hybrid-assembly.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="8-assembly-evaluation.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="6-hybrid-assembly.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Hybrid Long Read
        </a>
        <a class="chapter-link float-end" href="8-assembly-evaluation.html" rel="next">
          Next: Assembly Assessment...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/7-scaffolding-assemblies.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://rcac-bioinformatics.github.io/maize-genomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://rcac-bioinformatics.github.io/maize-genomics/" class="external-link">Source</a></p>
				<p><a href="https://rcac-bioinformatics.github.io/maize-genomics/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:rcac-help@purdue.edu">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.11" class="external-link">sandpaper (0.16.11)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/aseetharam/varnish/tree/db4f5c6f3c4799dfe8247fd057642c8099d5244b" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://maize-genomics.github.io/NA/7-scaffolding-assemblies.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene-annotation, braker, helixer, gemoma, entap, omark, busco, hifiasm",
  "name": "Scaffolding using Optical Genome Mapping",
  "creativeWorkStatus": "active",
  "url": "https://maize-genomics.github.io/NA/7-scaffolding-assemblies.html",
  "identifier": "https://maize-genomics.github.io/NA/7-scaffolding-assemblies.html",
  "dateCreated": "2025-02-05",
  "dateModified": "2025-03-01",
  "datePublished": "2025-03-01"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

