---
title: 'Annotation Strategies'
teaching: 10
exercises: 2
---

:::::::::::::::::::::::::::::::::::::: questions 

- What are the different strategies used for genome annotation?  
- How do various methods predict genes, and what are their strengths and limitations?  
- What role do evidence play in improving gene predictions?  
- How do deep learning and large models enhance genome annotation accuracy?  
- What tools are used in this workshop, and how do they fit into different strategies?  


::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Understand the principles behind different genome annotation approaches.  
- Learn how intrinsic, extrinsic, and hybrid methods predict gene structures.  
- Explore the role of multi-evidence approaches, including transcriptomics, proteomics, and functional annotation.  
- Introduce modern machine learning and large model-based methods for gene annotation.  
- Familiarize with annotation tools used in this workshop and their specific applications.  

::::::::::::::::::::::::::::::::::::::::::::::::

## Introduction

- Genome annotation identifies functional elements in a genome, including protein-coding genes, non-coding rnas, and regulatory regions  
- Annotation strategies vary between evidence-based, ab initio, and hybrid approaches, each with strengths and limitations  
- Selecting the right method depends on genome complexity, available transcriptomic and proteomic data, and the need for accuracy versus sensitivity

## Annotation Methods

**Intrinsic (Ab Initio) Methods**

- Predict genes based on sequence features without external evidence.  
- Use statistical models such as hidden Markov models (HMMs) to recognize coding regions, exon-intron boundaries, and gene structures.  
- Rely on nucleotide composition patterns like codon usage, GC content, and splice site signals.  
- Perform well in well-studied genomes but struggle with novel genes and complex gene structures.  

**Extrinsic (Evidence-Based) Methods**  

- Utilize external data such as RNA-seq, ESTs, and protein homology to guide gene predictions.  
- Align transcriptomic and proteomic data to the genome to refine exon-intron structures.  
- Improve annotation accuracy by incorporating evolutionary conservation and known gene models.  
- Require high-quality reference datasets, making them less effective in newly sequenced organisms with little prior data.  

**Hybrid Methods**  

- Combine ab initio predictions with evidence-based approaches to improve accuracy and completeness.  
- Use gene predictions from statistical models and refine them with transcriptomic and proteomic data.  
- Balance sensitivity (detecting novel genes) and specificity (avoiding false positives).  
- Often used in large-scale genome annotation projects where both approaches complement each other.  

**Multi-Evidence based methods** 

- Incorporate diverse biological data sources to improve gene prediction accuracy.  
- Use **ribosome profiling (Ribo-seq)** and **proteomics** to confirm protein-coding potential.  
- Leverage **poly(A) signal presence** to distinguish mature transcripts from non-coding RNA.  
- Integrate **RNA-seq, long-read sequencing, and EST data** to refine exon-intron structures.  
- Provide direct experimental validation, reducing reliance on purely computational models.  

**Large Model-Based Methods**  

- Utilize deep learning and large language models (LMMs) trained on vast genomic datasets.  
- Predict gene structures by recognizing sequence patterns learned from diverse organisms.  
- Adapt to various genome complexities without requiring manually curated training data.  
- Identify non-canonical gene structures, alternative splicing events, and novel genes.  
- Example: **Helixer**, a GPU-based model, predicts genes using high-dimensional sequence patterns, improving over traditional HMM-based methods.  


:::::::::::::::::::::::::::::::::::::::  prereq

## What do we annotate in the genome?

- **Protein-coding genes**: Genes that encode proteins, identified based on open reading frames (ORFs) and coding sequence features.  
- **Pseudogenes**: Non-functional remnants of once-active genes that have accumulated mutations.  
- **Non-coding RNAs (ncRNAs)**: Functional RNAs that do not code for proteins, including tRNAs, rRNAs, and lncRNAs.  
- **Upstream Open Reading Frames (uORFs)**: Small ORFs in the 5' untranslated region (UTR) that can regulate translation.  
- **Alternative Splicing Variants**: Different transcript isoforms from the same gene, generated by exon-intron rearrangements.  
- **Repeat Elements and Transposons**: Mobile DNA sequences that can impact gene function and genome structure.  
- **Regulatory Elements**: Promoters, enhancers, and other sequences controlling gene expression.  

For this workshop, we focus only on **protein-coding genes** and their annotation.

::::::::::::::::::::::::::::::::::::::::::::::::


## Annotation Workflows Used


**1. BRAKER3**: Combines GeneMark-ET and AUGUSTUS to predict genes in eukaryotic genomes.

- Integrates GeneMark and AUGUSTUS with RNA-Seq and/or protein evidence.
- Supports various data types or runs without them.
- Fully automated training and gene prediction.
- Requires evidence data for high accuracy; performs poorly without it.
- Struggles with scalability on larger genomes.

::: tab

### w/ RNA-seq

<img src="https://github.com/user-attachments/assets/201d283e-83cc-4d2a-bb2b-68104dc72cc3" alt="with RNA-Seq" height="500">



### w/ Proteins

<img src="https://github.com/user-attachments/assets/6a0cc45f-12f4-4363-81de-93c929166334" alt="with Proteins" height="550">



### w/ RNA-seq + Proteins

![with RNAseq and proteins](https://github.com/user-attachments/assets/b48adfaa-c357-411d-a252-8450cf720e5f)



### _ab initio_

<img src="https://github.com/user-attachments/assets/5049a336-398a-42b6-834c-3adf75d8be07" alt="_ab initio_" height="500">



### Pretrained model

<img src="https://github.com/user-attachments/assets/81dcb3b6-25a1-4c22-a550-f6906a44af4e" alt="_ab initio_" height="500">


:::



**2. Helixer**: A deep learning-based gene prediction tool that uses a convolutional neural network (CNN) to predict genes in eukaryotic genomes.

- Utilizes a deep learning model trained on diverse animal and plant genomes, reducing the need for retraining on closely related species.
- Provides probabilities for every base, enabling detailed predictions for intergenic, untranslated, coding, and intronic regions.
- Outperforms traditional tools like AUGUSTUS in both base-wise metrics and RNA-Seq consistency.
- Handles large and complex eukaryotic genomes effectively, even for species with longer genes.
- Requires no additional data except genome â€“ no repeatmasking! 
- Cannot predict alternative splicing events or spliced start codons.



![Helixer](https://github.com/user-attachments/assets/fd826e9d-30f9-48fe-9e82-e93cc8d779ce)


**3. EASEL**: Efficient, Accurate, Scalable Eukaryotic modeLs: eukaryotic genome annotation tool

- Combines machine learning, RNA folding, and functional annotations to enhance prediction accuracy.
- Leverages RNA-Seq, protein alignments, and additional evidence for gene prediction.
- Built on Nextflow, ensuring portability across HPC systems and compatibility with Docker and Singularity.
- Features EnTAP integration for streamlined functional annotation.
- Provides multiple formats (GFF, GTF, CDS, protein sequences) for downstream analysis.
- Employs BUSCO for quality assessment and automates parameter optimization.


![EASEL](https://github.com/user-attachments/assets/12e320eb-5ea6-43a9-bedb-2ae64097e44b)


**4. EnTAP**: eukaryotic non-model annotation pipeline (functional annotation)

- Optimized for functional annotation of transcriptomes without reference genomes, addressing challenges like fragmentation and assembly artifacts.
- Much faster than traditional annotation pipelines.
- Integrates taxonomic filtering, contaminant detection, and informativeness metrics for selecting optimal alignments.
- Enables annotation across diverse repositories (e.g., RefSeq, Swiss-Prot) with customizable database integration.
- Includes gene family assignments, protein domains, and GO terms for enriched annotation.
- User-friendly, Unix-based pipeline suitable for HPC environments, combining simplicity with high throughput.


![EnTAP](https://github.com/user-attachments/assets/11c2a5e7-2545-45a8-9d59-0b8d233f4de9)


::: callout

## How to we access gene predictions?

- **Busco and Omark** assess gene set completeness by checking the presence of conserved orthologs across species, helping identify missing or fragmented genes  
- **Gff3 metrics** provide structural statistics on gene models, including exon-intron distribution, gene lengths, and coding sequence properties, highlighting inconsistencies in annotation  
- **Feature assignment** using RNA-seq read mapping quantifies how well predicted genes capture transcriptomic evidence, indicating potential missing or incorrect annotations  
- **Reference comparison** with known annotations measures sensitivity and precision, identifying correctly predicted, missed, or falsely annotated genes  
- **Multiple validation methods** combine structural, functional, and comparative assessments to improve confidence in gene predictions and refine annotation quality  

:::


::::::::::::::::::::::::::::::::::::: keypoints 

- Genome annotation strategies include **ab initio, evidence-based, hybrid, multi-evidence, and large model-based methods**, each with different strengths and data requirements.  
- **Ab initio methods** predict genes using sequence patterns, while **evidence-based methods** use RNA-seq, proteins, and evolutionary conservation for refinement.  
- **Multi-evidence approaches** integrate diverse biological data such as ribosome profiling, proteomics, and long-read RNA-seq to improve accuracy.  
- **Large model-based methods**, like Helixer, use deep learning to predict genes without requiring manual training, outperforming traditional tools in complex genomes.  
- The workshop focuses on **protein-coding gene annotation** using tools like **BRAKER, Helixer, EASEL, and EnTAP**, followed by assessment with BUSCO, OMARK, and structural metrics.

::::::::::::::::::::::::::::::::::::::::::::::::

