<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Maize genome assembly and annotation: Hands-on training: Assembly Strategies</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/rcac/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicons/rcac/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicons/rcac/favicon-16x16.png"><link rel="manifest" href="../favicons/rcac/site.webmanifest"><link rel="mask-icon" href="../favicons/rcac/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav rcac"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="rcac-logo" alt="rcac" src="../assets/images/rcac-logo.svg"></div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../2-genome-assembly-strategies.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav rcac" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="rcac" src="../assets/images/rcac-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Maize genome assembly and annotation: Hands-on training
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Maize genome assembly and annotation: Hands-on training
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Maize genome assembly and annotation: Hands-on training
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 7%" class="percentage">
    7%
  </div>
  <div class="progress rcac">
    <div class="progress-bar rcac" role="progressbar" style="width: 7%" aria-valuenow="7" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../2-genome-assembly-strategies.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="1-introduction.html">1. Introduction to Genome assembly</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        2. Assembly Strategies
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#assembly-strategies">Assembly Strategies</a></li>
<li><a href="#comparative-assembly-strategies">Comparative Assembly Strategies</a></li>
<li><a href="#contig-vs.-scaffold-vs.-chromosome-level-assembly">Contig vs. Scaffold vs. Chromosome-Level Assembly</a></li>
<li><a href="#workflow-for-various-assemblies">Workflow for Various Assemblies</a></li>
<li><a href="#assembly-evaluation">Assembly Evaluation</a></li>
<li><a href="#bionano-and-hi-c-reads-in-genome-assembly">Bionano and Hi-C Reads in Genome Assembly</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="3-data-quality-control.html">3. Data Quality Control</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="4-pacbio-hifi-assembly.html">4. PacBio HiFi Assembly using HiFiasm</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="5-oxford-nanopore-assembly.html">5. Oxford Nanopore Assembly using Flye</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="6-hybrid-assembly.html">6. Hybrid Long Read Assembly (optional)</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="7-scaffolding-assemblies.html">7. Scaffolding using Optical Genome Mapping</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="8-assembly-evaluation.html">8. Assembly Assessment</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="9-annotation-strategies.html">9. Annotation Strategies</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="10-gene-annotation-with-braker.html">10. Annotation using BRAKER</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="11-gene-annotation-with-helixer.html">11. Annotation using Helixer</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush13">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading13">
        <a href="12-gene-annotation-with-gemoma.html">12. Homology-based gene annotation with GeMoMa</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush14">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading14">
        <a href="13-functional-annotation-with-entap.html">13. Functional annotation using EnTAP</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush15">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading15">
        <a href="14-annotation-assesment.html">14. Annotation Assesment</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/1-introduction.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/3-data-quality-control.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/1-introduction.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Introduction to
        </a>
        <a class="chapter-link float-end" href="../instructor/3-data-quality-control.html" rel="next">
          Next: Data Quality Control...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Assembly Strategies</h1>
        <p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/2-genome-assembly-strategies.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>What factors influence the choice of genome assembly strategy?</li>
<li>How do different assembly methods compare in terms of read length,
accuracy, and computational requirements?</li>
<li>What are the key steps in evaluating genome assemblies using BUSCO
and QUAST?</li>
<li>How do Bionano OGM and Hi-C sequencing improve genome continuity and
organization?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Understand factors influencing the choice of genome assembly
strategy.</li>
<li>Compare different assembly methods based on read length, accuracy,
and computational requirements.</li>
<li>Learn how to evaluate genome assemblies using BUSCO and QUAST.</li>
<li>Explore the role of Bionano OGM and Hi-C sequencing in improving
genome continuity.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="assembly-strategies">Assembly Strategies<a class="anchor" aria-label="anchor" href="#assembly-strategies"></a></h2>
<hr class="half-width"><p>Genome assembly involves choosing the right approach based on
sequencing technology, read type, genome complexity, and research
objectives. This chapter introduces key factors influencing assembly
strategy selection, from read length and coverage requirements to
computational trade-offs. We will explore different methods—PacBio HiFi
with HiFiasm, ONT with Flye, and hybrid assemblies—along with
scaffolding techniques like Bionano Optical Genome Mapping (OGM) and
Hi-C, which help improve genome continuity and organization. Finally,
we’ll discuss assembly evaluation tools such as BUSCO and QUAST to
assess the completeness and quality of assembled genomes.</p>
<div id="factors-influencing-the-choice-of-strategy" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="factors-influencing-the-choice-of-strategy" class="callout-inner">
<h3 class="callout-title">Factors Influencing the Choice of Strategy</h3>
<div class="callout-content">
<ul><li>
<strong>Read length</strong>: Affects the ability to resolve
repeats; short reads struggle with complex regions, while long reads
improve contiguity.<br></li>
<li>
<strong>Coverage depth</strong>: Crucial for assembly accuracy, with
HiFi requiring 20-30x, ONT needing 50-100x, and hybrid assemblies
depending on short-read support for polishing.<br></li>
<li>
<strong>Genome complexity</strong>: Includes repeat content,
heterozygosity, and polyploidy, which influence assembly success and
determine whether specialized tools or additional scaffolding methods
are needed.<br></li>
<li>
<strong>Computational resources</strong>: Vary across assembly
strategies, with HiFiasm being RAM-intensive, Flye being more
lightweight, and hybrid assemblies requiring additional processing for
polishing.<br></li>
<li>
<strong>Sequencing budget</strong>: Plays a role, as HiFi sequencing
is costlier but highly accurate, ONT is cheaper but requires more data
for error correction, and hybrid approaches balance cost and
quality.<br></li>
<li>
<strong>Downstream analyses</strong>: Structural variation
detection, gene annotation, or chromosome-level assemblies influence the
choice of assembler and the need for scaffolding methods like Hi-C or
Bionano OGM.</li>
</ul></div>
</div>
</div>
</section><section><h2 class="section-heading" id="comparative-assembly-strategies">Comparative Assembly Strategies<a class="anchor" aria-label="anchor" href="#comparative-assembly-strategies"></a></h2>
<hr class="half-width"><table class="table"><colgroup><col width="25%"><col width="21%"><col width="24%"><col width="28%"></colgroup><thead><tr class="header"><th>Factor</th>
<th>PacBio HiFi</th>
<th>ONT</th>
<th>Hybrid (ONT + Illumina)</th>
</tr></thead><tbody><tr class="odd"><td><strong>Read length</strong></td>
<td>~15-20kb</td>
<td>10-100kb+</td>
<td>Mix of short &amp; long</td>
</tr><tr class="even"><td><strong>Read accuracy</strong></td>
<td>High (~99%)</td>
<td>Moderate (~90%)</td>
<td>High (after polishing)</td>
</tr><tr class="odd"><td><strong>Coverage needed</strong></td>
<td>20-30x</td>
<td>50-100x</td>
<td>ONT: 50x + Illumina: 30x</td>
</tr><tr class="even"><td><strong>Cost per Gb</strong></td>
<td>Expensive</td>
<td>Lower</td>
<td>Medium</td>
</tr><tr class="odd"><td><strong>Error profile</strong></td>
<td>Random errors, low indels</td>
<td>Higher error rate, systematic errors</td>
<td>ONT errors corrected by Illumina</td>
</tr><tr class="even"><td><strong>Computational requirements</strong></td>
<td>High RAM required</td>
<td>Moderate RAM required</td>
<td>Moderate</td>
</tr><tr class="odd"><td><strong>Best for</strong></td>
<td>High-accuracy assemblies</td>
<td>Ultra-long contigs</td>
<td>Combines advantages of both</td>
</tr><tr class="even"><td><strong>Repeat resolution</strong></td>
<td>Good</td>
<td>Very good</td>
<td>Very good</td>
</tr><tr class="odd"><td><strong>Scaffolding needed</strong></td>
<td>Rarely needed</td>
<td>May be needed</td>
<td>Sometimes needed</td>
</tr><tr class="even"><td><strong>Polishing required</strong></td>
<td>Not required</td>
<td>Required (Racon + Medaka)</td>
<td>Required (Pilon)</td>
</tr><tr class="odd"><td><strong>Structural variant detection</strong></td>
<td>Good</td>
<td>Excellent</td>
<td>Good</td>
</tr><tr class="even"><td><strong>Haplotype phasing</strong></td>
<td>Excellent</td>
<td>Good</td>
<td>Moderate</td>
</tr><tr class="odd"><td><strong>Genome size suitability</strong></td>
<td>Suitable for large and small genomes</td>
<td>Best for large genomes</td>
<td>Best for complex genomes</td>
</tr><tr class="even"><td><strong>Downstream applications</strong></td>
<td>Reference-quality genome assembly, annotation</td>
<td>Structural variation analysis, de novo assembly</td>
<td>Genome correction, variant calling, scaffolding</td>
</tr></tbody></table></section><section><h2 class="section-heading" id="contig-vs--scaffold-vs--chromosome-level-assembly">Contig vs. Scaffold vs. Chromosome-Level Assembly<a class="anchor" aria-label="anchor" href="#contig-vs--scaffold-vs--chromosome-level-assembly"></a></h2>
<hr class="half-width"><p>Genome assemblies progress through different levels of completeness
and organization:</p>
<ul><li>
<strong>Contig-level assembly</strong>: The raw output of
assemblers, consisting of contiguous sequences without known order or
orientation. Longer contigs indicate better assembly continuity.<br></li>
<li>
<strong>Scaffold-level assembly</strong>: Contigs linked together
using additional data (e.g., long-range mate-pair reads, optical maps,
or Hi-C). Gaps (represented as ’N’s) remain where connections exist but
sequence information is missing.<br></li>
<li>
<strong>Chromosome-level assembly</strong>: The highest-quality
assembly, where scaffolds are further ordered and oriented into full
chromosomes using genetic maps, Hi-C data, or synteny with a reference
genome.</li>
</ul><p>Higher levels of assembly provide better genome context, but require
additional scaffolding methods beyond de novo assembly.</p>
<figure><img src="https://github.com/user-attachments/assets/4b7406b6-747a-4a34-b816-e1de2cf444a9" alt="Assembly levels" class="figure mx-auto d-block"><div class="figcaption">Assembly levels</div>
</figure></section><section><h2 class="section-heading" id="workflow-for-various-assemblies">Workflow for Various Assemblies<a class="anchor" aria-label="anchor" href="#workflow-for-various-assemblies"></a></h2>
<hr class="half-width"><p>In this workshop, we will use HiFiasm for PacBio HiFi assemblies,
Flye for ONT assemblies, and Flye in hybrid mode for ONT + Illumina
assemblies, followed by quality assessment using BUSCO and QUAST to
evaluate completeness and accuracy.</p>
<div class="section level3">
<h3 id="pacbio-hifi-assembly-with-hifiasm">PacBio HiFi Assembly with HiFiasm<a class="anchor" aria-label="anchor" href="#pacbio-hifi-assembly-with-hifiasm"></a></h3>
<ul><li>
<strong>PacBio HiFi reads</strong>: Highly accurate long reads with
low error rates, suitable for de novo assembly without polishing.</li>
<li>
<strong>HiFiasm</strong>: A specialized assembler for HiFi data,
leveraging read accuracy and length to resolve complex regions and
produce high-quality contigs.</li>
<li>
<strong>Workflow</strong>: Run HiFiasm with HiFi reads, adjust
parameters based on genome size and complexity, and evaluate the
assembly using Compleasm and QUAST.</li>
</ul><figure><img src="https://github.com/user-attachments/assets/1f87d81a-707b-44ff-a8e1-49648f61de95" alt="PacBio Assembly" class="figure mx-auto d-block"><div class="figcaption">PacBio Assembly</div>
</figure></div>
<div class="section level3">
<h3 id="ont-assembly-with-flye">ONT Assembly with Flye<a class="anchor" aria-label="anchor" href="#ont-assembly-with-flye"></a></h3>
<ul><li>
<strong>ONT reads</strong>: Ultra-long reads with higher error
rates, requiring additional error correction and polishing steps.</li>
<li>
<strong>Flye</strong>: A de novo assembler optimized for long reads,
capable of resolving complex repeats and generating high-quality
assemblies.</li>
<li>
<strong>Workflow</strong>: Run Flye with ONT reads, adjust
parameters based on genome size and complexity, and polish the assembly
using Medaka for basecalling and consensus polishing.</li>
</ul><figure><img src="https://github.com/user-attachments/assets/9864ae7d-a893-4aec-ad4a-ea57170c8c27" alt="ONT assembly" class="figure mx-auto d-block"><div class="figcaption">ONT assembly</div>
</figure></div>
<div class="section level3">
<h3 id="hybrid-ont-pacbio-assembly-with-flye">Hybrid (ONT + PacBio) Assembly with Flye<a class="anchor" aria-label="anchor" href="#hybrid-ont-pacbio-assembly-with-flye"></a></h3>
<ul><li>
<strong>Hybrid assembly</strong>: Combines the strengths of both
technologies for improved accuracy and contiguity.</li>
<li>
<strong>Workflow</strong>: Run Flye with both ONT and PacBio reads,
adjust parameters for hybrid mode, and polish the assembly using ONT or
PacBio reads for error correction and consensus polishing.</li>
</ul><figure><img src="https://github.com/user-attachments/assets/9b51523d-2358-4ad3-b6b3-bfe5d0189e4d" alt="Hybrid assembly" class="figure mx-auto d-block"><div class="figcaption">Hybrid assembly</div>
</figure></div>
</section><section><h2 class="section-heading" id="assembly-evaluation">Assembly Evaluation<a class="anchor" aria-label="anchor" href="#assembly-evaluation"></a></h2>
<hr class="half-width"><ul><li>Why assessing genome assembly quality is crucial before downstream
analyses.<br></li>
<li>Metrics to determine assembly completeness, accuracy, and
contiguity.</li>
</ul><p><strong>1. BUSCO (Benchmarking Universal Single-Copy Orthologs)/
Compleasm</strong></p>
<ul><li><p><a href="https://github.com/huangnengCSU/compleasm" class="external-link">Compleasm</a>
is a faster alternative to BUSCO for assessing genome completeness based
on single-copy orthologs. It evaluates genome completeness by checking
for highly conserved, single-copy genes expected to be present in nearly
all members of a given lineage. These genes are essential for basic
cellular functions, making them reliable markers for assessing genome
assembly quality.</p></li>
<li><p>We expect these genes to be present in our organism because they
are evolutionarily conserved and critical for survival. If many BUSCO
genes are missing or fragmented, it suggests gaps, misassemblies, or
sequencing errors, which can compromise downstream analyses like gene
annotation and functional studies. A high BUSCO completeness score
indicates a well-assembled genome with minimal missing data.</p></li>
<li><p>The BUSCO reports provide detailed statistics on the number of
complete, fragmented, and missing BUSCO genes, as well as the percentage
of genome completeness. The output helps identify areas for improvement
and guide further optimization steps in the assembly process.</p></li>
</ul><figure><img width="1030" alt="BUSCO genes" src="https://github.com/user-attachments/assets/e4c7951f-f4f1-4395-974c-abd6716d8cd1" class="figure mx-auto d-block"></figure><p><strong>2. QUAST (Quality Assessment Tool for Genome
Assemblies)</strong></p>
<ul><li><p><strong>Comprehensive Assembly Statistics</strong>: QUAST
provides detailed metrics beyond N50 and L50, including <strong>total
number of contigs, GC content, genome size estimation, and misassembly
rates</strong>, allowing in-depth evaluation of genome continuity and
structure.</p></li>
<li><p><strong>Reference-Based and Reference-Free Evaluation</strong>:
It can assess assemblies against a reference genome (identifying
misassemblies, inversions, and duplications) or work in
<strong>reference-free mode</strong>, making it useful for de novo
assemblies without a known genome sequence.</p></li>
<li><p><strong>Structural Error Detection and Gene Feature
Analysis</strong>: QUAST integrates gene annotation tools like
<strong>BUSCO and GeneMark</strong>, highlights misassemblies based on
alignment breaks, and detects <strong>gaps, relocations, and
translocations</strong>, making it particularly useful for validating
scaffolding approaches and hybrid assemblies.</p></li>
</ul><p><strong>3. Merqury: K-mer based assembly evaluation</strong></p>
<ul><li>
<strong>Assembly Accuracy Check</strong>: Merqury compares k-mers
from sequencing reads to the assembled genome, identifying mismatches,
missing k-mers, and sequencing errors without requiring a reference
genome.<br></li>
<li>
<strong>Haplotype Purity and Phasing</strong>: It calculates
<strong>QV (quality value)</strong> scores and provides
<strong>completeness metrics for haplotypes</strong>, helping assess
whether an assembly accurately represents both parental haplotypes or
contains chimeric sequences.<br></li>
<li>
<strong>Consensus and Read Support Validation</strong>: By analyzing
k-mer spectra, Merqury detects <strong>underrepresented or
overrepresented regions</strong>, highlighting assembly errors,
collapsed repeats, or sequencing biases that may impact downstream
analyses.</li>
</ul><figure><img src="https://github.com/user-attachments/assets/042f0ad9-e06e-4aa0-8faf-51f0d42e2983" alt="kmer spectra" class="figure mx-auto d-block"><div class="figcaption">kmer spectra</div>
</figure></section><section><h2 class="section-heading" id="bionano-and-hi-c-reads-in-genome-assembly">Bionano and Hi-C Reads in Genome Assembly<a class="anchor" aria-label="anchor" href="#bionano-and-hi-c-reads-in-genome-assembly"></a></h2>
<hr class="half-width"><p><strong>Bionano Optical Genome Mapping (OGM)</strong> provides
ultra-long, label-based maps of DNA molecules, helping to scaffold
contigs, detect misassemblies, and resolve large structural variations.
It improves genome continuity by linking fragmented sequences,
especially in repeat-rich or complex genomes.</p>
<p><strong>Hi-C sequencing</strong> captures chromatin interactions,
allowing scaffolding of contigs into chromosome-scale assemblies based
on physical proximity in the nucleus. It helps in ordering and orienting
scaffolds, identifying misassemblies, and resolving haplotypes, making
it essential for generating <strong>chromosome-level genome
assemblies</strong>.</p>
<figure><img src="https://github.com/user-attachments/assets/7f983713-7ea7-4522-8540-349955a9f12e" alt="Bionano and HiC for assembly improvement" class="figure mx-auto d-block"><div class="figcaption">Bionano and HiC for assembly improvement</div>
</figure><p><a href="https://doi.org/10.1038/ng.3802" class="external-link">ref</a></p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>Genome assembly strategy depends on read type, genome complexity,
and computational resources, with PacBio HiFi, ONT, and hybrid
approaches offering different advantages in accuracy, cost, and
contiguity.<br></li>
<li>Assembly evaluation is critical for assessing completeness and
accuracy, using tools like <strong>BUSCO for gene completeness, QUAST
for structural integrity, and Merqury for k-mer-based
validation</strong>.<br></li>
<li>Scaffolding methods like Bionano OGM and Hi-C improve genome
organization, resolving large structural variations and ordering contigs
into chromosome-level assemblies.<br></li>
<li>A well-assembled genome is essential for downstream applications
such as annotation, comparative genomics, and structural variation
analysis, with missing or misassembled regions potentially leading to
incorrect biological conclusions.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/1-introduction.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/3-data-quality-control.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/1-introduction.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Introduction to
        </a>
        <a class="chapter-link float-end" href="../instructor/3-data-quality-control.html" rel="next">
          Next: Data Quality Control...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/2-genome-assembly-strategies.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://rcac-bioinformatics.github.io/maize-genomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://rcac-bioinformatics.github.io/maize-genomics/" class="external-link">Source</a></p>
				<p><a href="https://rcac-bioinformatics.github.io/maize-genomics/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:rcac-help@purdue.edu">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.11" class="external-link">sandpaper (0.16.11)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/aseetharam/varnish/tree/db4f5c6f3c4799dfe8247fd057642c8099d5244b" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://maize-genomics.github.io/NA/instructor/2-genome-assembly-strategies.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene-annotation, braker, helixer, gemoma, entap, omark, busco, hifiasm",
  "name": "Assembly Strategies",
  "creativeWorkStatus": "active",
  "url": "https://maize-genomics.github.io/NA/instructor/2-genome-assembly-strategies.html",
  "identifier": "https://maize-genomics.github.io/NA/instructor/2-genome-assembly-strategies.html",
  "dateCreated": "2025-02-05",
  "dateModified": "2025-03-01",
  "datePublished": "2025-03-01"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

