<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Maize genome assembly and annotation: Hands-on training: Data Quality Control</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/rcac/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicons/rcac/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicons/rcac/favicon-16x16.png"><link rel="manifest" href="../favicons/rcac/site.webmanifest"><link rel="mask-icon" href="../favicons/rcac/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav rcac"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="rcac-logo" alt="rcac" src="../assets/images/rcac-logo.svg"></div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../3-data-quality-control.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav rcac" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="rcac" src="../assets/images/rcac-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Maize genome assembly and annotation: Hands-on training
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Maize genome assembly and annotation: Hands-on training
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Maize genome assembly and annotation: Hands-on training
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 14%" class="percentage">
    14%
  </div>
  <div class="progress rcac">
    <div class="progress-bar rcac" role="progressbar" style="width: 14%" aria-valuenow="14" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../3-data-quality-control.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="1-introduction.html">1. Introduction to Genome assembly</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="2-genome-assembly-strategies.html">2. Assembly Strategies</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        3. Data Quality Control
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#data-quality-check-and-filtering">Data Quality Check and Filtering</a></li>
<li><a href="#pacbio-hifi-reads-from-subreads">PacBio HiFi reads from Subreads</a></li>
<li><a href="#ont-reads-from-minion-sequencing">ONT Reads from MinION Sequencing</a></li>
<li><a href="#a.-nanoplot-for-quality-assessment">A. NanoPlot for Quality Assessment</a></li>
<li><a href="#b.-filtering-sequencing-reads">B. Filtering Sequencing Reads</a></li>
<li><a href="#c.-evaluating-data-quality-after-filtering">C. Evaluating Data Quality After Filtering</a></li>
<li><a href="#d.-k-mer-based-quality-checks-optional">D. K-mer Based Quality Checks (Optional)</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="4-pacbio-hifi-assembly.html">4. PacBio HiFi Assembly using HiFiasm</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="5-oxford-nanopore-assembly.html">5. Oxford Nanopore Assembly using Flye</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="6-hybrid-assembly.html">6. Hybrid Long Read Assembly (optional)</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="7-scaffolding-assemblies.html">7. Scaffolding using Optical Genome Mapping</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="8-assembly-evaluation.html">8. Assembly Assessment</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="9-annotation-strategies.html">9. Annotation Strategies</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="10-gene-annotation-with-braker.html">10. Annotation using BRAKER</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="11-gene-annotation-with-helixer.html">11. Annotation using Helixer</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush13">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading13">
        <a href="12-gene-annotation-with-gemoma.html">12. Homology-based gene annotation with GeMoMa</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush14">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading14">
        <a href="13-functional-annotation-with-entap.html">13. Functional annotation using EnTAP</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush15">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading15">
        <a href="14-annotation-assesment.html">14. Annotation Assesment</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/2-genome-assembly-strategies.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/4-pacbio-hifi-assembly.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/2-genome-assembly-strategies.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Assembly Strategies
        </a>
        <a class="chapter-link float-end" href="../instructor/4-pacbio-hifi-assembly.html" rel="next">
          Next: PacBio HiFi Assembly...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Data Quality Control</h1>
        <p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/3-data-quality-control.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>What is data quality checking and filtering?</li>
<li>Why is it necessary to assess the quality of raw sequencing
data?</li>
<li>What are the key steps in filtering long-read sequencing data?</li>
<li>How can visualization tools like NanoPlot help in quality
assessment?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Understand the importance of data quality checking and filtering in
genome assembly.</li>
<li>Learn how to assess raw sequencing data quality using NanoPlot.</li>
<li>Gain hands-on experience in filtering long-read sequencing data with
Filtlong.</li>
<li>Evaluate the impact of filtering on data quality using
NanoPlot.</li>
<li>Explore k-mer analysis for quality assessment (optional).</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="data-quality-check-and-filtering">Data Quality Check and Filtering<a class="anchor" aria-label="anchor" href="#data-quality-check-and-filtering"></a></h2>
<hr class="half-width"><ul><li>
<strong>What is data quality checking and filtering?</strong>
<ul><li>Assessing the quality of raw sequencing data before genome
assembly.</li>
<li>Identifying and removing low-quality or problematic reads before
assembly.<br></li>
<li>Ensuring that the data is suitable for downstream analysis.</li>
<li>Better quality ingredients make a better quality cake!</li>
</ul></li>
<li>
<strong>Why is it necessary?</strong>
<ul><li>Poor quality data can lead to errors in genome assembly.</li>
<li>Low-quality reads can introduce gaps, misassemblies, or incorrect
base calls.</li>
<li>Filtering out low-quality reads can improve the accuracy and
efficiency of assembly.</li>
<li>It is a critical step to ensure the success of downstream
analyses.</li>
</ul></li>
<li>
<strong>What we will do today?</strong>
<ul><li>Learn the process of long-read generation (PacBio HiFi and ONT) - a
brief overview.</li>
<li>Assess raw data quality using <code>NanoPlot</code> for PacBio HiFi
and ONT reads - hands-on.<br></li>
<li>Filter reads based on quality and length using <code>Filtlong</code>
- hands-on.</li>
<li>Re-evaluate data post-filtering using <code>NanoPlot</code> to
confirm improvements - hands-on.<br></li>
<li>K-mer analysis to ensure data quality (optional hands-on).</li>
</ul></li>
</ul><figure><img src="https://github.com/user-attachments/assets/21f98d7a-21b0-4e4d-aa48-b1b8edbb9dfc" alt="Overview of steps" class="figure mx-auto d-block"><div class="figcaption">Overview of steps</div>
</figure></section><section><h2 class="section-heading" id="pacbio-hifi-reads-from-subreads">PacBio HiFi reads from Subreads<a class="anchor" aria-label="anchor" href="#pacbio-hifi-reads-from-subreads"></a></h2>
<hr class="half-width"><p><strong>PacBio HiFi</strong> reads can be generated using <a href="https://ccs.how/" class="external-link">Circular Consensus Sequencing</a> (CCS) program
from <strong>PacBio</strong></p>
<ul><li>PacBio circular consensus sequencing (CCS) generates HiFi reads by
sequencing the same DNA molecule multiple times<br></li>
<li>The more passes over the same molecule, the higher the consensus
accuracy<br></li>
<li>Two main stages:
<ul><li>Generate a draft consensus from multiple subreads<br></li>
<li>Iteratively polish the consensus using all subreads to refine
accuracy</li>
</ul></li>
</ul><figure><img src="https://github.com/user-attachments/assets/566db0d0-93e8-40a3-82ce-89ec79c7a8b4" alt="PacBio CCS overview" class="figure mx-auto d-block"><div class="figcaption">PacBio CCS overview</div>
</figure><p><strong>What does CCS do?</strong></p>
<ol style="list-style-type: decimal"><li>Filter low-quality subreads based on length and signal-to-noise
ratio<br></li>
<li>Generate an initial consensus using overlapping subreads<br></li>
<li>Align all subreads to the draft consensus for refinement<br></li>
<li>Divide sequence into small overlapping windows to optimize
polishing<br></li>
<li>Identify and correct errors, heteroduplex artifacts, and large
insertions<br></li>
<li>Apply polishing algorithms to refine the sequence, removing
ambiguities<br></li>
<li>Compute read accuracy based on error likelihood<br></li>
<li>Output the final HiFi read if accuracy meets the threshold</li>
</ol><p><strong>Why is this important?</strong></p>
<ul><li>Produces highly accurate long reads (99 percent or higher)<br></li>
<li>Enables better genome assembly, especially in complex or repetitive
regions<br></li>
<li>Reduces the need for additional error correction</li>
</ul><p><strong>How was the data generated?</strong></p>
<p>Subreads (in bam format) were converted to ccs fastq as follows:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">ccs</span> <span class="dt">\</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>   <span class="at">--hifi-kinetics</span> <span class="dt">\</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>   <span class="at">--num-threads</span> <span class="va">$SLURM_CPUS_ON_NODE</span> <span class="dt">\</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>   input.subreads.bam <span class="dt">\</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>   output.hifi.bam</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="ex">samtools</span> fastq <span class="dt">\</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>   output.hifi.bam <span class="op">&gt;</span> output.hifi.fastq</span></code></pre>
</div>
<p><strong>What is the source of this data?</strong></p>
<ul><li>The PacBio HiFi reads are from the project <a href="https://www.ebi.ac.uk/ena/browser/view/PRJEB50694" class="external-link">PRJEB50694</a>.</li>
<li>Data is for <em>Arabidopsis thaliana</em> ecotype Col-0, sequenced
using PacBio HiFi technology (they also sequenced CLR data for this
project).</li>
<li>The data is publicly available on the European Nucleotide Archive
(ENA), and the <code>9994.q20.CCS.fastq.gz</code> reads were used for
analysis.</li>
<li>Data has been filtered to include only HiFi reads with a Q-score of
20 or higher.</li>
</ul></section><section><h2 class="section-heading" id="ont-reads-from-minion-sequencing">ONT Reads from MinION Sequencing<a class="anchor" aria-label="anchor" href="#ont-reads-from-minion-sequencing"></a></h2>
<hr class="half-width"><p><strong>Oxford Nanopore Technologies (ONT)</strong> sequencing
generates long reads by passing DNA through a biological nanopore</p>
<ul><li>ONT reads are generated in real-time as the DNA strand moves through
the nanopore.</li>
<li>Dorado is Oxford Nanopore Technologies’ basecaller that converts raw
electrical signals from nanopores into nucleotide sequences using
machine learning.</li>
</ul><figure><img src="https://github.com/user-attachments/assets/624b4209-5f34-4d03-ad12-de95ae85ad75" alt="Overview of ONT read generation" class="figure mx-auto d-block"><div class="figcaption">Overview of ONT read generation</div>
</figure><p><strong>Key steps in <code>dorado</code> base calling</strong></p>
<ol style="list-style-type: decimal"><li>
<strong>Signal detection</strong>
<ul><li>DNA or RNA molecules pass through a nanopore, disrupting an
electrical current.<br></li>
<li>These disruptions create a unique signal pattern called a
<strong>squiggle</strong>.</li>
</ul></li>
<li>
<strong>Real-time data processing</strong>
<ul><li>The MinKNOW software captures and processes the squiggle into
sequencing reads.<br></li>
<li>Reads include standard nucleotides and potential base modifications
like methylation.</li>
</ul></li>
<li>
<strong>Basecalling with machine learning</strong>
<ul><li>Neural networks, including transformer models, predict the
nucleotide sequence from raw signals.<br></li>
<li>Models continuously improve by training on diverse sequencing
data.</li>
</ul></li>
<li>
<strong>Error correction and refinement</strong>
<ul><li>Dorado refines base predictions to reduce errors, especially in
homopolymer regions.<br></li>
<li>Models are optimized for accuracy across various DNA/RNA types.</li>
</ul></li>
<li>
<strong>High-speed processing</strong>
<ul><li>Basecalling can be performed <strong>during sequencing</strong> for
real-time analysis or <strong>after sequencing</strong> for higher
accuracy.<br></li>
<li>GPUs accelerate computation, enabling rapid basecalling and
simultaneous modification detection.</li>
</ul></li>
</ol><p><strong>Why this matters</strong></p>
<ul><li>Enables <strong>real-time sequencing and analysis</strong> for quick
decision-making.<br></li>
<li>Uses <strong>advanced machine learning</strong> to improve sequence
accuracy over time.<br></li>
<li>Supports <strong>epigenetic modification detection</strong> without
extra processing steps.</li>
</ul><p><strong>How was the data generated?</strong></p>
<p><code>pod5</code> ONT reads were basecalled using Dorado as
follows:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># download</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">wget</span> ftp.sra.ebi.ac.uk/vol1/run/ERR791/ERR7919757/Arabidopsis-pass.tar.gz</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co"># extract</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">tar</span> <span class="at">-xvf</span> Arabidopsis-pass.tar.gz <span class="co"># fast5_pass is the extracted directory </span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co"># convert to pod5 format</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="ex">pod5</span> convert <span class="dt">\</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>   fast5 fast5_pass <span class="at">--output</span> pod5_pass</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co"># download model</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="ex">dorado</span> download <span class="dt">\</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>   <span class="at">--model</span> <span class="st">"dna_r10.4.1_e8.2_400bps_hac@v3.5.2"</span> <span class="dt">\</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>   <span class="at">--models-directory</span> models_dir/</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co"># basecall</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="ex">dorado</span> basecaller <span class="dt">\</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>   <span class="at">--emit-fastq</span> <span class="dt">\</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>   <span class="at">--output-dir</span> dorado_output_dir <span class="dt">\</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>   models_dir/dna_r10.4.1_e8.2_400bps_hac@v3.5.2 <span class="dt">\</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a>   input_pass.pod5</span></code></pre>
</div>
<p><strong>What is the source of this data?</strong></p>
<ul><li>The ONT reads are from the project <a href="https://www.ebi.ac.uk/ena/browser/view/PRJEB49840" class="external-link">PRJEB49840</a>.</li>
<li>Data is for <em>Arabidopsis thaliana</em> ecotype Col-0, sequenced
using R10.4/Q20+ chemistry from MinION cell</li>
<li>The data is publicly available on the European Nucleotide Archive
(ENA), and the <code>pass_fast5</code> reads were used for basecalling
(with commands above).</li>
</ul></section><section><h2 class="section-heading" id="a--nanoplot-for-quality-assessment">A. NanoPlot for Quality Assessment<a class="anchor" aria-label="anchor" href="#a--nanoplot-for-quality-assessment"></a></h2>
<hr class="half-width"><p><a href="https://github.com/wdecoster/NanoPlot" class="external-link">NanoPlot</a> <a href="https://doi.org/10.1093/bioinformatics/bty149" class="external-link">ref</a> is a
visualization tool designed for quality assessment of long-read
sequencing data. It generates a variety of plots, including read length
histograms, cumulative yield plots, violin plots of read length and
quality over time, and bivariate plots that compare read lengths,
quality scores, reference identity, and mapping quality. By providing
both single-variable and density-based visualizations, NanoPlot helps
users quickly assess sequencing run quality and detect potential issues.
The tool also allows downsampling, length and quality filtering, and
barcode-specific analysis for multiplexed experiments.</p>
<p><strong>1. Quality Assessment of PacBio HiFi Reads</strong></p>
<p>Assessing the quality of ONT reads using <code>NanoPlot</code>.
Create a slurm script to run NanoPlot on the HiFi reads.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="ex">ml</span> nanoplot</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">NanoPlot</span> <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>   <span class="at">--verbose</span> <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>   <span class="at">--outdir</span> nanoplot_pacbio_pre <span class="dt">\</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>   <span class="at">--prefix</span> At_PacBio_ <span class="dt">\</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>   <span class="at">--plots</span> kde <span class="dt">\</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>   <span class="at">--N50</span> <span class="dt">\</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>   <span class="at">--dpi</span> 300 <span class="dt">\</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>   <span class="at">--fastq</span> At_pacbio-hifi.fastq.gz</span></code></pre>
</div>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Show details </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" aria-labelledby="headingSpoiler1" data-bs-parent="#accordionSpoiler1">
<div class="accordion-body">
<p>The stdout from the NanoPlot run will look like this:</p>
<pre><code>2025-02-13 12:13:19,155 NanoPlot 1.44.1 started with arguments Namespace(threads=16, verbose=True, store=False, raw=False, huge=False, outdir='nanoplot_pacbio_output', no_static=False, prefix='At_PacBio_', tsv_stats=False, only_report=False, info_in_report=False, maxlength=None, minlength=None, drop_outliers=False, downsample=None, loglength=False, percentqual=False, alength=False, minqual=None, runtime_until=None, readtype='1D', barcoded=False, no_supplementary=False, color='#4CB391', colormap='Greens', format=['png'], plots=['kde'], legacy=None, listcolors=False, listcolormaps=False, no_N50=False, N50=True, title=None, font_scale=1, dpi=300, hide_stats=False, fastq=['9994.q20.CCS.fastq.gz'], fasta=None, fastq_rich=None, fastq_minimal=None, summary=None, bam=None, ubam=None, cram=None, pickle=None, feather=None, path='nanoplot_pacbio_output/At_PacBio_')
2025-02-13 12:13:19,156 Python version is: 3.9.21 | packaged by conda-forge | (main, Dec  5 2024, 13:51:40)  [GCC 13.3.0]
2025-02-13 12:13:19,186 Nanoget: Starting to collect statistics from plain fastq file.
2025-02-13 12:13:19,187 Nanoget: Decompressing gzipped fastq 9994.q20.CCS.fastq.gz
2025-02-13 12:29:10,170 Reduced DataFrame memory usage from 12.780670166015625Mb to 12.780670166015625Mb
2025-02-13 12:29:10,194 Nanoget: Gathered all metrics of 837586 reads
2025-02-13 12:29:10,538 Calculated statistics
2025-02-13 12:29:10,539 Using sequenced read lengths for plotting.
2025-02-13 12:29:10,556 NanoPlot:  Valid color #4CB391.
2025-02-13 12:29:10,557 NanoPlot:  Valid colormap Greens.
2025-02-13 12:29:10,582 NanoPlot:  Creating length plots for Read length.
2025-02-13 12:29:10,583 NanoPlot: Using 837586 reads with read length N50 of 22587bp and maximum of 57055bp.
2025-02-13 12:29:11,933 Saved nanoplot_pacbio_output/At_PacBio_WeightedHistogramReadlength  as png (or png for --legacy)
2025-02-13 12:29:12,443 Saved nanoplot_pacbio_output/At_PacBio_WeightedLogTransformed_HistogramReadlength  as png (or png for --legacy)
2025-02-13 12:29:12,899 Saved nanoplot_pacbio_output/At_PacBio_Non_weightedHistogramReadlength  as png (or png for --legacy)
2025-02-13 12:29:13,371 Saved nanoplot_pacbio_output/At_PacBio_Non_weightedLogTransformed_HistogramReadlength  as png (or png for --legacy)
2025-02-13 12:29:13,372 NanoPlot: Creating yield by minimal length plot for Read length.
2025-02-13 12:29:14,465 Saved nanoplot_pacbio_output/At_PacBio_Yield_By_Length  as png (or png for --legacy)
2025-02-13 12:29:14,466 Created length plots
2025-02-13 12:29:14,474 NanoPlot: Creating Read lengths vs Average read quality plots using 837586 reads.
2025-02-13 12:29:15,012 Saved nanoplot_pacbio_output/At_PacBio_LengthvsQualityScatterPlot_kde  as png (or png for --legacy)
2025-02-13 12:29:15,013 Created LengthvsQual plot
2025-02-13 12:29:15,013 Writing html report.
2025-02-13 12:29:15,029 Finished!
</code></pre>
</div>
</div>
</div>
</div>
<p><strong>Evaluate the quality of HiFi reads:</strong></p>
<p>Examine the <code>At_PacBio_NanoPlot-report.html</code> file</p>
<ul><li>
<strong>Read length distribution</strong>: Histogram of read
lengths, showing the distribution of read lengths in the dataset.</li>
<li>
<strong>Read length vs. Quality</strong>: Scatter plot showing the
relationship between read length and quality score.</li>
<li>
<strong>Yield (number of bases) by read length</strong>: Plot
showing the cumulative yield of reads based on their length.</li>
<li>
<strong>Log-Transformed histograms</strong>: Histogram of read
lengths with a log-transformed scale for better visualization.</li>
<li>
<strong>KDE plots</strong>: Kernel Density Estimation plots for read
length and quality score distributions.</li>
<li>
<strong>Summary statistics</strong>: N50 value, maximum read length,
and other key metrics.</li>
</ul><div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p><strong>What filtering should be applied to the PacBio HiFi reads
based on the quality assessment?</strong></p>
<p>Our genome (<em>A. thaliana</em>) has a genome size of ~135 Mb. Our
target coverage is ~40x. Currently we have ~18Gb of HiFi reads (~138X
depth of coverage). We need to filter the reads to ensure we have good
quality reads of desired length and coverage.</p>
</div>
</div>
</div>
<p><strong>2. Quality Assessment of ONT Reads</strong></p>
<p>Assessing the quality of ONT reads using <code>NanoPlot</code>.
Create a slurm script to run NanoPlot on the basecalled ONT reads.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="ex">ml</span> nanoplot</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="ex">NanoPlot</span> <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>   <span class="at">--verbose</span> <span class="dt">\</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>   <span class="at">--outdir</span> nanoplot_ont_pre <span class="dt">\</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>   <span class="at">--prefix</span> At_ONT_ <span class="dt">\</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>   <span class="at">--readtype</span> 1D <span class="dt">\</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>   <span class="at">--plots</span> kde <span class="dt">\</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>   <span class="at">--N50</span> <span class="dt">\</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>   <span class="at">--dpi</span> 300 <span class="dt">\</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>   <span class="at">--fastq</span> At_ont-reads.fastq.gz</span></code></pre>
</div>
<div id="accordionSpoiler2" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler2" aria-expanded="false" aria-controls="collapseSpoiler2">
  <h3 class="accordion-header" id="headingSpoiler2">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Show details </h3>
</button>
<div id="collapseSpoiler2" class="accordion-collapse collapse" aria-labelledby="headingSpoiler2" data-bs-parent="#accordionSpoiler2">
<div class="accordion-body">
<p>The stdout from the NanoPlot run will look like this:</p>
<pre><code>2025-02-13 12:15:51,066 NanoPlot 1.44.1 started with arguments Namespace(threads=8, verbose=True, store=False, raw=False, huge=False, outdir='nanoplot_pacbio_output', no_static=False, prefix='At_ONT_', tsv_stats=False, only_report=False, info_in_report=False, maxlength=None, minlength=None, drop_outliers=False, downsample=None, loglength=False, percentqual=False, alength=False, minqual=None, runtime_until=None, readtype='1D', barcoded=False, no_supplementary=False, color='#4CB391', colormap='Greens', format=['png'], plots=['kde'], legacy=None, listcolors=False, listcolormaps=False, no_N50=False, N50=True, title=None, font_scale=1, dpi=300, hide_stats=False, fastq=['basecalled_2025-02-12.fastq'], fasta=None, fastq_rich=None, fastq_minimal=None, summary=None, bam=None, ubam=None, cram=None, pickle=None, feather=None, path='nanoplot_pacbio_output/At_ONT_')
2025-02-13 12:15:51,067 Python version is: 3.9.21 | packaged by conda-forge | (main, Dec  5 2024, 13:51:40)  [GCC 13.3.0]
2025-02-13 12:15:51,096 Nanoget: Starting to collect statistics from plain fastq file.
2025-02-13 12:25:11,429 Reduced DataFrame memory usage from 8.842315673828125Mb to 8.842315673828125Mb
2025-02-13 12:25:11,455 Nanoget: Gathered all metrics of 579482 reads
2025-02-13 12:25:11,692 Calculated statistics
2025-02-13 12:25:11,693 Using sequenced read lengths for plotting.
2025-02-13 12:25:11,707 NanoPlot:  Valid color #4CB391.
2025-02-13 12:25:11,707 NanoPlot:  Valid colormap Greens.
2025-02-13 12:25:11,725 NanoPlot:  Creating length plots for Read length.
2025-02-13 12:25:11,725 NanoPlot: Using 579482 reads with read length N50 of 36292bp and maximum of 298974bp.
2025-02-13 12:25:13,096 Saved nanoplot_pacbio_output/At_ONT_WeightedHistogramReadlength  as png (or png for --legacy)
2025-02-13 12:25:13,571 Saved nanoplot_pacbio_output/At_ONT_WeightedLogTransformed_HistogramReadlength  as png (or png for --legacy)
2025-02-13 12:25:14,971 Saved nanoplot_pacbio_output/At_ONT_Non_weightedHistogramReadlength  as png (or png for --legacy)
2025-02-13 12:25:15,440 Saved nanoplot_pacbio_output/At_ONT_Non_weightedLogTransformed_HistogramReadlength  as png (or png for --legacy)
2025-02-13 12:25:15,441 NanoPlot: Creating yield by minimal length plot for Read length.
2025-02-13 12:25:16,485 Saved nanoplot_pacbio_output/At_ONT_Yield_By_Length  as png (or png for --legacy)
2025-02-13 12:25:16,486 Created length plots
2025-02-13 12:25:16,495 NanoPlot: Creating Read lengths vs Average read quality plots using 579482 reads.
2025-02-13 12:25:17,029 Saved nanoplot_pacbio_output/At_ONT_LengthvsQualityScatterPlot_kde  as png (or png for --legacy)
2025-02-13 12:25:17,030 Created LengthvsQual plot
2025-02-13 12:25:17,030 Writing html report.
2025-02-13 12:25:17,047 Finished!</code></pre>
</div>
</div>
</div>
</div>
<p><strong>Evaluate the quality of ONT reads:</strong></p>
<p>Examine the <code>At_ONT_NanoPlot-report.html</code> file.</p>
<ul><li>
<strong>Read length distribution</strong>: Histogram of read
lengths, showing the distribution of read lengths in the dataset.</li>
<li>
<strong>Read length vs. Quality</strong>: Scatter plot showing the
relationship between read length and quality score.</li>
<li>
<strong>Yield (number of bases) by read length</strong>: Plot
showing the cumulative yield of reads based on their length.</li>
<li>
<strong>Log-Transformed histograms</strong>: Histogram of read
lengths with a log-transformed scale for better visualization.</li>
<li>
<strong>KDE plots</strong>: Kernel Density Estimation plots for read
length and quality score distributions.</li>
<li>
<strong>Summary statistics</strong>: N50 value, maximum read length,
and other key metrics.</li>
</ul><div id="callout2" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p><strong>What filtering should be applied to the ONT reads based on
the quality assessment?</strong></p>
<p>Our genome (<em>A. thaliana</em>) has a genome size of ~135 Mb. Our
target coverage is ~40x. Currently we have ~14Gb of ONT reads (104X
depth of coverage). We need to filter the reads to ensure we have good
quality reads of desired length and coverage.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="b--filtering-sequencing-reads">B. Filtering Sequencing Reads<a class="anchor" aria-label="anchor" href="#b--filtering-sequencing-reads"></a></h2>
<hr class="half-width"><p><a href="https://github.com/rrwick/Filtlong" class="external-link"><code>Filtlong</code></a> is a
tool designed to filter long-read sequencing data by selecting a
smaller, higher-quality subset of reads based on length and identity. It
prioritizes longer reads with higher sequence identity while discarding
shorter or lower-quality reads, ensuring that the retained data
contributes to more accurate genome assemblies. This filtering step is
crucial for improving assembly contiguity, reducing errors, and
optimizing computational efficiency by removing excess low-quality
data.</p>
<p><strong>1. Filtering PacBio HiFi Reads</strong></p>
<p>Filter the PacBio HiFi reads using <code>Filtlong</code> to retain
only high-quality reads.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="ex">ml</span> filtlong</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="ex">filtlong</span> <span class="dt">\</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>   <span class="at">--target_bases</span> 5400000000 <span class="dt">\</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>   <span class="at">--keep_percent</span> 90 <span class="dt">\</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>   <span class="at">--min_length</span> 1000 <span class="dt">\</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>     At_pacbio-hifi.fastq.gz <span class="op">&gt;</span> At_pacbio-hifi-filtered.fastq</span></code></pre>
</div>
<p><strong>2. Filtering ONT Reads</strong></p>
<p>Filter the ONT reads using <code>Filtlong</code> to retain only
high-quality reads.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="ex">ml</span> filtlong</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="ex">filtlong</span> <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>   <span class="at">--target_bases</span> 5400000000 <span class="dt">\</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>   <span class="at">--keep_percent</span> 90 <span class="dt">\</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>   <span class="at">--min_length</span> 1000 <span class="dt">\</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>     At_ont-reads.fastq.gz <span class="op">&gt;</span> At_ont-reads-filtered.fastq</span></code></pre>
</div>
<div id="callout3" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p><strong>What does this command do?</strong></p>
<ul><li>
<code>--target_bases 5400000000</code>: Target number of bases to
retain in the filtered dataset (5.4 Gb).</li>
<li>
<code>--keep_percent 90</code>: Retain reads that cover 90% of the
target bases.</li>
<li>
<code>--min_length 1000</code>: Minimum read length to keep in the
filtered dataset (1000 bp).</li>
</ul></div>
</div>
</div>
</section><section><h2 class="section-heading" id="c--evaluating-data-quality-after-filtering">C. Evaluating Data Quality After Filtering<a class="anchor" aria-label="anchor" href="#c--evaluating-data-quality-after-filtering"></a></h2>
<hr class="half-width"><p>We will re-run <code>NanoPlot</code> on the filtered HiFi and ONT
reads to assess the quality of the filtered datasets.</p>
<p><strong>1. For PacBio HiFi Reads:</strong></p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="ex">ml</span> nanoplot</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="ex">NanoPlot</span> <span class="dt">\</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>   <span class="at">--verbose</span> <span class="dt">\</span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>   <span class="at">--outdir</span> nanoplot_pacbio_post <span class="dt">\</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>   <span class="at">--prefix</span> At_PacBio_post_ <span class="dt">\</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>   <span class="at">--plots</span> kde <span class="dt">\</span></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>   <span class="at">--N50</span> <span class="dt">\</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>   <span class="at">--dpi</span> 300 <span class="dt">\</span></span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>   <span class="at">--fastq</span> At_pacbio-hifi-filtered.fastq</span></code></pre>
</div>
<p><strong>2. For ONT Reads:</strong></p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="ex">ml</span> nanoplot</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="ex">NanoPlot</span> <span class="dt">\</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>   <span class="at">--verbose</span> <span class="dt">\</span></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>   <span class="at">--outdir</span> nanoplot_ont_post <span class="dt">\</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>   <span class="at">--prefix</span> At_ONT_post_ <span class="dt">\</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>   <span class="at">--readtype</span> 1D <span class="dt">\</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>   <span class="at">--plots</span> kde <span class="dt">\</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>   <span class="at">--N50</span> <span class="dt">\</span></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a>   <span class="at">--dpi</span> 300 <span class="dt">\</span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a>   <span class="at">--fastq</span> At_ont-reads-filtered.fastq</span></code></pre>
</div>
<p>Now, examine the <code>At_PacBio_post_NanoPlot-report.html</code> and
<code>At_ONT_post_NanoPlot-report.html</code> files to assess the
quality of the filtered HiFi and ONT reads. Do you observe any
improvements in read quality after filtering? We will use these filtered
reads for downstream genome assembly.</p>
</section><section><h2 class="section-heading" id="d--k-mer-based-quality-checks-optional">D. K-mer Based Quality Checks (Optional)<a class="anchor" aria-label="anchor" href="#d--k-mer-based-quality-checks-optional"></a></h2>
<hr class="half-width"><p><a href="https://github.com/tbenavi1/genomescope2.0" class="external-link">GenomeScope</a>
is a k-mer-based tool used to profile genomes without requiring a
reference, providing estimates of genome size, heterozygosity, and
repeat content. It uses k-mer frequency distributions from raw
sequencing reads to model genome characteristics, making it especially
useful for detecting sequencing artifacts and assessing data quality
before assembly. In this optional section, we will use GenomeScope to
evaluate the quality of our Oxford Nanopore and PacBio reads by
identifying potential errors, biases, and coverage issues, helping to
refine filtering strategies and improve downstream assembly results <a href="https://www.nature.com/articles/s41467-020-14998-3" class="external-link">ref</a>.</p>
<p><strong>1. For PacBio HiFi Reads:</strong></p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="ex">ml</span> kmc</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="fu">mkdir</span> tmp</span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="fu">ls</span> At_pacbio-hifi-filtered.fastq <span class="op">&gt;</span> FILES</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="ex">kmc</span> <span class="at">-k21</span> <span class="at">-t10</span> <span class="at">-m64</span> <span class="at">-ci1</span> <span class="at">-cs10000</span> @FILES reads tmp/</span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="ex">kmc_tools</span> transform reads histogram reads-pacbio.histo <span class="at">-cx10000</span></span></code></pre>
</div>
<p><strong>2. For ONT Reads:</strong></p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="ex">ml</span> kmc</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="fu">mkdir</span> tmp</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="fu">ls</span> At_ont-reads-filtered.fastq <span class="op">&gt;</span> FILES</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a><span class="ex">kmc</span> <span class="at">-k21</span> <span class="at">-t10</span> <span class="at">-m64</span> <span class="at">-ci1</span> <span class="at">-cs10000</span> @FILES reads tmp/</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a><span class="ex">kmc_tools</span> transform reads histogram reads-ont.histo <span class="at">-cx10000</span></span></code></pre>
</div>
<p>Now you can visualize the k-mer frequency distributions using
GenomeScope to assess the quality of the HiFi and ONT reads. This
analysis can help identify potential issues and guide further filtering
or processing steps to improve data quality.</p>
<p>To visualize the k-mer frequency distributions:</p>
<ol style="list-style-type: decimal"><li>Visit the <a href="http://genomescope.org/genomescope2.0/" class="external-link">GenomeScope website</a>
and upload the <code>reads-pacbio.histo</code> OR
<code>reads-ont.histo</code> files (drag and drop).</li>
<li>If you uploaded the <code>reads-pacbio.histo</code> file, enter
description with <code>PacBio HiFi</code>. If you uploaded the
<code>reads-ont.histo</code> file, enter description with
<code>Oxford Nanopore</code>.</li>
<li>Click on the <code>Submit</code> button to generate the k-mer
frequency distribution plots.</li>
</ol><div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>Q: <em>Why did the ONT k-mer analysis fail?</em></p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>A: High error rates in ONT reads can lead to k-mer counting errors,
causing the analysis to fail. K-mer analyses reads rely on accuracy to
generate reliable frequency distributions. Only reads higher than Q20
are recommended for k-mer analysis.</p>
</div>
</div>
</div>
</div>
<div id="callout4" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p><strong>What insights can you gain from the k-mer frequency
distributions?</strong></p>
<ul><li>Look for peaks and patterns in the k-mer frequency
distributions.</li>
<li>Identify potential issues such as heterozygosity, repeat content, or
sequencing errors.</li>
<li>Did your models converge? What does this indicate about the quality
of your data?</li>
</ul></div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>
<strong>Data Quality Control</strong>: Assessing and filtering raw
sequencing data is essential for accurate genome assembly.</li>
<li>
<strong>NanoPlot</strong>: Visualizes read length distributions,
quality scores, and other metrics to evaluate sequencing data
quality.</li>
<li>
<strong>Filtlong</strong>: Filters long-read sequencing data based
on length and quality to retain high-quality reads.</li>
<li>
<strong>GenomeScope</strong>: Profiles genomes using k-mer frequency
distributions to estimate genome size, heterozygosity, and repeat
content.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/2-genome-assembly-strategies.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/4-pacbio-hifi-assembly.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/2-genome-assembly-strategies.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Assembly Strategies
        </a>
        <a class="chapter-link float-end" href="../instructor/4-pacbio-hifi-assembly.html" rel="next">
          Next: PacBio HiFi Assembly...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/3-data-quality-control.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://rcac-bioinformatics.github.io/maize-genomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://rcac-bioinformatics.github.io/maize-genomics/" class="external-link">Source</a></p>
				<p><a href="https://rcac-bioinformatics.github.io/maize-genomics/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:rcac-help@purdue.edu">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.11" class="external-link">sandpaper (0.16.11)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/aseetharam/varnish/tree/db4f5c6f3c4799dfe8247fd057642c8099d5244b" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://maize-genomics.github.io/NA/instructor/3-data-quality-control.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene-annotation, braker, helixer, gemoma, entap, omark, busco, hifiasm",
  "name": "Data Quality Control",
  "creativeWorkStatus": "active",
  "url": "https://maize-genomics.github.io/NA/instructor/3-data-quality-control.html",
  "identifier": "https://maize-genomics.github.io/NA/instructor/3-data-quality-control.html",
  "dateCreated": "2025-02-05",
  "dateModified": "2025-03-01",
  "datePublished": "2025-03-01"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

