<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Maize genome assembly and annotation: Hands-on training: PacBio HiFi Assembly using HiFiasm</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/rcac/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicons/rcac/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicons/rcac/favicon-16x16.png"><link rel="manifest" href="../favicons/rcac/site.webmanifest"><link rel="mask-icon" href="../favicons/rcac/safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav rcac"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="rcac-logo" alt="rcac" src="../assets/images/rcac-logo.svg"></div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../4-pacbio-hifi-assembly.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav rcac" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="rcac" src="../assets/images/rcac-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Maize genome assembly and annotation: Hands-on training
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Maize genome assembly and annotation: Hands-on training
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Maize genome assembly and annotation: Hands-on training
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 21%" class="percentage">
    21%
  </div>
  <div class="progress rcac">
    <div class="progress-bar rcac" role="progressbar" style="width: 21%" aria-valuenow="21" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../4-pacbio-hifi-assembly.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="1-introduction.html">1. Introduction to Genome assembly</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="2-genome-assembly-strategies.html">2. Assembly Strategies</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="3-data-quality-control.html">3. Data Quality Control</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        4. PacBio HiFi Assembly using HiFiasm
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#introduction-to-hifiasm">Introduction to HiFiasm</a></li>
<li><a href="#installation-and-setup">Installation and Setup</a></li>
<li><a href="#overview-of-hifiasm-read-assembly">Overview of HiFiASM Read Assembly</a></li>
<li><a href="#hifiasm-basic-workflow">HiFiasm: basic workflow</a></li>
<li><a href="#understanding-hifiasm-output">Understanding HiFiasm Output</a></li>
<li><a href="#handling-haplotype-resolved-contigs">Handling haplotype-resolved contigs</a></li>
<li><a href="#improving-assembly-quality">Improving Assembly Quality</a></li>
<li><a href="#alternative-assembler-flye-for-hifi">Alternative Assembler: Flye for HiFi</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="5-oxford-nanopore-assembly.html">5. Oxford Nanopore Assembly using Flye</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="6-hybrid-assembly.html">6. Hybrid Long Read Assembly (optional)</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="7-scaffolding-assemblies.html">7. Scaffolding using Optical Genome Mapping</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="8-assembly-evaluation.html">8. Assembly Assessment</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="9-annotation-strategies.html">9. Annotation Strategies</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="10-gene-annotation-with-braker.html">10. Annotation using BRAKER</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="11-gene-annotation-with-helixer.html">11. Annotation using Helixer</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush13">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading13">
        <a href="12-gene-annotation-with-gemoma.html">12. Homology-based gene annotation with GeMoMa</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush14">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading14">
        <a href="13-functional-annotation-with-entap.html">13. Functional annotation using EnTAP</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush15">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading15">
        <a href="14-annotation-assesment.html">14. Annotation Assesment</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr><li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/3-data-quality-control.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/5-oxford-nanopore-assembly.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/3-data-quality-control.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Data Quality Control
        </a>
        <a class="chapter-link float-end" href="../instructor/5-oxford-nanopore-assembly.html" rel="next">
          Next: Oxford Nanopore...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>PacBio HiFi Assembly using HiFiasm</h1>
        <p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/4-pacbio-hifi-assembly.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>

        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>What is HiFiasm, and how does it improve genome assembly using
PacBio HiFi reads?</li>
<li>What are the key steps in running HiFiasm for haplotype-resolved
assembly?</li>
<li>How does HiFiasm handle haplotype resolution and purging of
duplications?</li>
<li>What are the benefits of using HiFiasm for assembling complex and
heterozygous genomes?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Understand the purpose and function of HiFiasm for
haplotype-resolved genome assembly.</li>
<li>Learn to set up and run HiFiasm for assembling genomes using PacBio
HiFi reads.</li>
<li>Gain hands-on experience with haplotype resolution and purging of
duplications in HiFiasm.</li>
<li>Analyze and interpret HiFiasm output to assess assembly quality and
completeness.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction-to-hifiasm">Introduction to HiFiasm<a class="anchor" aria-label="anchor" href="#introduction-to-hifiasm"></a></h2>
<hr class="half-width"><p><a href="https://www.nature.com/articles/s41592-020-01056-5" class="external-link">HiFiasm</a> is
a specialized de novo assembler designed for PacBio HiFi reads,
providing high-quality, haplotype-resolved genome assemblies. Unlike
traditional assemblers that collapse heterozygous regions into a
consensus sequence, HiFiasm preserves haplotype information using a
phased assembly graph approach. This enables more accurate
representation of genetic variations and structural differences.</p>
<p>Leveraging the low error rate of HiFi reads, HiFiasm constructs
phased assembly graphs that allow for haplotype separation without
requiring external polishing or duplication-purging tools. It
significantly improves assembly contiguity, resolving complex regions
more effectively than alternative methods. HiFiasm is widely used in
genome projects, including the Human Pangenome Project, and has been
successfully applied to large and highly heterozygous genomes such as
Sequoia sempervirens (~30 Gb).</p>
<p>With its ability to generate fast and accurate assemblies, HiFiasm
has become the preferred tool for haplotype-resolved genome assembly,
especially when parental reads or Hi-C data are available.</p>
<div id="latest-version-of-hifiasm" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="latest-version-of-hifiasm" class="callout-inner">
<h3 class="callout-title">Latest version of HiFiasm</h3>
<div class="callout-content">
<p>The latest version of HiFiasm supports assembling ONT reads as well.
It has also added support to integrate ultra-long ONT reads for improved
contiguity, as well as hybrid assembly (using both ONT and HiFi reads).
Apart from ONT data, HiFiasm can handle Hi-C data for scaffolding, as
well as kmer profiles from parents to resolve haplotypes. The latest
version includes several bug fixes and performance improvements, making
it more efficient and user-friendly.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="installation-and-setup">Installation and Setup<a class="anchor" aria-label="anchor" href="#installation-and-setup"></a></h2>
<hr class="half-width"><p>HiFiasm is available as module on RCAC clusters. You can load the
module using the following command:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">ml</span> hifiasm</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="at">--version</span></span></code></pre>
</div>
<p>You can also use the Singularity container for HiFiasm, which
provides a consistent environment across different systems. The
container can be pulled from the BioContainers registry using the
following command:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">apptainer</span> pull docker://quay.io/biocontainers/hifiasm:0.24.0--h5ca1c30_0</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">apptainer</span> exec hifiasm_0.24.0--h5ca1c30_0.sif hifiasm <span class="at">--version</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="overview-of-hifiasm-read-assembly">Overview of HiFiASM Read Assembly<a class="anchor" aria-label="anchor" href="#overview-of-hifiasm-read-assembly"></a></h2>
<hr class="half-width"><p>HiFiasm can assemble high-quality, contiguous genome sequences from
PacBio High-Fidelity (HiFi) reads. HiFi reads are long and highly
accurate (99%+), making them ideal for assembling complex genomes,
resolving repetitive regions, and distinguishing haplotypes in diploid
or polyploid organisms.</p>
<p>The assembly workflow typically involves:</p>
<ol style="list-style-type: decimal"><li>Preprocessing reads – filtering and quality-checking raw hifi
reads</li>
<li>Read overlap detection – identifying how reads align to each
other</li>
<li>Error correction – resolving sequencing errors while maintaining
true haplotype differences</li>
<li>Graph construction – building an assembly graph to represent contig
relationships</li>
<li>Contig generation – extracting the final set of contiguous
sequences</li>
<li>Post-processing – refining assemblies by purging duplications or
scaffolding</li>
</ol><p>HiFiasm is optimized for this process, leveraging the high accuracy
of HiFi reads to generate contigs with minimal fragmentation and greater
haplotype resolution compared to traditional assemblers.</p>
<figure><img src="https://github.com/user-attachments/assets/1ab2a17c-ac29-42c7-a890-5c86577e2f19" alt="HiFiasm assembler" class="figure mx-auto d-block"><div class="figcaption">HiFiasm assembler</div>
</figure></section><section><h2 class="section-heading" id="hifiasm-basic-workflow">HiFiasm: basic workflow<a class="anchor" aria-label="anchor" href="#hifiasm-basic-workflow"></a></h2>
<hr class="half-width"><p>To run HiFiasm, you need to provide the input HiFi reads in FASTA or
FASTQ format. The basic command structure is as follows:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="ex">ml</span> hifiasm</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>    <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>    <span class="at">-o</span> Athaliana_hifiasm_default.asm <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>    9994.q20.CCS-filtered.fastq</span></code></pre>
</div>
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>In this command:</p>
<ul><li>
<code>-t</code> specifies the number of threads to use</li>
<li>
<code>-o</code> specifies the output prefix for the assembly</li>
<li>last argument is the input HiFi reads file (fastq format)</li>
</ul><p>The input can either be fastq or fasta, compressed or uncompressed.
The output will be stored in the same directory with the specified
prefix.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="understanding-hifiasm-output">Understanding HiFiasm Output<a class="anchor" aria-label="anchor" href="#understanding-hifiasm-output"></a></h2>
<hr class="half-width"><p>The run generates several output files. Here are all the files and
their descriptions:</p>
<table class="table"><colgroup><col width="38%"><col width="61%"></colgroup><thead><tr class="header"><th align="left">filename</th>
<th>description</th>
</tr></thead><tbody><tr class="odd"><td align="left"><code>Athaliana_hifiasm_default.asm.ec.bin</code></td>
<td>error-corrected reads stored in binary format</td>
</tr><tr class="even"><td align="left"><code>Athaliana_hifiasm_default.asm.ovlp.source.bin</code></td>
<td>source overlap data between reads in binary format</td>
</tr><tr class="odd"><td align="left"><code>Athaliana_hifiasm_default.asm.ovlp.reverse.bin</code></td>
<td>reverse overlap data between reads in binary format</td>
</tr><tr class="even"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.r_utg.noseq.gfa</code></td>
<td>assembly graph of raw unitigs (without sequence)</td>
</tr><tr class="odd"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.r_utg.gfa</code></td>
<td>assembly graph of raw unitigs (with sequence)</td>
</tr><tr class="even"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.r_utg.lowQ.bed</code></td>
<td>low-quality regions in raw unitigs</td>
</tr><tr class="odd"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.p_utg.noseq.gfa</code></td>
<td>assembly graph of purged unitigs (without sequence)</td>
</tr><tr class="even"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.p_utg.gfa</code></td>
<td>assembly graph of purged unitigs (with sequence)</td>
</tr><tr class="odd"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.p_utg.lowQ.bed</code></td>
<td>low-quality regions in purged unitigs</td>
</tr><tr class="even"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.p_ctg.noseq.gfa</code></td>
<td>assembly graph of primary contigs (without sequence)</td>
</tr><tr class="odd"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.p_ctg.gfa</code></td>
<td>assembly graph of primary contigs (with sequence)</td>
</tr><tr class="even"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.p_ctg.lowQ.bed</code></td>
<td>low-quality regions in primary contigs</td>
</tr><tr class="odd"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.hap1.p_ctg.noseq.gfa</code></td>
<td>haplotype 1 primary contigs (without sequence)</td>
</tr><tr class="even"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.hap1.p_ctg.gfa</code></td>
<td>haplotype 1 primary contigs (with sequence)</td>
</tr><tr class="odd"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.hap2.p_ctg.noseq.gfa</code></td>
<td>haplotype 2 primary contigs (without sequence)</td>
</tr><tr class="even"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.hap2.p_ctg.gfa</code></td>
<td>haplotype 2 primary contigs (with sequence)</td>
</tr><tr class="odd"><td align="left"><code>Athaliana_hifiasm_default.asm.bp.hap1.p_ctg.lowQ.bed</code></td>
<td>low-quality regions in haplotype 1 primary contigs</td>
</tr></tbody></table><div id="where-are-the-assemblycontig-sequences" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="where-are-the-assemblycontig-sequences" class="callout-inner">
<h3 class="callout-title">Where are the assembly/contig sequences?</h3>
<div class="callout-content">
<p>The <code>*_ctg.gfa</code> file contains the contigs
(haplotype-resolved, and primary only) in GFA (Graphical Fragment
Assembly) format. You can extract the sequences from this file using
<code>awk</code>. The sequences are represented as lines starting with
<code>S</code> followed by the contig ID and the sequence.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="cf">for</span> ctg <span class="kw">in</span> <span class="pp">*</span>_ctg.gfa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>    <span class="fu">awk</span> <span class="st">'/^S/{print "&gt;"$2"\n"$3}'</span> <span class="va">${ctg}</span> <span class="op">&gt;</span> <span class="va">${ctg</span><span class="op">%</span>.gfa<span class="va">}</span>.fasta</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
</div>
</div>
</div>
<p>Let’s take a look at the sats of this assembly:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="ex">ml</span> quast</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="ex">quast.py</span> <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>    <span class="at">--fast</span> <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>    <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>    <span class="at">-o</span> quast_basic_stats <span class="dt">\</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>    <span class="pp">*</span>p_ctg.fasta</span></code></pre>
</div>
<div id="quast-metrics" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="quast-metrics" class="callout-inner">
<h3 class="callout-title">Quast metrics</h3>
<div class="callout-content">
<p><strong>Key metrics for assembly quality assessment</strong></p>
<table class="table"><colgroup><col width="12%"><col width="87%"></colgroup><thead><tr class="header"><th align="left"><strong>Metric</strong></th>
<th><strong>Description &amp; Importance</strong></th>
</tr></thead><tbody><tr class="odd"><td align="left"><strong># Contigs</strong></td>
<td>The number of contiguous sequences in the assembly. Fewer, larger
contigs indicate a more contiguous assembly.</td>
</tr><tr class="even"><td align="left"><strong>Largest Contig</strong></td>
<td>The length of the longest assembled sequence. A larger value
suggests better resolution of large genomic regions.</td>
</tr><tr class="odd"><td align="left"><strong>Total Length</strong></td>
<td>The sum of all contig lengths. Should approximately match the
expected genome size.</td>
</tr><tr class="even"><td align="left"><strong>N50</strong></td>
<td>The contig length at which 50% of the assembly is covered. Higher
values indicate a more contiguous assembly.</td>
</tr><tr class="odd"><td align="left"><strong>N90</strong></td>
<td>The contig length at which 90% of the assembly is covered. Provides
insight into the distribution of smaller contigs.</td>
</tr><tr class="even"><td align="left"><strong>L50</strong></td>
<td>The number of contigs that make up 50% of the assembly. Lower values
indicate higher contiguity.</td>
</tr><tr class="odd"><td align="left"><strong>L90</strong></td>
<td>The number of contigs that make up 90% of the assembly. Lower values
suggest fewer, larger contigs.</td>
</tr><tr class="even"><td align="left"><strong>auN</strong></td>
<td>Weighted average of contig lengths, emphasizing longer contigs.
Higher values indicate better continuity.</td>
</tr><tr class="odd"><td align="left"><strong># N/100 kbp</strong></td>
<td>Measures the presence of gaps (<code>N</code>s) in the assembly.
Ideally should be 0, meaning no unresolved bases.</td>
</tr></tbody></table><p><strong>How to interpret</strong></p>
<ul><li>
<strong>High N50 and low L50</strong> suggest a well-assembled
genome with fewer, larger contigs.<br></li>
<li>
<strong>Total Length</strong> should be close to the estimated
genome size, ensuring completeness.<br></li>
<li>
<strong>Low # of contigs</strong> indicates better continuity,
meaning fewer breaks in the genome.<br></li>
<li>
<strong>No <code>N</code> bases</strong> means the assembly is
gap-free and doesn’t contain unresolved regions.<br></li>
</ul></div>
</div>
</div>
</section><section><h2 class="section-heading" id="handling-haplotype-resolved-contigs">Handling haplotype-resolved contigs<a class="anchor" aria-label="anchor" href="#handling-haplotype-resolved-contigs"></a></h2>
<hr class="half-width"><p>HiFiasm generates haplotype-resolved contigs. With the default
options above, you saw that it generated <code>hap1.p_ctg</code>,
<code>hap2.p_ctg</code> and <code>.p_ctg</code> GFA files, which
corresponds to haplotype 1, haplotype 2, and primary contigs,
respectively. Although HiFiasm separates the haplotypes, it is unable to
phase (assign the actual regions of hap1 and hap2 to their respective
haplotypes consistently across the genome) them without additional data.
The haplotype-resolved contigs, as-is, is still valuable information,
and can be used for downstream analyses requiring haplotype-specific
information. The primary contigs represent the consensus sequence, and
is usually more complete than either of the haplotype only
assemblies.</p>
<ul><li>Hifiasm purges haplotig duplications by default (to produce two sets
of partially phased contigs)</li>
<li>For inbred or homozygous genomes, you may disable purging with
option <code>-l 0</code>
((<code>hifiasm -o prefix.asm -l 0 -t ${SLURM_CPUS_ON_NODE} input.fq.gz</code>)</li>
<li>To get primary/alternate assemblies, the option
<code>--primary</code> should be set
(<code>hifiasm -o prefix.asm --primary -t ${SLURM_CPUS_ON_NODE} input.fq.gz</code>)</li>
<li>For heterozygous genomes, you can set <code>-l 1</code>,
<code>-l 2</code>, or <code>-l 3</code>, to adjust purging of haplotigs
<ul><li>
<code>-l 1</code> to only purge contained haplotigs</li>
<li>
<code>-l 2</code> to purge all types of haplotigs</li>
<li>
<code>-l 3</code> to purge all types of haplotigs in the most
aggressive way</li>
</ul></li>
<li>If you have parental kmer profiles, you can use them to resolve
haplotypes</li>
</ul><p>We can try running HiFiasm with various <code>-l</code> options to
see how it affects the assembly quality.</p>
<div class="tabs">
<nav><div id="nav-tab-1" class="nav nav-tabs" role="tablist">
<button class="nav-link active" id="nav-tab-1-1" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-1" type="button" role="tab" aria-controls="nav-tabpanel-1-1" aria-selected="true">
  <h3 class="tab-header" id="nav-tab-heading-1-1">
  purge l=0
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-2" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-2" type="button" role="tab" aria-controls="nav-tabpanel-1-2" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-2">
  purge l=1
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-3" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-3" type="button" role="tab" aria-controls="nav-tabpanel-1-3" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-3">
  purge l=2
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-4" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-4" type="button" role="tab" aria-controls="nav-tabpanel-1-4" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-4">
  purge l=3
  </h3>
</button>
</div>
</nav><div id="nav-tabContent-1" class="tab-content">
<div id="nav-tabpanel-1-1" class="tab-pane show active" aria-labelledby="nav-tab-1-1" role="tabpanel">
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="ex">ml</span> hifiasm</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co"># purge level 0</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> hifiasm_purge_level_0</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="bu">cd</span> hifiasm_purge_level_0</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ../9994.q20.CCS-filtered.fastq</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="dt">\</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>  <span class="at">-o</span> Athaliana_hifiasm_purge_level_0.asm <span class="dt">\</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>  <span class="at">-l</span> 0 <span class="dt">\</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>  <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>  9994.q20.CCS-filtered.fastq</span></code></pre>
</div>
</div>
<div id="nav-tabpanel-1-2" class="tab-pane" aria-labelledby="nav-tab-1-2" role="tabpanel">
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="ex">ml</span> hifiasm</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co"># purge level 1</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> hifiasm_purge_level_1</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="bu">cd</span> hifiasm_purge_level_1</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ../9994.q20.CCS-filtered.fastq</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="dt">\</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>  <span class="at">-o</span> Athaliana_hifiasm_purge_level_1.asm <span class="dt">\</span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>  <span class="at">-l</span> 1 <span class="dt">\</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>  <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>  9994.q20.CCS-filtered.fastq</span></code></pre>
</div>
</div>
<div id="nav-tabpanel-1-3" class="tab-pane" aria-labelledby="nav-tab-1-3" role="tabpanel">
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="ex">ml</span> hifiasm</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="co"># purge level 2</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> hifiasm_purge_level_2</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="bu">cd</span> hifiasm_purge_level_2</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ../9994.q20.CCS-filtered.fastq</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="dt">\</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>  <span class="at">-o</span> Athaliana_hifiasm_purge_level_2.asm <span class="dt">\</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>  <span class="at">-l</span> 2 <span class="dt">\</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>  <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>  9994.q20.CCS-filtered.fastq</span></code></pre>
</div>
</div>
<div id="nav-tabpanel-1-4" class="tab-pane" aria-labelledby="nav-tab-1-4" role="tabpanel">
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="ex">ml</span> hifiasm</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="co"># purge level 3</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> hifiasm_purge_level_3</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a><span class="bu">cd</span> hifiasm_purge_level_3</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ../9994.q20.CCS-filtered.fastq</span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="dt">\</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>  <span class="at">-o</span> Athaliana_hifiasm_purge_level_3.asm <span class="dt">\</span></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>  <span class="at">-l</span> 3 <span class="dt">\</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>  <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>  9994.q20.CCS-filtered.fastq</span></code></pre>
</div>
</div>
</div>
</div>
<div id="callout3" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>Each of these will run in about ~15 minutes with 32 cores. You can
either run them in parallel or sequentially or request more cores to run
them faster.</p>
</div>
</div>
</div>
<div id="comparing-assemblies" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="comparing-assemblies" class="callout-inner">
<h3 class="callout-title">Comparing assemblies</h3>
<div class="callout-content">
<p>Convert GFA files to FASTA format</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="cf">for</span> dir <span class="kw">in</span> hifiasm_purge_level_<span class="dt">{</span><span class="dv">0</span><span class="dt">..</span><span class="dv">3</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span> </span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>    <span class="bu">cd</span> <span class="va">${dir}</span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>    <span class="cf">for</span> ctg <span class="kw">in</span> <span class="pp">*</span>_ctg.gfa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>        <span class="fu">awk</span> <span class="st">'/^S/{print "&gt;"$2"\n"$3}'</span> <span class="va">${ctg}</span> <span class="op">&gt;</span> <span class="va">${ctg</span><span class="op">%</span>.gfa<span class="va">}</span>.fasta</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>    <span class="cf">done</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>    <span class="bu">cd</span> ..</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
<p>Run QUAST to compare the assemblies</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> quast_stats</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="cf">for</span> fasta <span class="kw">in</span> hifiasm_purge_level_{0..3}/<span class="pp">*</span>_p_ctg.fasta<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>    <span class="fu">ln</span> <span class="at">-s</span> <span class="va">${fasta}</span> quast_stats/</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="bu">cd</span> quast_stats</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="ex">quast.py</span> <span class="dt">\</span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>    <span class="at">--fast</span> <span class="dt">\</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>    <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a>    <span class="at">-o</span> quast_purge_level_stats <span class="dt">\</span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a>    <span class="pp">*</span>_p_ctg.fasta</span></code></pre>
</div>
<p>Run Compleasm to compare the assemblies</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="ex">ml</span> compleasm</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> compleasm_stats</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="cf">for</span> fasta <span class="kw">in</span> hifiasm_purge_level_{0..3}/<span class="pp">*</span>_p_ctg.fasta<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>    <span class="fu">ln</span> <span class="at">-s</span> <span class="va">${fasta}</span> compleasm_stats/</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a><span class="bu">cd</span> compleasm_stats</span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a><span class="cf">for</span> fasta <span class="kw">in</span> <span class="pp">*</span>_p_ctg.fasta<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a>    <span class="ex">compleasm</span> run <span class="dt">\</span></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a>       <span class="at">-a</span> <span class="va">${fasta}</span> <span class="dt">\</span></span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a>       <span class="at">-o</span> <span class="va">${fasta</span><span class="op">%</span>.<span class="pp">*</span><span class="va">}</span> <span class="dt">\</span></span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a>       <span class="at">-l</span> brassicales_odb10 <span class="dt">\</span></span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a>       <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
<p>Examining the results from QUAST and Compleasm, compare the assembly
statistics and assess the impact of different purging levels on the
assembly quality. Look for metrics like N50, L50, and total assembly
size to evaluate the contiguity and completeness of the assemblies.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="improving-assembly-quality">Improving Assembly Quality<a class="anchor" aria-label="anchor" href="#improving-assembly-quality"></a></h2>
<hr class="half-width"><p>After the first round of assembly, you will have the files
<code>*.ec.bin</code>, <code>*.ovlp.source.bin</code>, and
<code>*.ovlp.reverse.bin</code>. Save these files and try various
options to see if you can improve the assembly. First, make a folder to
move the .gfa, .fasta, and .bed files. These are the results from the
first round of assembly. Second, adjust the parameters in the hifiasm
command and run the assembler again. Third, move results to a new folder
and compare the results of the first folder. You can re-run the assembly
quickly and generate statistics for each of these folders and compare
them to see if the changes improved the assembly.</p>
</section><section><h2 class="section-heading" id="alternative-assembler-flye-for-hifi">Alternative Assembler: Flye for HiFi<a class="anchor" aria-label="anchor" href="#alternative-assembler-flye-for-hifi"></a></h2>
<hr class="half-width"><p>Flye is another popular assembler specialized for ONT reads, offering
a different approach to haplotype-resolved assembly. The latest version
can also use HiFi reads to generate great quality assemblies. We will
explore Flye in a separate episode to compare its performance with
HiFiasm. But in this optional section, you can try running Flye with
HiFi reads to see how it performs compared to HiFiasm.</p>
<div class="section level3">
<h3 id="running-flye-with-hifi-reads">Running Flye with HiFi Reads<a class="anchor" aria-label="anchor" href="#running-flye-with-hifi-reads"></a></h3>
<p>To run Flye with HiFi reads, you can use the following command
structure:</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a><span class="ex">ml</span> flye</span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="ex">flye</span> <span class="dt">\</span></span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>  <span class="at">--pacbio-hifi</span> 9994.q20.CCS-filtered-60x.fastq <span class="dt">\</span></span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>  <span class="at">--genome-size</span> 135m <span class="dt">\</span></span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>  <span class="at">--out-dir</span> flye_pcb <span class="dt">\</span></span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>  <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<div id="options-used" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="options-used" class="callout-inner">
<h3 class="callout-title">Options used</h3>
<div class="callout-content">
<p>In this command:</p>
<ul><li>
<code>--pacbio-hifi</code> specifies the input HiFi reads file</li>
<li>
<code>--genome-size</code> provides an estimate of the genome size
(optional)</li>
<li>
<code>--out-dir</code> specifies the output directory for Flye
results</li>
<li>
<code>--threads</code> specifies the number of threads to use</li>
</ul><p>The output will be stored in the specified directory, containing the
assembly graph, contigs, and other relevant files.</p>
<p>With 64 cores, this will run in about ~40 mins. It needs about
~80-90Gb of memory.</p>
</div>
</div>
</div>
<div id="quality-metrics" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="quality-metrics" class="callout-inner">
<h3 class="callout-title">Quality metrics</h3>
<div class="callout-content">
<p>Run quality metrics on Flye assembly:</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="ex">ml</span> quast</span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a><span class="ex">quast.py</span> <span class="dt">\</span></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>    <span class="at">--fast</span> <span class="dt">\</span></span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>    <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>    <span class="at">-o</span> quast_flye_stats <span class="dt">\</span></span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a>    flye_pcb/assembly.fasta</span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a><span class="ex">ml</span> compleasm</span>
<span id="cb14-10"><a href="#cb14-10" tabindex="-1"></a><span class="ex">compleasm</span> run <span class="dt">\</span></span>
<span id="cb14-11"><a href="#cb14-11" tabindex="-1"></a>  <span class="at">-a</span> flye_pcb/assembly.fasta <span class="dt">\</span></span>
<span id="cb14-12"><a href="#cb14-12" tabindex="-1"></a>  <span class="at">-o</span> flye_pcb <span class="dt">\</span></span>
<span id="cb14-13"><a href="#cb14-13" tabindex="-1"></a>  <span class="at">-l</span> brassicales_odb10 <span class="dt">\</span></span>
<span id="cb14-14"><a href="#cb14-14" tabindex="-1"></a>  <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>Which assembler did a better job at assembling the genome? Compare
the statistics from QUAST and Compleasm for Flye and HiFiasm assemblies
to evaluate their performance.</p>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li>
<strong>HiFiasm</strong> is a specialized assembler for PacBio HiFi
reads, providing high-quality, haplotype-resolved genome
assemblies.</li>
<li>It leverages the high accuracy of HiFi reads to generate phased
assembly graphs, preserving haplotype information.</li>
<li>HiFiasm is optimized for resolving complex regions and
distinguishing haplotypes in diploid or polyploid organisms.</li>
<li>The assembler generates primary contigs and haplotype-resolved
contigs, offering valuable information for downstream analyses.</li>
<li>By adjusting purging levels and using parental kmer profiles, users
can improve haplotype resolution and assembly quality.</li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/3-data-quality-control.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/5-oxford-nanopore-assembly.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/3-data-quality-control.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Data Quality Control
        </a>
        <a class="chapter-link float-end" href="../instructor/5-oxford-nanopore-assembly.html" rel="next">
          Next: Oxford Nanopore...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/4-pacbio-hifi-assembly.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://rcac-bioinformatics.github.io/maize-genomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://rcac-bioinformatics.github.io/maize-genomics/" class="external-link">Source</a></p>
				<p><a href="https://rcac-bioinformatics.github.io/maize-genomics/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:rcac-help@purdue.edu">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.11" class="external-link">sandpaper (0.16.11)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/aseetharam/varnish/tree/db4f5c6f3c4799dfe8247fd057642c8099d5244b" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://maize-genomics.github.io/NA/instructor/4-pacbio-hifi-assembly.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene-annotation, braker, helixer, gemoma, entap, omark, busco, hifiasm",
  "name": "PacBio HiFi Assembly using HiFiasm",
  "creativeWorkStatus": "active",
  "url": "https://maize-genomics.github.io/NA/instructor/4-pacbio-hifi-assembly.html",
  "identifier": "https://maize-genomics.github.io/NA/instructor/4-pacbio-hifi-assembly.html",
  "dateCreated": "2025-02-05",
  "dateModified": "2025-03-01",
  "datePublished": "2025-03-01"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

