<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>Maize genome assembly and annotation: Hands-on training: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css">
<script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/rcac/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicons/rcac/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="../favicons/rcac/favicon-16x16.png">
<link rel="manifest" href="../favicons/rcac/site.webmanifest">
<link rel="mask-icon" href="../favicons/rcac/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav rcac"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="rcac-logo" alt="rcac" src="../assets/images/rcac-logo.svg">
</div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='../aio.html';">Learner View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav rcac" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="rcac" src="../assets/images/rcac-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      Maize genome assembly and annotation: Hands-on training
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            Maize genome assembly and annotation: Hands-on training
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Maize genome assembly and annotation: Hands-on training
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress rcac">
    <div class="progress-bar rcac" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../aio.html">Learner View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="1-introduction.html">1. Introduction to Genome assembly</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="2-genome-assembly-strategies.html">2. Assembly Strategies</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="3-data-quality-control.html">3. Data Quality Control</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="4-pacbio-hifi-assembly.html">4. PacBio HiFi Assembly using HiFiasm</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="5-oxford-nanopore-assembly.html">5. Oxford Nanopore Assembly using Flye</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="6-hybrid-assembly.html">6. Hybrid Long Read Assembly (optional)</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="7-scaffolding-assemblies.html">7. Scaffolding using Optical Genome Mapping</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="8-assembly-evaluation.html">8. Assembly Assessment</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="9-annotation-strategies.html">9. Annotation Strategies</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="10-gene-annotation-with-braker.html">10. Annotation using BRAKER</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="11-gene-annotation-with-helixer.html">11. Annotation using Helixer</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush13">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading13">
        <a href="12-gene-annotation-with-gemoma.html">12. Homology-based gene annotation with GeMoMa</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush14">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading14">
        <a href="13-functional-annotation-with-entap.html">13. Functional annotation using EnTAP</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush15">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading15">
        <a href="14-annotation-assesment.html">14. Annotation Assesment</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr>
<li><a class="dropdown-item" href="reference.html">Reference</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources">
<a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-1-introduction"><p>Content from <a href="1-introduction.html">Introduction to Genome assembly</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/1-introduction.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is genome assembly, and why is it important?</li>
<li>What sequencing technologies can be used for genome assembly?</li>
<li>What are <em>de novo</em> and reference-guided assemblies?</li>
<li>What challenges arise when generating high-quality assemblies?</li>
<li>What software tools are used for assembling genomes?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Learn key concepts and terminology related to genome assembly.</li>
<li>Understand datasets and tools used for genome assembly.</li>
<li>Describe major sequencing technologies and their impact on
assembly.</li>
<li>Identify common computational challenges in genome assembly.</li>
<li>Explain fundamental strategies and algorithms used in genome
assembly.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="what-is-genome-assembly">What is Genome Assembly?<a class="anchor" aria-label="anchor" href="#what-is-genome-assembly"></a>
</h2>
<hr class="half-width">
<p>Genome assembly is the process of reconstructing a complete genome
sequence by arranging fragmented DNA sequences (reads) into a continuous
sequence.</p>
<ul>
<li>Goal is to achieve a high-quality reference genome that accurately
represents the structure and sequence of an organism’s DNA.</li>
<li>It enables deeper understanding of genes, their function, and
evolutionary history.</li>
<li>Vital for studying complex traits, species diversity, and disease
mechanisms.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/8f1b2924-4af4-46f5-a925-b927b0fdfc81" alt="Genome Assembly" class="figure mx-auto d-block"><div class="figcaption">Genome Assembly</div>
</figure></section><section><h2 class="section-heading" id="imporance-of-genome-assembly">Imporance of Genome Assembly<a class="anchor" aria-label="anchor" href="#imporance-of-genome-assembly"></a>
</h2>
<hr class="half-width">
<ul>
<li>Applications in medicine, agriculture, conservation, and
biotechnology:
<ul>
<li>Human genetics: Assembling genomes to identify disease-causing
mutations.</li>
<li>Crop improvement: Identifying beneficial traits in plant
genomes.</li>
<li>Conservation biology: Sequencing endangered species to understand
genetic diversity.</li>
</ul>
</li>
<li>Examples of major genome sequencing projects (Human Genome Project,
Vertebrate Genome Project).</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/bd5e664b-cf76-4c33-a2ad-250040444a0b" alt="Number of genomes in NCBI" class="figure mx-auto d-block"><div class="figcaption">Number of genomes in NCBI</div>
</figure></section><section><h2 class="section-heading" id="de-novo-vs--reference-guided-assembly">De Novo vs. Reference-Guided Assembly<a class="anchor" aria-label="anchor" href="#de-novo-vs--reference-guided-assembly"></a>
</h2>
<hr class="half-width">
<ul>
<li>De novo assembly:
<ul>
<li>Used when no reference genome exists.</li>
<li>Requires assembling the genome from scratch using computational
methods.</li>
<li>Example: Assembling a new plant species genome.</li>
</ul>
</li>
<li>Reference-guided assembly:
<ul>
<li>Aligns reads to a closely related reference genome.</li>
<li>Useful for identifying variations but limited by reference
bias.</li>
<li>Example: Human genome resequencing for variant detection.</li>
</ul>
</li>
</ul>
<figure><img width="755" alt="Assembly types" src="https://github.com/user-attachments/assets/002b3956-0679-41b9-aac5-f3c1e1f72a4b" class="figure mx-auto d-block"></figure></section><section><h2 class="section-heading" id="basic-steps-in-genome-assembly">Basic Steps in Genome Assembly<a class="anchor" aria-label="anchor" href="#basic-steps-in-genome-assembly"></a>
</h2>
<hr class="half-width">
<ol style="list-style-type: decimal">
<li>Sequencing: Generating raw reads from a genome.</li>
<li>Preprocessing &amp; Quality Control: Filtering and trimming
reads.</li>
<li>Assembly: Aligning and merging overlapping reads into contigs.</li>
<li>Scaffolding: Ordering contigs into larger scaffolds using long-range
sequencing data or mapping techniques.</li>
<li>Polishing: Correcting errors using additional data.</li>
<li>Quality Assessment: Evaluating assembly completeness and
accuracy.</li>
<li>Downstream Analysis: Annotating genes, identifying variants, and
studying genome structure.</li>
</ol>
<figure><img src="https://github.com/user-attachments/assets/ac4f82b2-1488-4ba8-b2c7-b1976a393f82" alt="Steps Overview" class="figure mx-auto d-block"><div class="figcaption">Steps Overview</div>
</figure><p>reference: <a href="https://doi.org/10.1016/j.xpro.2022.101506" class="external-link">10.1016/j.xpro.2022.101506</a></p>
</section><section><h2 class="section-heading" id="data-types-for-genome-assembly">Data types for Genome Assembly<a class="anchor" aria-label="anchor" href="#data-types-for-genome-assembly"></a>
</h2>
<hr class="half-width">
<p><strong>Illumina</strong></p>
<ul>
<li><p>Excels at high-throughput, short-read sequencing with high
accuracy.</p></li>
<li><p>Uses Sequencing-by-Synthesis (SBS). DNA is fragmented, adapters
are attached, and the fragments are immobilized on a flowcell. A
polymerase incorporates fluorescently labeled nucleotides, and a camera
captures the emitted signals in real time. Each cycle represents one
nucleotide added to the growing DNA strand.</p></li>
</ul>
<p><strong>PacBio HiFi</strong></p>
<ul>
<li><p>provide accurate long reads, balancing throughput and error
correction.</p></li>
<li><p>Uses Single Molecule, Real-Time (SMRT) sequencing. DNA is ligated
into circular molecules and loaded onto a chip with zero-mode waveguides
(ZMWs). A polymerase synthesizes the complementary strand, incorporating
fluorescently labeled nucleotides. The system detects light pulses in
real time, capturing multiple passes of the same molecule to generate
highly accurate HiFi reads.</p></li>
</ul>
<p><strong>Oxford Nanopore</strong></p>
<ul>
<li><p>enables ultra-long reads but requires more advanced error
correction.</p></li>
<li><p>A single DNA strand is passed through a biological nanopore
embedded in a membrane. As nucleotides move through the pore, they cause
characteristic disruptions in an electrical current, which are
interpreted by machine-learning algorithms to determine the
sequence.</p></li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/62914627-e17d-4fd6-aceb-ae5dc48626d8" alt="Sequencing Instruments" class="figure mx-auto d-block"><div class="figcaption">Sequencing Instruments</div>
</figure></section><section><h2 class="section-heading" id="challenges-in-genome-assembly">Challenges in Genome Assembly<a class="anchor" aria-label="anchor" href="#challenges-in-genome-assembly"></a>
</h2>
<hr class="half-width">
<ul>
<li>
<strong>Repetitive Elements</strong>: Identical or similar sequences
that occur multiple times in the genome, making it difficult to resolve
unique regions.</li>
<li>
<strong>Heterozygosity</strong>: Presence of two or more alleles at
a given locus, leading to ambiguous read alignments.</li>
<li>
<strong>Polyploidy</strong>: Multiple copies of chromosomes,
complicating assembly due to similar sequences.</li>
<li>
<strong>Genome Size</strong>: Large genomes require more
computational resources and specialized algorithms.</li>
<li>
<strong>Error Correction</strong>: Addressing sequencing errors and
distinguishing true variants from artifacts.</li>
<li>
<strong>Structural Variants</strong>: Large-scale rearrangements,
duplications, deletions, and inversions that disrupt contiguity.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/f1ec5463-6a81-466e-b4ce-a0a4d8344005" alt="Challenges in Genome Assembly" class="figure mx-auto d-block"><div class="figcaption">Challenges in Genome Assembly</div>
</figure></section><section><h2 class="section-heading" id="main-programs-used-for-genome-assembly">Main programs used for Genome Assembly<a class="anchor" aria-label="anchor" href="#main-programs-used-for-genome-assembly"></a>
</h2>
<hr class="half-width">
<ul>
<li>Data QC:
<ul>
<li>NanoPlot: Visualization of sequencing data quality.</li>
<li>FiltLong: Filtering long reads based on quality and length.</li>
</ul>
</li>
<li>Assembly:
<ul>
<li>HiFiasm: HiFi assembler for PacBio data.</li>
<li>Flye: de novo assembler for long reads.</li>
</ul>
</li>
<li>Post-processing:
<ul>
<li>Medaka: Basecaller and consensus polishing for Flye assembly.</li>
<li>Bionano Solve: Optical mapping for scaffolding and validation.</li>
</ul>
</li>
<li>Evaluation:
<ul>
<li>QUAST: Quality assessment tool for evaluating assemblies.</li>
<li>Compleasm (BUSCO alternative): Benchmarking tool for assessing
genome completeness.</li>
<li>KAT: Kmer-based evaluation of assembly accuracy and
completeness.</li>
</ul>
</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/17b9ec7a-343e-4c80-baf5-066b4ac130ec" alt="Programs used in this workshop" width="50%" class="figure mx-auto d-block"></figure><div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>
<strong>Genome assembly</strong> reconstructs complete genome
sequences from fragmented DNA reads.</li>
<li>
<strong>De novo</strong> assembly builds genomes without a
reference, while <strong>reference-guided</strong> assembly uses
existing genomes.</li>
<li>
<strong>Sequencing technologies</strong> like Illumina, PacBio HiFi,
and Oxford Nanopore offer different read lengths and error rates.</li>
<li>
<strong>Challenges</strong> include repetitive elements,
heterozygosity, and error correction.</li>
<li>
<strong>Tools</strong> many programs are available for data QC,
assembly, post-processing, and evaluation - choice depends on data type
and research goals.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-2-genome-assembly-strategies"><p>Content from <a href="2-genome-assembly-strategies.html">Assembly Strategies</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/2-genome-assembly-strategies.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What factors influence the choice of genome assembly strategy?</li>
<li>How do different assembly methods compare in terms of read length,
accuracy, and computational requirements?</li>
<li>What are the key steps in evaluating genome assemblies using BUSCO
and QUAST?</li>
<li>How do Bionano OGM and Hi-C sequencing improve genome continuity and
organization?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand factors influencing the choice of genome assembly
strategy.</li>
<li>Compare different assembly methods based on read length, accuracy,
and computational requirements.</li>
<li>Learn how to evaluate genome assemblies using BUSCO and QUAST.</li>
<li>Explore the role of Bionano OGM and Hi-C sequencing in improving
genome continuity.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="assembly-strategies">Assembly Strategies<a class="anchor" aria-label="anchor" href="#assembly-strategies"></a>
</h2>
<hr class="half-width">
<p>Genome assembly involves choosing the right approach based on
sequencing technology, read type, genome complexity, and research
objectives. This chapter introduces key factors influencing assembly
strategy selection, from read length and coverage requirements to
computational trade-offs. We will explore different methods—PacBio HiFi
with HiFiasm, ONT with Flye, and hybrid assemblies—along with
scaffolding techniques like Bionano Optical Genome Mapping (OGM) and
Hi-C, which help improve genome continuity and organization. Finally,
we’ll discuss assembly evaluation tools such as BUSCO and QUAST to
assess the completeness and quality of assembled genomes.</p>
<div id="factors-influencing-the-choice-of-strategy" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="factors-influencing-the-choice-of-strategy" class="callout-inner">
<h3 class="callout-title">Factors Influencing the Choice of Strategy</h3>
<div class="callout-content">
<ul>
<li>
<strong>Read length</strong>: Affects the ability to resolve
repeats; short reads struggle with complex regions, while long reads
improve contiguity.<br>
</li>
<li>
<strong>Coverage depth</strong>: Crucial for assembly accuracy, with
HiFi requiring 20-30x, ONT needing 50-100x, and hybrid assemblies
depending on short-read support for polishing.<br>
</li>
<li>
<strong>Genome complexity</strong>: Includes repeat content,
heterozygosity, and polyploidy, which influence assembly success and
determine whether specialized tools or additional scaffolding methods
are needed.<br>
</li>
<li>
<strong>Computational resources</strong>: Vary across assembly
strategies, with HiFiasm being RAM-intensive, Flye being more
lightweight, and hybrid assemblies requiring additional processing for
polishing.<br>
</li>
<li>
<strong>Sequencing budget</strong>: Plays a role, as HiFi sequencing
is costlier but highly accurate, ONT is cheaper but requires more data
for error correction, and hybrid approaches balance cost and
quality.<br>
</li>
<li>
<strong>Downstream analyses</strong>: Structural variation
detection, gene annotation, or chromosome-level assemblies influence the
choice of assembler and the need for scaffolding methods like Hi-C or
Bionano OGM.</li>
</ul>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="comparative-assembly-strategies">Comparative Assembly Strategies<a class="anchor" aria-label="anchor" href="#comparative-assembly-strategies"></a>
</h2>
<hr class="half-width">
<table class="table">
<colgroup>
<col width="25%">
<col width="21%">
<col width="24%">
<col width="28%">
</colgroup>
<thead><tr class="header">
<th>Factor</th>
<th>PacBio HiFi</th>
<th>ONT</th>
<th>Hybrid (ONT + Illumina)</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Read length</strong></td>
<td>~15-20kb</td>
<td>10-100kb+</td>
<td>Mix of short &amp; long</td>
</tr>
<tr class="even">
<td><strong>Read accuracy</strong></td>
<td>High (~99%)</td>
<td>Moderate (~90%)</td>
<td>High (after polishing)</td>
</tr>
<tr class="odd">
<td><strong>Coverage needed</strong></td>
<td>20-30x</td>
<td>50-100x</td>
<td>ONT: 50x + Illumina: 30x</td>
</tr>
<tr class="even">
<td><strong>Cost per Gb</strong></td>
<td>Expensive</td>
<td>Lower</td>
<td>Medium</td>
</tr>
<tr class="odd">
<td><strong>Error profile</strong></td>
<td>Random errors, low indels</td>
<td>Higher error rate, systematic errors</td>
<td>ONT errors corrected by Illumina</td>
</tr>
<tr class="even">
<td><strong>Computational requirements</strong></td>
<td>High RAM required</td>
<td>Moderate RAM required</td>
<td>Moderate</td>
</tr>
<tr class="odd">
<td><strong>Best for</strong></td>
<td>High-accuracy assemblies</td>
<td>Ultra-long contigs</td>
<td>Combines advantages of both</td>
</tr>
<tr class="even">
<td><strong>Repeat resolution</strong></td>
<td>Good</td>
<td>Very good</td>
<td>Very good</td>
</tr>
<tr class="odd">
<td><strong>Scaffolding needed</strong></td>
<td>Rarely needed</td>
<td>May be needed</td>
<td>Sometimes needed</td>
</tr>
<tr class="even">
<td><strong>Polishing required</strong></td>
<td>Not required</td>
<td>Required (Racon + Medaka)</td>
<td>Required (Pilon)</td>
</tr>
<tr class="odd">
<td><strong>Structural variant detection</strong></td>
<td>Good</td>
<td>Excellent</td>
<td>Good</td>
</tr>
<tr class="even">
<td><strong>Haplotype phasing</strong></td>
<td>Excellent</td>
<td>Good</td>
<td>Moderate</td>
</tr>
<tr class="odd">
<td><strong>Genome size suitability</strong></td>
<td>Suitable for large and small genomes</td>
<td>Best for large genomes</td>
<td>Best for complex genomes</td>
</tr>
<tr class="even">
<td><strong>Downstream applications</strong></td>
<td>Reference-quality genome assembly, annotation</td>
<td>Structural variation analysis, de novo assembly</td>
<td>Genome correction, variant calling, scaffolding</td>
</tr>
</tbody>
</table></section><section><h2 class="section-heading" id="contig-vs--scaffold-vs--chromosome-level-assembly">Contig vs. Scaffold vs. Chromosome-Level Assembly<a class="anchor" aria-label="anchor" href="#contig-vs--scaffold-vs--chromosome-level-assembly"></a>
</h2>
<hr class="half-width">
<p>Genome assemblies progress through different levels of completeness
and organization:</p>
<ul>
<li>
<strong>Contig-level assembly</strong>: The raw output of
assemblers, consisting of contiguous sequences without known order or
orientation. Longer contigs indicate better assembly continuity.<br>
</li>
<li>
<strong>Scaffold-level assembly</strong>: Contigs linked together
using additional data (e.g., long-range mate-pair reads, optical maps,
or Hi-C). Gaps (represented as ’N’s) remain where connections exist but
sequence information is missing.<br>
</li>
<li>
<strong>Chromosome-level assembly</strong>: The highest-quality
assembly, where scaffolds are further ordered and oriented into full
chromosomes using genetic maps, Hi-C data, or synteny with a reference
genome.</li>
</ul>
<p>Higher levels of assembly provide better genome context, but require
additional scaffolding methods beyond de novo assembly.</p>
<figure><img src="https://github.com/user-attachments/assets/4b7406b6-747a-4a34-b816-e1de2cf444a9" alt="Assembly levels" class="figure mx-auto d-block"><div class="figcaption">Assembly levels</div>
</figure></section><section><h2 class="section-heading" id="workflow-for-various-assemblies">Workflow for Various Assemblies<a class="anchor" aria-label="anchor" href="#workflow-for-various-assemblies"></a>
</h2>
<hr class="half-width">
<p>In this workshop, we will use HiFiasm for PacBio HiFi assemblies,
Flye for ONT assemblies, and Flye in hybrid mode for ONT + Illumina
assemblies, followed by quality assessment using BUSCO and QUAST to
evaluate completeness and accuracy.</p>
<div class="section level3">
<h3 id="pacbio-hifi-assembly-with-hifiasm">PacBio HiFi Assembly with HiFiasm<a class="anchor" aria-label="anchor" href="#pacbio-hifi-assembly-with-hifiasm"></a>
</h3>
<ul>
<li>
<strong>PacBio HiFi reads</strong>: Highly accurate long reads with
low error rates, suitable for de novo assembly without polishing.</li>
<li>
<strong>HiFiasm</strong>: A specialized assembler for HiFi data,
leveraging read accuracy and length to resolve complex regions and
produce high-quality contigs.</li>
<li>
<strong>Workflow</strong>: Run HiFiasm with HiFi reads, adjust
parameters based on genome size and complexity, and evaluate the
assembly using Compleasm and QUAST.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/1f87d81a-707b-44ff-a8e1-49648f61de95" alt="PacBio Assembly" class="figure mx-auto d-block"><div class="figcaption">PacBio Assembly</div>
</figure>
</div>
<div class="section level3">
<h3 id="ont-assembly-with-flye">ONT Assembly with Flye<a class="anchor" aria-label="anchor" href="#ont-assembly-with-flye"></a>
</h3>
<ul>
<li>
<strong>ONT reads</strong>: Ultra-long reads with higher error
rates, requiring additional error correction and polishing steps.</li>
<li>
<strong>Flye</strong>: A de novo assembler optimized for long reads,
capable of resolving complex repeats and generating high-quality
assemblies.</li>
<li>
<strong>Workflow</strong>: Run Flye with ONT reads, adjust
parameters based on genome size and complexity, and polish the assembly
using Medaka for basecalling and consensus polishing.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/9864ae7d-a893-4aec-ad4a-ea57170c8c27" alt="ONT assembly" class="figure mx-auto d-block"><div class="figcaption">ONT assembly</div>
</figure>
</div>
<div class="section level3">
<h3 id="hybrid-ont-pacbio-assembly-with-flye">Hybrid (ONT + PacBio) Assembly with Flye<a class="anchor" aria-label="anchor" href="#hybrid-ont-pacbio-assembly-with-flye"></a>
</h3>
<ul>
<li>
<strong>Hybrid assembly</strong>: Combines the strengths of both
technologies for improved accuracy and contiguity.</li>
<li>
<strong>Workflow</strong>: Run Flye with both ONT and PacBio reads,
adjust parameters for hybrid mode, and polish the assembly using ONT or
PacBio reads for error correction and consensus polishing.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/9b51523d-2358-4ad3-b6b3-bfe5d0189e4d" alt="Hybrid assembly" class="figure mx-auto d-block"><div class="figcaption">Hybrid assembly</div>
</figure>
</div>
</section><section><h2 class="section-heading" id="assembly-evaluation">Assembly Evaluation<a class="anchor" aria-label="anchor" href="#assembly-evaluation"></a>
</h2>
<hr class="half-width">
<ul>
<li>Why assessing genome assembly quality is crucial before downstream
analyses.<br>
</li>
<li>Metrics to determine assembly completeness, accuracy, and
contiguity.</li>
</ul>
<p><strong>1. BUSCO (Benchmarking Universal Single-Copy Orthologs)/
Compleasm</strong></p>
<ul>
<li><p><a href="https://github.com/huangnengCSU/compleasm" class="external-link">Compleasm</a>
is a faster alternative to BUSCO for assessing genome completeness based
on single-copy orthologs. It evaluates genome completeness by checking
for highly conserved, single-copy genes expected to be present in nearly
all members of a given lineage. These genes are essential for basic
cellular functions, making them reliable markers for assessing genome
assembly quality.</p></li>
<li><p>We expect these genes to be present in our organism because they
are evolutionarily conserved and critical for survival. If many BUSCO
genes are missing or fragmented, it suggests gaps, misassemblies, or
sequencing errors, which can compromise downstream analyses like gene
annotation and functional studies. A high BUSCO completeness score
indicates a well-assembled genome with minimal missing data.</p></li>
<li><p>The BUSCO reports provide detailed statistics on the number of
complete, fragmented, and missing BUSCO genes, as well as the percentage
of genome completeness. The output helps identify areas for improvement
and guide further optimization steps in the assembly process.</p></li>
</ul>
<figure><img width="1030" alt="BUSCO genes" src="https://github.com/user-attachments/assets/e4c7951f-f4f1-4395-974c-abd6716d8cd1" class="figure mx-auto d-block"></figure><p><strong>2. QUAST (Quality Assessment Tool for Genome
Assemblies)</strong></p>
<ul>
<li><p><strong>Comprehensive Assembly Statistics</strong>: QUAST
provides detailed metrics beyond N50 and L50, including <strong>total
number of contigs, GC content, genome size estimation, and misassembly
rates</strong>, allowing in-depth evaluation of genome continuity and
structure.</p></li>
<li><p><strong>Reference-Based and Reference-Free Evaluation</strong>:
It can assess assemblies against a reference genome (identifying
misassemblies, inversions, and duplications) or work in
<strong>reference-free mode</strong>, making it useful for de novo
assemblies without a known genome sequence.</p></li>
<li><p><strong>Structural Error Detection and Gene Feature
Analysis</strong>: QUAST integrates gene annotation tools like
<strong>BUSCO and GeneMark</strong>, highlights misassemblies based on
alignment breaks, and detects <strong>gaps, relocations, and
translocations</strong>, making it particularly useful for validating
scaffolding approaches and hybrid assemblies.</p></li>
</ul>
<p><strong>3. Merqury: K-mer based assembly evaluation</strong></p>
<ul>
<li>
<strong>Assembly Accuracy Check</strong>: Merqury compares k-mers
from sequencing reads to the assembled genome, identifying mismatches,
missing k-mers, and sequencing errors without requiring a reference
genome.<br>
</li>
<li>
<strong>Haplotype Purity and Phasing</strong>: It calculates
<strong>QV (quality value)</strong> scores and provides
<strong>completeness metrics for haplotypes</strong>, helping assess
whether an assembly accurately represents both parental haplotypes or
contains chimeric sequences.<br>
</li>
<li>
<strong>Consensus and Read Support Validation</strong>: By analyzing
k-mer spectra, Merqury detects <strong>underrepresented or
overrepresented regions</strong>, highlighting assembly errors,
collapsed repeats, or sequencing biases that may impact downstream
analyses.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/042f0ad9-e06e-4aa0-8faf-51f0d42e2983" alt="kmer spectra" class="figure mx-auto d-block"><div class="figcaption">kmer spectra</div>
</figure></section><section><h2 class="section-heading" id="bionano-and-hi-c-reads-in-genome-assembly">Bionano and Hi-C Reads in Genome Assembly<a class="anchor" aria-label="anchor" href="#bionano-and-hi-c-reads-in-genome-assembly"></a>
</h2>
<hr class="half-width">
<p><strong>Bionano Optical Genome Mapping (OGM)</strong> provides
ultra-long, label-based maps of DNA molecules, helping to scaffold
contigs, detect misassemblies, and resolve large structural variations.
It improves genome continuity by linking fragmented sequences,
especially in repeat-rich or complex genomes.</p>
<p><strong>Hi-C sequencing</strong> captures chromatin interactions,
allowing scaffolding of contigs into chromosome-scale assemblies based
on physical proximity in the nucleus. It helps in ordering and orienting
scaffolds, identifying misassemblies, and resolving haplotypes, making
it essential for generating <strong>chromosome-level genome
assemblies</strong>.</p>
<figure><img src="https://github.com/user-attachments/assets/7f983713-7ea7-4522-8540-349955a9f12e" alt="Bionano and HiC for assembly improvement" class="figure mx-auto d-block"><div class="figcaption">Bionano and HiC for assembly improvement</div>
</figure><p><a href="https://doi.org/10.1038/ng.3802" class="external-link">ref</a></p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Genome assembly strategy depends on read type, genome complexity,
and computational resources, with PacBio HiFi, ONT, and hybrid
approaches offering different advantages in accuracy, cost, and
contiguity.<br>
</li>
<li>Assembly evaluation is critical for assessing completeness and
accuracy, using tools like <strong>BUSCO for gene completeness, QUAST
for structural integrity, and Merqury for k-mer-based
validation</strong>.<br>
</li>
<li>Scaffolding methods like Bionano OGM and Hi-C improve genome
organization, resolving large structural variations and ordering contigs
into chromosome-level assemblies.<br>
</li>
<li>A well-assembled genome is essential for downstream applications
such as annotation, comparative genomics, and structural variation
analysis, with missing or misassembled regions potentially leading to
incorrect biological conclusions.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-3-data-quality-control"><p>Content from <a href="3-data-quality-control.html">Data Quality Control</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/3-data-quality-control.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is data quality checking and filtering?</li>
<li>Why is it necessary to assess the quality of raw sequencing
data?</li>
<li>What are the key steps in filtering long-read sequencing data?</li>
<li>How can visualization tools like NanoPlot help in quality
assessment?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the importance of data quality checking and filtering in
genome assembly.</li>
<li>Learn how to assess raw sequencing data quality using NanoPlot.</li>
<li>Gain hands-on experience in filtering long-read sequencing data with
Filtlong.</li>
<li>Evaluate the impact of filtering on data quality using
NanoPlot.</li>
<li>Explore k-mer analysis for quality assessment (optional).</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="data-quality-check-and-filtering">Data Quality Check and Filtering<a class="anchor" aria-label="anchor" href="#data-quality-check-and-filtering"></a>
</h2>
<hr class="half-width">
<ul>
<li>
<strong>What is data quality checking and filtering?</strong>
<ul>
<li>Assessing the quality of raw sequencing data before genome
assembly.</li>
<li>Identifying and removing low-quality or problematic reads before
assembly.<br>
</li>
<li>Ensuring that the data is suitable for downstream analysis.</li>
<li>Better quality ingredients make a better quality cake!</li>
</ul>
</li>
<li>
<strong>Why is it necessary?</strong>
<ul>
<li>Poor quality data can lead to errors in genome assembly.</li>
<li>Low-quality reads can introduce gaps, misassemblies, or incorrect
base calls.</li>
<li>Filtering out low-quality reads can improve the accuracy and
efficiency of assembly.</li>
<li>It is a critical step to ensure the success of downstream
analyses.</li>
</ul>
</li>
<li>
<strong>What we will do today?</strong>
<ul>
<li>Learn the process of long-read generation (PacBio HiFi and ONT) - a
brief overview.</li>
<li>Assess raw data quality using <code>NanoPlot</code> for PacBio HiFi
and ONT reads - hands-on.<br>
</li>
<li>Filter reads based on quality and length using <code>Filtlong</code>
- hands-on.</li>
<li>Re-evaluate data post-filtering using <code>NanoPlot</code> to
confirm improvements - hands-on.<br>
</li>
<li>K-mer analysis to ensure data quality (optional hands-on).</li>
</ul>
</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/21f98d7a-21b0-4e4d-aa48-b1b8edbb9dfc" alt="Overview of steps" class="figure mx-auto d-block"><div class="figcaption">Overview of steps</div>
</figure></section><section><h2 class="section-heading" id="pacbio-hifi-reads-from-subreads">PacBio HiFi reads from Subreads<a class="anchor" aria-label="anchor" href="#pacbio-hifi-reads-from-subreads"></a>
</h2>
<hr class="half-width">
<p><strong>PacBio HiFi</strong> reads can be generated using <a href="https://ccs.how/" class="external-link">Circular Consensus Sequencing</a> (CCS) program
from <strong>PacBio</strong></p>
<ul>
<li>PacBio circular consensus sequencing (CCS) generates HiFi reads by
sequencing the same DNA molecule multiple times<br>
</li>
<li>The more passes over the same molecule, the higher the consensus
accuracy<br>
</li>
<li>Two main stages:
<ul>
<li>Generate a draft consensus from multiple subreads<br>
</li>
<li>Iteratively polish the consensus using all subreads to refine
accuracy</li>
</ul>
</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/566db0d0-93e8-40a3-82ce-89ec79c7a8b4" alt="PacBio CCS overview" class="figure mx-auto d-block"><div class="figcaption">PacBio CCS overview</div>
</figure><p><strong>What does CCS do?</strong></p>
<ol style="list-style-type: decimal">
<li>Filter low-quality subreads based on length and signal-to-noise
ratio<br>
</li>
<li>Generate an initial consensus using overlapping subreads<br>
</li>
<li>Align all subreads to the draft consensus for refinement<br>
</li>
<li>Divide sequence into small overlapping windows to optimize
polishing<br>
</li>
<li>Identify and correct errors, heteroduplex artifacts, and large
insertions<br>
</li>
<li>Apply polishing algorithms to refine the sequence, removing
ambiguities<br>
</li>
<li>Compute read accuracy based on error likelihood<br>
</li>
<li>Output the final HiFi read if accuracy meets the threshold</li>
</ol>
<p><strong>Why is this important?</strong></p>
<ul>
<li>Produces highly accurate long reads (99 percent or higher)<br>
</li>
<li>Enables better genome assembly, especially in complex or repetitive
regions<br>
</li>
<li>Reduces the need for additional error correction</li>
</ul>
<p><strong>How was the data generated?</strong></p>
<p>Subreads (in bam format) were converted to ccs fastq as follows:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">ccs</span> <span class="dt">\</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>   <span class="at">--hifi-kinetics</span> <span class="dt">\</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>   <span class="at">--num-threads</span> <span class="va">$SLURM_CPUS_ON_NODE</span> <span class="dt">\</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>   input.subreads.bam <span class="dt">\</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>   output.hifi.bam</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="ex">samtools</span> fastq <span class="dt">\</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>   output.hifi.bam <span class="op">&gt;</span> output.hifi.fastq</span></code></pre>
</div>
<p><strong>What is the source of this data?</strong></p>
<ul>
<li>The PacBio HiFi reads are from the project <a href="https://www.ebi.ac.uk/ena/browser/view/PRJEB50694" class="external-link">PRJEB50694</a>.</li>
<li>Data is for <em>Arabidopsis thaliana</em> ecotype Col-0, sequenced
using PacBio HiFi technology (they also sequenced CLR data for this
project).</li>
<li>The data is publicly available on the European Nucleotide Archive
(ENA), and the <code>9994.q20.CCS.fastq.gz</code> reads were used for
analysis.</li>
<li>Data has been filtered to include only HiFi reads with a Q-score of
20 or higher.</li>
</ul></section><section><h2 class="section-heading" id="ont-reads-from-minion-sequencing">ONT Reads from MinION Sequencing<a class="anchor" aria-label="anchor" href="#ont-reads-from-minion-sequencing"></a>
</h2>
<hr class="half-width">
<p><strong>Oxford Nanopore Technologies (ONT)</strong> sequencing
generates long reads by passing DNA through a biological nanopore</p>
<ul>
<li>ONT reads are generated in real-time as the DNA strand moves through
the nanopore.</li>
<li>Dorado is Oxford Nanopore Technologies’ basecaller that converts raw
electrical signals from nanopores into nucleotide sequences using
machine learning.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/624b4209-5f34-4d03-ad12-de95ae85ad75" alt="Overview of ONT read generation" class="figure mx-auto d-block"><div class="figcaption">Overview of ONT read generation</div>
</figure><p><strong>Key steps in <code>dorado</code> base calling</strong></p>
<ol style="list-style-type: decimal">
<li>
<strong>Signal detection</strong>
<ul>
<li>DNA or RNA molecules pass through a nanopore, disrupting an
electrical current.<br>
</li>
<li>These disruptions create a unique signal pattern called a
<strong>squiggle</strong>.</li>
</ul>
</li>
<li>
<strong>Real-time data processing</strong>
<ul>
<li>The MinKNOW software captures and processes the squiggle into
sequencing reads.<br>
</li>
<li>Reads include standard nucleotides and potential base modifications
like methylation.</li>
</ul>
</li>
<li>
<strong>Basecalling with machine learning</strong>
<ul>
<li>Neural networks, including transformer models, predict the
nucleotide sequence from raw signals.<br>
</li>
<li>Models continuously improve by training on diverse sequencing
data.</li>
</ul>
</li>
<li>
<strong>Error correction and refinement</strong>
<ul>
<li>Dorado refines base predictions to reduce errors, especially in
homopolymer regions.<br>
</li>
<li>Models are optimized for accuracy across various DNA/RNA types.</li>
</ul>
</li>
<li>
<strong>High-speed processing</strong>
<ul>
<li>Basecalling can be performed <strong>during sequencing</strong> for
real-time analysis or <strong>after sequencing</strong> for higher
accuracy.<br>
</li>
<li>GPUs accelerate computation, enabling rapid basecalling and
simultaneous modification detection.</li>
</ul>
</li>
</ol>
<p><strong>Why this matters</strong></p>
<ul>
<li>Enables <strong>real-time sequencing and analysis</strong> for quick
decision-making.<br>
</li>
<li>Uses <strong>advanced machine learning</strong> to improve sequence
accuracy over time.<br>
</li>
<li>Supports <strong>epigenetic modification detection</strong> without
extra processing steps.</li>
</ul>
<p><strong>How was the data generated?</strong></p>
<p><code>pod5</code> ONT reads were basecalled using Dorado as
follows:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># download</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">wget</span> ftp.sra.ebi.ac.uk/vol1/run/ERR791/ERR7919757/Arabidopsis-pass.tar.gz</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co"># extract</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">tar</span> <span class="at">-xvf</span> Arabidopsis-pass.tar.gz <span class="co"># fast5_pass is the extracted directory </span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co"># convert to pod5 format</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="ex">pod5</span> convert <span class="dt">\</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>   fast5 fast5_pass <span class="at">--output</span> pod5_pass</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co"># download model</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="ex">dorado</span> download <span class="dt">\</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>   <span class="at">--model</span> <span class="st">"dna_r10.4.1_e8.2_400bps_hac@v3.5.2"</span> <span class="dt">\</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>   <span class="at">--models-directory</span> models_dir/</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co"># basecall</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="ex">dorado</span> basecaller <span class="dt">\</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>   <span class="at">--emit-fastq</span> <span class="dt">\</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>   <span class="at">--output-dir</span> dorado_output_dir <span class="dt">\</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>   models_dir/dna_r10.4.1_e8.2_400bps_hac@v3.5.2 <span class="dt">\</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a>   input_pass.pod5</span></code></pre>
</div>
<p><strong>What is the source of this data?</strong></p>
<ul>
<li>The ONT reads are from the project <a href="https://www.ebi.ac.uk/ena/browser/view/PRJEB49840" class="external-link">PRJEB49840</a>.</li>
<li>Data is for <em>Arabidopsis thaliana</em> ecotype Col-0, sequenced
using R10.4/Q20+ chemistry from MinION cell</li>
<li>The data is publicly available on the European Nucleotide Archive
(ENA), and the <code>pass_fast5</code> reads were used for basecalling
(with commands above).</li>
</ul></section><section><h2 class="section-heading" id="a--nanoplot-for-quality-assessment">A. NanoPlot for Quality Assessment<a class="anchor" aria-label="anchor" href="#a--nanoplot-for-quality-assessment"></a>
</h2>
<hr class="half-width">
<p><a href="https://github.com/wdecoster/NanoPlot" class="external-link">NanoPlot</a> <a href="https://doi.org/10.1093/bioinformatics/bty149" class="external-link">ref</a> is a
visualization tool designed for quality assessment of long-read
sequencing data. It generates a variety of plots, including read length
histograms, cumulative yield plots, violin plots of read length and
quality over time, and bivariate plots that compare read lengths,
quality scores, reference identity, and mapping quality. By providing
both single-variable and density-based visualizations, NanoPlot helps
users quickly assess sequencing run quality and detect potential issues.
The tool also allows downsampling, length and quality filtering, and
barcode-specific analysis for multiplexed experiments.</p>
<p><strong>1. Quality Assessment of PacBio HiFi Reads</strong></p>
<p>Assessing the quality of ONT reads using <code>NanoPlot</code>.
Create a slurm script to run NanoPlot on the HiFi reads.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="ex">ml</span> nanoplot</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">NanoPlot</span> <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>   <span class="at">--verbose</span> <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>   <span class="at">--outdir</span> nanoplot_pacbio_pre <span class="dt">\</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>   <span class="at">--prefix</span> At_PacBio_ <span class="dt">\</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>   <span class="at">--plots</span> kde <span class="dt">\</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>   <span class="at">--N50</span> <span class="dt">\</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>   <span class="at">--dpi</span> 300 <span class="dt">\</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>   <span class="at">--fastq</span> At_pacbio-hifi.fastq.gz</span></code></pre>
</div>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Show details </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" aria-labelledby="headingSpoiler1" data-bs-parent="#accordionSpoiler1">
<div class="accordion-body">
<p>The stdout from the NanoPlot run will look like this:</p>
<pre><code>2025-02-13 12:13:19,155 NanoPlot 1.44.1 started with arguments Namespace(threads=16, verbose=True, store=False, raw=False, huge=False, outdir='nanoplot_pacbio_output', no_static=False, prefix='At_PacBio_', tsv_stats=False, only_report=False, info_in_report=False, maxlength=None, minlength=None, drop_outliers=False, downsample=None, loglength=False, percentqual=False, alength=False, minqual=None, runtime_until=None, readtype='1D', barcoded=False, no_supplementary=False, color='#4CB391', colormap='Greens', format=['png'], plots=['kde'], legacy=None, listcolors=False, listcolormaps=False, no_N50=False, N50=True, title=None, font_scale=1, dpi=300, hide_stats=False, fastq=['9994.q20.CCS.fastq.gz'], fasta=None, fastq_rich=None, fastq_minimal=None, summary=None, bam=None, ubam=None, cram=None, pickle=None, feather=None, path='nanoplot_pacbio_output/At_PacBio_')
2025-02-13 12:13:19,156 Python version is: 3.9.21 | packaged by conda-forge | (main, Dec  5 2024, 13:51:40)  [GCC 13.3.0]
2025-02-13 12:13:19,186 Nanoget: Starting to collect statistics from plain fastq file.
2025-02-13 12:13:19,187 Nanoget: Decompressing gzipped fastq 9994.q20.CCS.fastq.gz
2025-02-13 12:29:10,170 Reduced DataFrame memory usage from 12.780670166015625Mb to 12.780670166015625Mb
2025-02-13 12:29:10,194 Nanoget: Gathered all metrics of 837586 reads
2025-02-13 12:29:10,538 Calculated statistics
2025-02-13 12:29:10,539 Using sequenced read lengths for plotting.
2025-02-13 12:29:10,556 NanoPlot:  Valid color #4CB391.
2025-02-13 12:29:10,557 NanoPlot:  Valid colormap Greens.
2025-02-13 12:29:10,582 NanoPlot:  Creating length plots for Read length.
2025-02-13 12:29:10,583 NanoPlot: Using 837586 reads with read length N50 of 22587bp and maximum of 57055bp.
2025-02-13 12:29:11,933 Saved nanoplot_pacbio_output/At_PacBio_WeightedHistogramReadlength  as png (or png for --legacy)
2025-02-13 12:29:12,443 Saved nanoplot_pacbio_output/At_PacBio_WeightedLogTransformed_HistogramReadlength  as png (or png for --legacy)
2025-02-13 12:29:12,899 Saved nanoplot_pacbio_output/At_PacBio_Non_weightedHistogramReadlength  as png (or png for --legacy)
2025-02-13 12:29:13,371 Saved nanoplot_pacbio_output/At_PacBio_Non_weightedLogTransformed_HistogramReadlength  as png (or png for --legacy)
2025-02-13 12:29:13,372 NanoPlot: Creating yield by minimal length plot for Read length.
2025-02-13 12:29:14,465 Saved nanoplot_pacbio_output/At_PacBio_Yield_By_Length  as png (or png for --legacy)
2025-02-13 12:29:14,466 Created length plots
2025-02-13 12:29:14,474 NanoPlot: Creating Read lengths vs Average read quality plots using 837586 reads.
2025-02-13 12:29:15,012 Saved nanoplot_pacbio_output/At_PacBio_LengthvsQualityScatterPlot_kde  as png (or png for --legacy)
2025-02-13 12:29:15,013 Created LengthvsQual plot
2025-02-13 12:29:15,013 Writing html report.
2025-02-13 12:29:15,029 Finished!
</code></pre>
</div>
</div>
</div>
</div>
<p><strong>Evaluate the quality of HiFi reads:</strong></p>
<p>Examine the <code>At_PacBio_NanoPlot-report.html</code> file</p>
<ul>
<li>
<strong>Read length distribution</strong>: Histogram of read
lengths, showing the distribution of read lengths in the dataset.</li>
<li>
<strong>Read length vs. Quality</strong>: Scatter plot showing the
relationship between read length and quality score.</li>
<li>
<strong>Yield (number of bases) by read length</strong>: Plot
showing the cumulative yield of reads based on their length.</li>
<li>
<strong>Log-Transformed histograms</strong>: Histogram of read
lengths with a log-transformed scale for better visualization.</li>
<li>
<strong>KDE plots</strong>: Kernel Density Estimation plots for read
length and quality score distributions.</li>
<li>
<strong>Summary statistics</strong>: N50 value, maximum read length,
and other key metrics.</li>
</ul>
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p><strong>What filtering should be applied to the PacBio HiFi reads
based on the quality assessment?</strong></p>
<p>Our genome (<em>A. thaliana</em>) has a genome size of ~135 Mb. Our
target coverage is ~40x. Currently we have ~18Gb of HiFi reads (~138X
depth of coverage). We need to filter the reads to ensure we have good
quality reads of desired length and coverage.</p>
</div>
</div>
</div>
<p><strong>2. Quality Assessment of ONT Reads</strong></p>
<p>Assessing the quality of ONT reads using <code>NanoPlot</code>.
Create a slurm script to run NanoPlot on the basecalled ONT reads.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="ex">ml</span> nanoplot</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="ex">NanoPlot</span> <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>   <span class="at">--verbose</span> <span class="dt">\</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>   <span class="at">--outdir</span> nanoplot_ont_pre <span class="dt">\</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>   <span class="at">--prefix</span> At_ONT_ <span class="dt">\</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>   <span class="at">--readtype</span> 1D <span class="dt">\</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>   <span class="at">--plots</span> kde <span class="dt">\</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>   <span class="at">--N50</span> <span class="dt">\</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>   <span class="at">--dpi</span> 300 <span class="dt">\</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>   <span class="at">--fastq</span> At_ont-reads.fastq.gz</span></code></pre>
</div>
<div id="accordionSpoiler2" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler2" aria-expanded="false" aria-controls="collapseSpoiler2">
  <h3 class="accordion-header" id="headingSpoiler2">  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div> Show details </h3>
</button>
<div id="collapseSpoiler2" class="accordion-collapse collapse" aria-labelledby="headingSpoiler2" data-bs-parent="#accordionSpoiler2">
<div class="accordion-body">
<p>The stdout from the NanoPlot run will look like this:</p>
<pre><code>2025-02-13 12:15:51,066 NanoPlot 1.44.1 started with arguments Namespace(threads=8, verbose=True, store=False, raw=False, huge=False, outdir='nanoplot_pacbio_output', no_static=False, prefix='At_ONT_', tsv_stats=False, only_report=False, info_in_report=False, maxlength=None, minlength=None, drop_outliers=False, downsample=None, loglength=False, percentqual=False, alength=False, minqual=None, runtime_until=None, readtype='1D', barcoded=False, no_supplementary=False, color='#4CB391', colormap='Greens', format=['png'], plots=['kde'], legacy=None, listcolors=False, listcolormaps=False, no_N50=False, N50=True, title=None, font_scale=1, dpi=300, hide_stats=False, fastq=['basecalled_2025-02-12.fastq'], fasta=None, fastq_rich=None, fastq_minimal=None, summary=None, bam=None, ubam=None, cram=None, pickle=None, feather=None, path='nanoplot_pacbio_output/At_ONT_')
2025-02-13 12:15:51,067 Python version is: 3.9.21 | packaged by conda-forge | (main, Dec  5 2024, 13:51:40)  [GCC 13.3.0]
2025-02-13 12:15:51,096 Nanoget: Starting to collect statistics from plain fastq file.
2025-02-13 12:25:11,429 Reduced DataFrame memory usage from 8.842315673828125Mb to 8.842315673828125Mb
2025-02-13 12:25:11,455 Nanoget: Gathered all metrics of 579482 reads
2025-02-13 12:25:11,692 Calculated statistics
2025-02-13 12:25:11,693 Using sequenced read lengths for plotting.
2025-02-13 12:25:11,707 NanoPlot:  Valid color #4CB391.
2025-02-13 12:25:11,707 NanoPlot:  Valid colormap Greens.
2025-02-13 12:25:11,725 NanoPlot:  Creating length plots for Read length.
2025-02-13 12:25:11,725 NanoPlot: Using 579482 reads with read length N50 of 36292bp and maximum of 298974bp.
2025-02-13 12:25:13,096 Saved nanoplot_pacbio_output/At_ONT_WeightedHistogramReadlength  as png (or png for --legacy)
2025-02-13 12:25:13,571 Saved nanoplot_pacbio_output/At_ONT_WeightedLogTransformed_HistogramReadlength  as png (or png for --legacy)
2025-02-13 12:25:14,971 Saved nanoplot_pacbio_output/At_ONT_Non_weightedHistogramReadlength  as png (or png for --legacy)
2025-02-13 12:25:15,440 Saved nanoplot_pacbio_output/At_ONT_Non_weightedLogTransformed_HistogramReadlength  as png (or png for --legacy)
2025-02-13 12:25:15,441 NanoPlot: Creating yield by minimal length plot for Read length.
2025-02-13 12:25:16,485 Saved nanoplot_pacbio_output/At_ONT_Yield_By_Length  as png (or png for --legacy)
2025-02-13 12:25:16,486 Created length plots
2025-02-13 12:25:16,495 NanoPlot: Creating Read lengths vs Average read quality plots using 579482 reads.
2025-02-13 12:25:17,029 Saved nanoplot_pacbio_output/At_ONT_LengthvsQualityScatterPlot_kde  as png (or png for --legacy)
2025-02-13 12:25:17,030 Created LengthvsQual plot
2025-02-13 12:25:17,030 Writing html report.
2025-02-13 12:25:17,047 Finished!</code></pre>
</div>
</div>
</div>
</div>
<p><strong>Evaluate the quality of ONT reads:</strong></p>
<p>Examine the <code>At_ONT_NanoPlot-report.html</code> file.</p>
<ul>
<li>
<strong>Read length distribution</strong>: Histogram of read
lengths, showing the distribution of read lengths in the dataset.</li>
<li>
<strong>Read length vs. Quality</strong>: Scatter plot showing the
relationship between read length and quality score.</li>
<li>
<strong>Yield (number of bases) by read length</strong>: Plot
showing the cumulative yield of reads based on their length.</li>
<li>
<strong>Log-Transformed histograms</strong>: Histogram of read
lengths with a log-transformed scale for better visualization.</li>
<li>
<strong>KDE plots</strong>: Kernel Density Estimation plots for read
length and quality score distributions.</li>
<li>
<strong>Summary statistics</strong>: N50 value, maximum read length,
and other key metrics.</li>
</ul>
<div id="callout2" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p><strong>What filtering should be applied to the ONT reads based on
the quality assessment?</strong></p>
<p>Our genome (<em>A. thaliana</em>) has a genome size of ~135 Mb. Our
target coverage is ~40x. Currently we have ~14Gb of ONT reads (104X
depth of coverage). We need to filter the reads to ensure we have good
quality reads of desired length and coverage.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="b--filtering-sequencing-reads">B. Filtering Sequencing Reads<a class="anchor" aria-label="anchor" href="#b--filtering-sequencing-reads"></a>
</h2>
<hr class="half-width">
<p><a href="https://github.com/rrwick/Filtlong" class="external-link"><code>Filtlong</code></a> is a
tool designed to filter long-read sequencing data by selecting a
smaller, higher-quality subset of reads based on length and identity. It
prioritizes longer reads with higher sequence identity while discarding
shorter or lower-quality reads, ensuring that the retained data
contributes to more accurate genome assemblies. This filtering step is
crucial for improving assembly contiguity, reducing errors, and
optimizing computational efficiency by removing excess low-quality
data.</p>
<p><strong>1. Filtering PacBio HiFi Reads</strong></p>
<p>Filter the PacBio HiFi reads using <code>Filtlong</code> to retain
only high-quality reads.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="ex">ml</span> filtlong</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="ex">filtlong</span> <span class="dt">\</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>   <span class="at">--target_bases</span> 5400000000 <span class="dt">\</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>   <span class="at">--keep_percent</span> 90 <span class="dt">\</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>   <span class="at">--min_length</span> 1000 <span class="dt">\</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>     At_pacbio-hifi.fastq.gz <span class="op">&gt;</span> At_pacbio-hifi-filtered.fastq</span></code></pre>
</div>
<p><strong>2. Filtering ONT Reads</strong></p>
<p>Filter the ONT reads using <code>Filtlong</code> to retain only
high-quality reads.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="ex">ml</span> filtlong</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="ex">filtlong</span> <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>   <span class="at">--target_bases</span> 5400000000 <span class="dt">\</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>   <span class="at">--keep_percent</span> 90 <span class="dt">\</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>   <span class="at">--min_length</span> 1000 <span class="dt">\</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>     At_ont-reads.fastq.gz <span class="op">&gt;</span> At_ont-reads-filtered.fastq</span></code></pre>
</div>
<div id="callout3" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p><strong>What does this command do?</strong></p>
<ul>
<li>
<code>--target_bases 5400000000</code>: Target number of bases to
retain in the filtered dataset (5.4 Gb).</li>
<li>
<code>--keep_percent 90</code>: Retain reads that cover 90% of the
target bases.</li>
<li>
<code>--min_length 1000</code>: Minimum read length to keep in the
filtered dataset (1000 bp).</li>
</ul>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="c--evaluating-data-quality-after-filtering">C. Evaluating Data Quality After Filtering<a class="anchor" aria-label="anchor" href="#c--evaluating-data-quality-after-filtering"></a>
</h2>
<hr class="half-width">
<p>We will re-run <code>NanoPlot</code> on the filtered HiFi and ONT
reads to assess the quality of the filtered datasets.</p>
<p><strong>1. For PacBio HiFi Reads:</strong></p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="ex">ml</span> nanoplot</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="ex">NanoPlot</span> <span class="dt">\</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>   <span class="at">--verbose</span> <span class="dt">\</span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>   <span class="at">--outdir</span> nanoplot_pacbio_post <span class="dt">\</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>   <span class="at">--prefix</span> At_PacBio_post_ <span class="dt">\</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>   <span class="at">--plots</span> kde <span class="dt">\</span></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>   <span class="at">--N50</span> <span class="dt">\</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>   <span class="at">--dpi</span> 300 <span class="dt">\</span></span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>   <span class="at">--fastq</span> At_pacbio-hifi-filtered.fastq</span></code></pre>
</div>
<p><strong>2. For ONT Reads:</strong></p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="ex">ml</span> nanoplot</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="ex">NanoPlot</span> <span class="dt">\</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>   <span class="at">--verbose</span> <span class="dt">\</span></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>   <span class="at">--outdir</span> nanoplot_ont_post <span class="dt">\</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>   <span class="at">--prefix</span> At_ONT_post_ <span class="dt">\</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>   <span class="at">--readtype</span> 1D <span class="dt">\</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>   <span class="at">--plots</span> kde <span class="dt">\</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>   <span class="at">--N50</span> <span class="dt">\</span></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a>   <span class="at">--dpi</span> 300 <span class="dt">\</span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a>   <span class="at">--fastq</span> At_ont-reads-filtered.fastq</span></code></pre>
</div>
<p>Now, examine the <code>At_PacBio_post_NanoPlot-report.html</code> and
<code>At_ONT_post_NanoPlot-report.html</code> files to assess the
quality of the filtered HiFi and ONT reads. Do you observe any
improvements in read quality after filtering? We will use these filtered
reads for downstream genome assembly.</p>
</section><section><h2 class="section-heading" id="d--k-mer-based-quality-checks-optional">D. K-mer Based Quality Checks (Optional)<a class="anchor" aria-label="anchor" href="#d--k-mer-based-quality-checks-optional"></a>
</h2>
<hr class="half-width">
<p><a href="https://github.com/tbenavi1/genomescope2.0" class="external-link">GenomeScope</a>
is a k-mer-based tool used to profile genomes without requiring a
reference, providing estimates of genome size, heterozygosity, and
repeat content. It uses k-mer frequency distributions from raw
sequencing reads to model genome characteristics, making it especially
useful for detecting sequencing artifacts and assessing data quality
before assembly. In this optional section, we will use GenomeScope to
evaluate the quality of our Oxford Nanopore and PacBio reads by
identifying potential errors, biases, and coverage issues, helping to
refine filtering strategies and improve downstream assembly results <a href="https://www.nature.com/articles/s41467-020-14998-3" class="external-link">ref</a>.</p>
<p><strong>1. For PacBio HiFi Reads:</strong></p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="ex">ml</span> kmc</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="fu">mkdir</span> tmp</span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="fu">ls</span> At_pacbio-hifi-filtered.fastq <span class="op">&gt;</span> FILES</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="ex">kmc</span> <span class="at">-k21</span> <span class="at">-t10</span> <span class="at">-m64</span> <span class="at">-ci1</span> <span class="at">-cs10000</span> @FILES reads tmp/</span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="ex">kmc_tools</span> transform reads histogram reads-pacbio.histo <span class="at">-cx10000</span></span></code></pre>
</div>
<p><strong>2. For ONT Reads:</strong></p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="ex">ml</span> kmc</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="fu">mkdir</span> tmp</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="fu">ls</span> At_ont-reads-filtered.fastq <span class="op">&gt;</span> FILES</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a><span class="ex">kmc</span> <span class="at">-k21</span> <span class="at">-t10</span> <span class="at">-m64</span> <span class="at">-ci1</span> <span class="at">-cs10000</span> @FILES reads tmp/</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a><span class="ex">kmc_tools</span> transform reads histogram reads-ont.histo <span class="at">-cx10000</span></span></code></pre>
</div>
<p>Now you can visualize the k-mer frequency distributions using
GenomeScope to assess the quality of the HiFi and ONT reads. This
analysis can help identify potential issues and guide further filtering
or processing steps to improve data quality.</p>
<p>To visualize the k-mer frequency distributions:</p>
<ol style="list-style-type: decimal">
<li>Visit the <a href="http://genomescope.org/genomescope2.0/" class="external-link">GenomeScope website</a>
and upload the <code>reads-pacbio.histo</code> OR
<code>reads-ont.histo</code> files (drag and drop).</li>
<li>If you uploaded the <code>reads-pacbio.histo</code> file, enter
description with <code>PacBio HiFi</code>. If you uploaded the
<code>reads-ont.histo</code> file, enter description with
<code>Oxford Nanopore</code>.</li>
<li>Click on the <code>Submit</code> button to generate the k-mer
frequency distribution plots.</li>
</ol>
<div id="challenge1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Challenge</h3>
<div class="callout-content">
<p>Q: <em>Why did the ONT k-mer analysis fail?</em></p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>A: High error rates in ONT reads can lead to k-mer counting errors,
causing the analysis to fail. K-mer analyses reads rely on accuracy to
generate reliable frequency distributions. Only reads higher than Q20
are recommended for k-mer analysis.</p>
</div>
</div>
</div>
</div>
<div id="callout4" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p><strong>What insights can you gain from the k-mer frequency
distributions?</strong></p>
<ul>
<li>Look for peaks and patterns in the k-mer frequency
distributions.</li>
<li>Identify potential issues such as heterozygosity, repeat content, or
sequencing errors.</li>
<li>Did your models converge? What does this indicate about the quality
of your data?</li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>
<strong>Data Quality Control</strong>: Assessing and filtering raw
sequencing data is essential for accurate genome assembly.</li>
<li>
<strong>NanoPlot</strong>: Visualizes read length distributions,
quality scores, and other metrics to evaluate sequencing data
quality.</li>
<li>
<strong>Filtlong</strong>: Filters long-read sequencing data based
on length and quality to retain high-quality reads.</li>
<li>
<strong>GenomeScope</strong>: Profiles genomes using k-mer frequency
distributions to estimate genome size, heterozygosity, and repeat
content.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-4-pacbio-hifi-assembly"><p>Content from <a href="4-pacbio-hifi-assembly.html">PacBio HiFi Assembly using HiFiasm</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/4-pacbio-hifi-assembly.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is HiFiasm, and how does it improve genome assembly using
PacBio HiFi reads?</li>
<li>What are the key steps in running HiFiasm for haplotype-resolved
assembly?</li>
<li>How does HiFiasm handle haplotype resolution and purging of
duplications?</li>
<li>What are the benefits of using HiFiasm for assembling complex and
heterozygous genomes?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the purpose and function of HiFiasm for
haplotype-resolved genome assembly.</li>
<li>Learn to set up and run HiFiasm for assembling genomes using PacBio
HiFi reads.</li>
<li>Gain hands-on experience with haplotype resolution and purging of
duplications in HiFiasm.</li>
<li>Analyze and interpret HiFiasm output to assess assembly quality and
completeness.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction-to-hifiasm">Introduction to HiFiasm<a class="anchor" aria-label="anchor" href="#introduction-to-hifiasm"></a>
</h2>
<hr class="half-width">
<p><a href="https://www.nature.com/articles/s41592-020-01056-5" class="external-link">HiFiasm</a> is
a specialized de novo assembler designed for PacBio HiFi reads,
providing high-quality, haplotype-resolved genome assemblies. Unlike
traditional assemblers that collapse heterozygous regions into a
consensus sequence, HiFiasm preserves haplotype information using a
phased assembly graph approach. This enables more accurate
representation of genetic variations and structural differences.</p>
<p>Leveraging the low error rate of HiFi reads, HiFiasm constructs
phased assembly graphs that allow for haplotype separation without
requiring external polishing or duplication-purging tools. It
significantly improves assembly contiguity, resolving complex regions
more effectively than alternative methods. HiFiasm is widely used in
genome projects, including the Human Pangenome Project, and has been
successfully applied to large and highly heterozygous genomes such as
Sequoia sempervirens (~30 Gb).</p>
<p>With its ability to generate fast and accurate assemblies, HiFiasm
has become the preferred tool for haplotype-resolved genome assembly,
especially when parental reads or Hi-C data are available.</p>
<div id="latest-version-of-hifiasm" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="latest-version-of-hifiasm" class="callout-inner">
<h3 class="callout-title">Latest version of HiFiasm</h3>
<div class="callout-content">
<p>The latest version of HiFiasm supports assembling ONT reads as well.
It has also added support to integrate ultra-long ONT reads for improved
contiguity, as well as hybrid assembly (using both ONT and HiFi reads).
Apart from ONT data, HiFiasm can handle Hi-C data for scaffolding, as
well as kmer profiles from parents to resolve haplotypes. The latest
version includes several bug fixes and performance improvements, making
it more efficient and user-friendly.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="installation-and-setup">Installation and Setup<a class="anchor" aria-label="anchor" href="#installation-and-setup"></a>
</h2>
<hr class="half-width">
<p>HiFiasm is available as module on RCAC clusters. You can load the
module using the following command:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">ml</span> hifiasm</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="at">--version</span></span></code></pre>
</div>
<p>You can also use the Singularity container for HiFiasm, which
provides a consistent environment across different systems. The
container can be pulled from the BioContainers registry using the
following command:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">apptainer</span> pull docker://quay.io/biocontainers/hifiasm:0.24.0--h5ca1c30_0</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">apptainer</span> exec hifiasm_0.24.0--h5ca1c30_0.sif hifiasm <span class="at">--version</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="overview-of-hifiasm-read-assembly">Overview of HiFiASM Read Assembly<a class="anchor" aria-label="anchor" href="#overview-of-hifiasm-read-assembly"></a>
</h2>
<hr class="half-width">
<p>HiFiasm can assemble high-quality, contiguous genome sequences from
PacBio High-Fidelity (HiFi) reads. HiFi reads are long and highly
accurate (99%+), making them ideal for assembling complex genomes,
resolving repetitive regions, and distinguishing haplotypes in diploid
or polyploid organisms.</p>
<p>The assembly workflow typically involves:</p>
<ol style="list-style-type: decimal">
<li>Preprocessing reads – filtering and quality-checking raw hifi
reads</li>
<li>Read overlap detection – identifying how reads align to each
other</li>
<li>Error correction – resolving sequencing errors while maintaining
true haplotype differences</li>
<li>Graph construction – building an assembly graph to represent contig
relationships</li>
<li>Contig generation – extracting the final set of contiguous
sequences</li>
<li>Post-processing – refining assemblies by purging duplications or
scaffolding</li>
</ol>
<p>HiFiasm is optimized for this process, leveraging the high accuracy
of HiFi reads to generate contigs with minimal fragmentation and greater
haplotype resolution compared to traditional assemblers.</p>
<figure><img src="https://github.com/user-attachments/assets/1ab2a17c-ac29-42c7-a890-5c86577e2f19" alt="HiFiasm assembler" class="figure mx-auto d-block"><div class="figcaption">HiFiasm assembler</div>
</figure></section><section><h2 class="section-heading" id="hifiasm-basic-workflow">HiFiasm: basic workflow<a class="anchor" aria-label="anchor" href="#hifiasm-basic-workflow"></a>
</h2>
<hr class="half-width">
<p>To run HiFiasm, you need to provide the input HiFi reads in FASTA or
FASTQ format. The basic command structure is as follows:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="ex">ml</span> hifiasm</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>    <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>    <span class="at">-o</span> Athaliana_hifiasm_default.asm <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>    9994.q20.CCS-filtered.fastq</span></code></pre>
</div>
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>In this command:</p>
<ul>
<li>
<code>-t</code> specifies the number of threads to use</li>
<li>
<code>-o</code> specifies the output prefix for the assembly</li>
<li>last argument is the input HiFi reads file (fastq format)</li>
</ul>
<p>The input can either be fastq or fasta, compressed or uncompressed.
The output will be stored in the same directory with the specified
prefix.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="understanding-hifiasm-output">Understanding HiFiasm Output<a class="anchor" aria-label="anchor" href="#understanding-hifiasm-output"></a>
</h2>
<hr class="half-width">
<p>The run generates several output files. Here are all the files and
their descriptions:</p>
<table class="table">
<colgroup>
<col width="38%">
<col width="61%">
</colgroup>
<thead><tr class="header">
<th align="left">filename</th>
<th>description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>Athaliana_hifiasm_default.asm.ec.bin</code></td>
<td>error-corrected reads stored in binary format</td>
</tr>
<tr class="even">
<td align="left"><code>Athaliana_hifiasm_default.asm.ovlp.source.bin</code></td>
<td>source overlap data between reads in binary format</td>
</tr>
<tr class="odd">
<td align="left"><code>Athaliana_hifiasm_default.asm.ovlp.reverse.bin</code></td>
<td>reverse overlap data between reads in binary format</td>
</tr>
<tr class="even">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.r_utg.noseq.gfa</code></td>
<td>assembly graph of raw unitigs (without sequence)</td>
</tr>
<tr class="odd">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.r_utg.gfa</code></td>
<td>assembly graph of raw unitigs (with sequence)</td>
</tr>
<tr class="even">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.r_utg.lowQ.bed</code></td>
<td>low-quality regions in raw unitigs</td>
</tr>
<tr class="odd">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.p_utg.noseq.gfa</code></td>
<td>assembly graph of purged unitigs (without sequence)</td>
</tr>
<tr class="even">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.p_utg.gfa</code></td>
<td>assembly graph of purged unitigs (with sequence)</td>
</tr>
<tr class="odd">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.p_utg.lowQ.bed</code></td>
<td>low-quality regions in purged unitigs</td>
</tr>
<tr class="even">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.p_ctg.noseq.gfa</code></td>
<td>assembly graph of primary contigs (without sequence)</td>
</tr>
<tr class="odd">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.p_ctg.gfa</code></td>
<td>assembly graph of primary contigs (with sequence)</td>
</tr>
<tr class="even">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.p_ctg.lowQ.bed</code></td>
<td>low-quality regions in primary contigs</td>
</tr>
<tr class="odd">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.hap1.p_ctg.noseq.gfa</code></td>
<td>haplotype 1 primary contigs (without sequence)</td>
</tr>
<tr class="even">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.hap1.p_ctg.gfa</code></td>
<td>haplotype 1 primary contigs (with sequence)</td>
</tr>
<tr class="odd">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.hap2.p_ctg.noseq.gfa</code></td>
<td>haplotype 2 primary contigs (without sequence)</td>
</tr>
<tr class="even">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.hap2.p_ctg.gfa</code></td>
<td>haplotype 2 primary contigs (with sequence)</td>
</tr>
<tr class="odd">
<td align="left"><code>Athaliana_hifiasm_default.asm.bp.hap1.p_ctg.lowQ.bed</code></td>
<td>low-quality regions in haplotype 1 primary contigs</td>
</tr>
</tbody>
</table>
<div id="where-are-the-assemblycontig-sequences" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="where-are-the-assemblycontig-sequences" class="callout-inner">
<h3 class="callout-title">Where are the assembly/contig sequences?</h3>
<div class="callout-content">
<p>The <code>*_ctg.gfa</code> file contains the contigs
(haplotype-resolved, and primary only) in GFA (Graphical Fragment
Assembly) format. You can extract the sequences from this file using
<code>awk</code>. The sequences are represented as lines starting with
<code>S</code> followed by the contig ID and the sequence.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="cf">for</span> ctg <span class="kw">in</span> <span class="pp">*</span>_ctg.gfa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>    <span class="fu">awk</span> <span class="st">'/^S/{print "&gt;"$2"\n"$3}'</span> <span class="va">${ctg}</span> <span class="op">&gt;</span> <span class="va">${ctg</span><span class="op">%</span>.gfa<span class="va">}</span>.fasta</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
</div>
</div>
</div>
<p>Let’s take a look at the sats of this assembly:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="ex">ml</span> quast</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="ex">quast.py</span> <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>    <span class="at">--fast</span> <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>    <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>    <span class="at">-o</span> quast_basic_stats <span class="dt">\</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>    <span class="pp">*</span>p_ctg.fasta</span></code></pre>
</div>
<div id="quast-metrics" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="quast-metrics" class="callout-inner">
<h3 class="callout-title">Quast metrics</h3>
<div class="callout-content">
<p><strong>Key metrics for assembly quality assessment</strong></p>
<table class="table">
<colgroup>
<col width="12%">
<col width="87%">
</colgroup>
<thead><tr class="header">
<th align="left"><strong>Metric</strong></th>
<th><strong>Description &amp; Importance</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><strong># Contigs</strong></td>
<td>The number of contiguous sequences in the assembly. Fewer, larger
contigs indicate a more contiguous assembly.</td>
</tr>
<tr class="even">
<td align="left"><strong>Largest Contig</strong></td>
<td>The length of the longest assembled sequence. A larger value
suggests better resolution of large genomic regions.</td>
</tr>
<tr class="odd">
<td align="left"><strong>Total Length</strong></td>
<td>The sum of all contig lengths. Should approximately match the
expected genome size.</td>
</tr>
<tr class="even">
<td align="left"><strong>N50</strong></td>
<td>The contig length at which 50% of the assembly is covered. Higher
values indicate a more contiguous assembly.</td>
</tr>
<tr class="odd">
<td align="left"><strong>N90</strong></td>
<td>The contig length at which 90% of the assembly is covered. Provides
insight into the distribution of smaller contigs.</td>
</tr>
<tr class="even">
<td align="left"><strong>L50</strong></td>
<td>The number of contigs that make up 50% of the assembly. Lower values
indicate higher contiguity.</td>
</tr>
<tr class="odd">
<td align="left"><strong>L90</strong></td>
<td>The number of contigs that make up 90% of the assembly. Lower values
suggest fewer, larger contigs.</td>
</tr>
<tr class="even">
<td align="left"><strong>auN</strong></td>
<td>Weighted average of contig lengths, emphasizing longer contigs.
Higher values indicate better continuity.</td>
</tr>
<tr class="odd">
<td align="left"><strong># N/100 kbp</strong></td>
<td>Measures the presence of gaps (<code>N</code>s) in the assembly.
Ideally should be 0, meaning no unresolved bases.</td>
</tr>
</tbody>
</table>
<p><strong>How to interpret</strong></p>
<ul>
<li>
<strong>High N50 and low L50</strong> suggest a well-assembled
genome with fewer, larger contigs.<br>
</li>
<li>
<strong>Total Length</strong> should be close to the estimated
genome size, ensuring completeness.<br>
</li>
<li>
<strong>Low # of contigs</strong> indicates better continuity,
meaning fewer breaks in the genome.<br>
</li>
<li>
<strong>No <code>N</code> bases</strong> means the assembly is
gap-free and doesn’t contain unresolved regions.<br>
</li>
</ul>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="handling-haplotype-resolved-contigs">Handling haplotype-resolved contigs<a class="anchor" aria-label="anchor" href="#handling-haplotype-resolved-contigs"></a>
</h2>
<hr class="half-width">
<p>HiFiasm generates haplotype-resolved contigs. With the default
options above, you saw that it generated <code>hap1.p_ctg</code>,
<code>hap2.p_ctg</code> and <code>.p_ctg</code> GFA files, which
corresponds to haplotype 1, haplotype 2, and primary contigs,
respectively. Although HiFiasm separates the haplotypes, it is unable to
phase (assign the actual regions of hap1 and hap2 to their respective
haplotypes consistently across the genome) them without additional data.
The haplotype-resolved contigs, as-is, is still valuable information,
and can be used for downstream analyses requiring haplotype-specific
information. The primary contigs represent the consensus sequence, and
is usually more complete than either of the haplotype only
assemblies.</p>
<ul>
<li>Hifiasm purges haplotig duplications by default (to produce two sets
of partially phased contigs)</li>
<li>For inbred or homozygous genomes, you may disable purging with
option <code>-l 0</code>
((<code>hifiasm -o prefix.asm -l 0 -t ${SLURM_CPUS_ON_NODE} input.fq.gz</code>)</li>
<li>To get primary/alternate assemblies, the option
<code>--primary</code> should be set
(<code>hifiasm -o prefix.asm --primary -t ${SLURM_CPUS_ON_NODE} input.fq.gz</code>)</li>
<li>For heterozygous genomes, you can set <code>-l 1</code>,
<code>-l 2</code>, or <code>-l 3</code>, to adjust purging of haplotigs
<ul>
<li>
<code>-l 1</code> to only purge contained haplotigs</li>
<li>
<code>-l 2</code> to purge all types of haplotigs</li>
<li>
<code>-l 3</code> to purge all types of haplotigs in the most
aggressive way</li>
</ul>
</li>
<li>If you have parental kmer profiles, you can use them to resolve
haplotypes</li>
</ul>
<p>We can try running HiFiasm with various <code>-l</code> options to
see how it affects the assembly quality.</p>
<div class="tabs">
<nav><div id="nav-tab-1" class="nav nav-tabs" role="tablist">
<button class="nav-link active" id="nav-tab-1-1" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-1" type="button" role="tab" aria-controls="nav-tabpanel-1-1" aria-selected="true">
  <h3 class="tab-header" id="nav-tab-heading-1-1">
  purge l=0
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-2" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-2" type="button" role="tab" aria-controls="nav-tabpanel-1-2" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-2">
  purge l=1
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-3" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-3" type="button" role="tab" aria-controls="nav-tabpanel-1-3" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-3">
  purge l=2
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-4" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-4" type="button" role="tab" aria-controls="nav-tabpanel-1-4" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-4">
  purge l=3
  </h3>
</button>
</div>
</nav><div id="nav-tabContent-1" class="tab-content">
<div id="nav-tabpanel-1-1" class="tab-pane show active" aria-labelledby="nav-tab-1-1" role="tabpanel">
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="ex">ml</span> hifiasm</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co"># purge level 0</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> hifiasm_purge_level_0</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="bu">cd</span> hifiasm_purge_level_0</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ../9994.q20.CCS-filtered.fastq</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="dt">\</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>  <span class="at">-o</span> Athaliana_hifiasm_purge_level_0.asm <span class="dt">\</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>  <span class="at">-l</span> 0 <span class="dt">\</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>  <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>  9994.q20.CCS-filtered.fastq</span></code></pre>
</div>
</div>
<div id="nav-tabpanel-1-2" class="tab-pane" aria-labelledby="nav-tab-1-2" role="tabpanel">
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="ex">ml</span> hifiasm</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co"># purge level 1</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> hifiasm_purge_level_1</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="bu">cd</span> hifiasm_purge_level_1</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ../9994.q20.CCS-filtered.fastq</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="dt">\</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>  <span class="at">-o</span> Athaliana_hifiasm_purge_level_1.asm <span class="dt">\</span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>  <span class="at">-l</span> 1 <span class="dt">\</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>  <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>  9994.q20.CCS-filtered.fastq</span></code></pre>
</div>
</div>
<div id="nav-tabpanel-1-3" class="tab-pane" aria-labelledby="nav-tab-1-3" role="tabpanel">
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="ex">ml</span> hifiasm</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="co"># purge level 2</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> hifiasm_purge_level_2</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="bu">cd</span> hifiasm_purge_level_2</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ../9994.q20.CCS-filtered.fastq</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="dt">\</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>  <span class="at">-o</span> Athaliana_hifiasm_purge_level_2.asm <span class="dt">\</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>  <span class="at">-l</span> 2 <span class="dt">\</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>  <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>  9994.q20.CCS-filtered.fastq</span></code></pre>
</div>
</div>
<div id="nav-tabpanel-1-4" class="tab-pane" aria-labelledby="nav-tab-1-4" role="tabpanel">
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="ex">ml</span> hifiasm</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="co"># purge level 3</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> hifiasm_purge_level_3</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a><span class="bu">cd</span> hifiasm_purge_level_3</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ../9994.q20.CCS-filtered.fastq</span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="dt">\</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>  <span class="at">-o</span> Athaliana_hifiasm_purge_level_3.asm <span class="dt">\</span></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>  <span class="at">-l</span> 3 <span class="dt">\</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>  <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>  9994.q20.CCS-filtered.fastq</span></code></pre>
</div>
</div>
</div>
</div>
<div id="callout3" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p>Each of these will run in about ~15 minutes with 32 cores. You can
either run them in parallel or sequentially or request more cores to run
them faster.</p>
</div>
</div>
</div>
<div id="comparing-assemblies" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="comparing-assemblies" class="callout-inner">
<h3 class="callout-title">Comparing assemblies</h3>
<div class="callout-content">
<p>Convert GFA files to FASTA format</p>
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="cf">for</span> dir <span class="kw">in</span> hifiasm_purge_level_<span class="dt">{</span><span class="dv">0</span><span class="dt">..</span><span class="dv">3</span><span class="dt">}</span><span class="kw">;</span> <span class="cf">do</span> </span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>    <span class="bu">cd</span> <span class="va">${dir}</span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>    <span class="cf">for</span> ctg <span class="kw">in</span> <span class="pp">*</span>_ctg.gfa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>        <span class="fu">awk</span> <span class="st">'/^S/{print "&gt;"$2"\n"$3}'</span> <span class="va">${ctg}</span> <span class="op">&gt;</span> <span class="va">${ctg</span><span class="op">%</span>.gfa<span class="va">}</span>.fasta</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>    <span class="cf">done</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>    <span class="bu">cd</span> ..</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
<p>Run QUAST to compare the assemblies</p>
<div class="codewrapper sourceCode" id="cb11">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> quast_stats</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="cf">for</span> fasta <span class="kw">in</span> hifiasm_purge_level_{0..3}/<span class="pp">*</span>_p_ctg.fasta<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>    <span class="fu">ln</span> <span class="at">-s</span> <span class="va">${fasta}</span> quast_stats/</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="bu">cd</span> quast_stats</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="ex">quast.py</span> <span class="dt">\</span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>    <span class="at">--fast</span> <span class="dt">\</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>    <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a>    <span class="at">-o</span> quast_purge_level_stats <span class="dt">\</span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a>    <span class="pp">*</span>_p_ctg.fasta</span></code></pre>
</div>
<p>Run Compleasm to compare the assemblies</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="ex">ml</span> compleasm</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> compleasm_stats</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="cf">for</span> fasta <span class="kw">in</span> hifiasm_purge_level_{0..3}/<span class="pp">*</span>_p_ctg.fasta<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>    <span class="fu">ln</span> <span class="at">-s</span> <span class="va">${fasta}</span> compleasm_stats/</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a><span class="bu">cd</span> compleasm_stats</span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a><span class="cf">for</span> fasta <span class="kw">in</span> <span class="pp">*</span>_p_ctg.fasta<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a>    <span class="ex">compleasm</span> run <span class="dt">\</span></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a>       <span class="at">-a</span> <span class="va">${fasta}</span> <span class="dt">\</span></span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a>       <span class="at">-o</span> <span class="va">${fasta</span><span class="op">%</span>.<span class="pp">*</span><span class="va">}</span> <span class="dt">\</span></span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a>       <span class="at">-l</span> brassicales_odb10 <span class="dt">\</span></span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a>       <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
<p>Examining the results from QUAST and Compleasm, compare the assembly
statistics and assess the impact of different purging levels on the
assembly quality. Look for metrics like N50, L50, and total assembly
size to evaluate the contiguity and completeness of the assemblies.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="improving-assembly-quality">Improving Assembly Quality<a class="anchor" aria-label="anchor" href="#improving-assembly-quality"></a>
</h2>
<hr class="half-width">
<p>After the first round of assembly, you will have the files
<code>*.ec.bin</code>, <code>*.ovlp.source.bin</code>, and
<code>*.ovlp.reverse.bin</code>. Save these files and try various
options to see if you can improve the assembly. First, make a folder to
move the .gfa, .fasta, and .bed files. These are the results from the
first round of assembly. Second, adjust the parameters in the hifiasm
command and run the assembler again. Third, move results to a new folder
and compare the results of the first folder. You can re-run the assembly
quickly and generate statistics for each of these folders and compare
them to see if the changes improved the assembly.</p>
</section><section><h2 class="section-heading" id="alternative-assembler-flye-for-hifi">Alternative Assembler: Flye for HiFi<a class="anchor" aria-label="anchor" href="#alternative-assembler-flye-for-hifi"></a>
</h2>
<hr class="half-width">
<p>Flye is another popular assembler specialized for ONT reads, offering
a different approach to haplotype-resolved assembly. The latest version
can also use HiFi reads to generate great quality assemblies. We will
explore Flye in a separate episode to compare its performance with
HiFiasm. But in this optional section, you can try running Flye with
HiFi reads to see how it performs compared to HiFiasm.</p>
<div class="section level3">
<h3 id="running-flye-with-hifi-reads">Running Flye with HiFi Reads<a class="anchor" aria-label="anchor" href="#running-flye-with-hifi-reads"></a>
</h3>
<p>To run Flye with HiFi reads, you can use the following command
structure:</p>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a><span class="ex">ml</span> flye</span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="ex">flye</span> <span class="dt">\</span></span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>  <span class="at">--pacbio-hifi</span> 9994.q20.CCS-filtered-60x.fastq <span class="dt">\</span></span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>  <span class="at">--genome-size</span> 135m <span class="dt">\</span></span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>  <span class="at">--out-dir</span> flye_pcb <span class="dt">\</span></span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>  <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<div id="options-used" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="options-used" class="callout-inner">
<h3 class="callout-title">Options used</h3>
<div class="callout-content">
<p>In this command:</p>
<ul>
<li>
<code>--pacbio-hifi</code> specifies the input HiFi reads file</li>
<li>
<code>--genome-size</code> provides an estimate of the genome size
(optional)</li>
<li>
<code>--out-dir</code> specifies the output directory for Flye
results</li>
<li>
<code>--threads</code> specifies the number of threads to use</li>
</ul>
<p>The output will be stored in the specified directory, containing the
assembly graph, contigs, and other relevant files.</p>
<p>With 64 cores, this will run in about ~40 mins. It needs about
~80-90Gb of memory.</p>
</div>
</div>
</div>
<div id="quality-metrics" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="quality-metrics" class="callout-inner">
<h3 class="callout-title">Quality metrics</h3>
<div class="callout-content">
<p>Run quality metrics on Flye assembly:</p>
<div class="codewrapper sourceCode" id="cb14">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="ex">ml</span> quast</span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a><span class="ex">quast.py</span> <span class="dt">\</span></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>    <span class="at">--fast</span> <span class="dt">\</span></span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>    <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>    <span class="at">-o</span> quast_flye_stats <span class="dt">\</span></span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a>    flye_pcb/assembly.fasta</span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a><span class="ex">ml</span> compleasm</span>
<span id="cb14-10"><a href="#cb14-10" tabindex="-1"></a><span class="ex">compleasm</span> run <span class="dt">\</span></span>
<span id="cb14-11"><a href="#cb14-11" tabindex="-1"></a>  <span class="at">-a</span> flye_pcb/assembly.fasta <span class="dt">\</span></span>
<span id="cb14-12"><a href="#cb14-12" tabindex="-1"></a>  <span class="at">-o</span> flye_pcb <span class="dt">\</span></span>
<span id="cb14-13"><a href="#cb14-13" tabindex="-1"></a>  <span class="at">-l</span> brassicales_odb10 <span class="dt">\</span></span>
<span id="cb14-14"><a href="#cb14-14" tabindex="-1"></a>  <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>Which assembler did a better job at assembling the genome? Compare
the statistics from QUAST and Compleasm for Flye and HiFiasm assemblies
to evaluate their performance.</p>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>
<strong>HiFiasm</strong> is a specialized assembler for PacBio HiFi
reads, providing high-quality, haplotype-resolved genome
assemblies.</li>
<li>It leverages the high accuracy of HiFi reads to generate phased
assembly graphs, preserving haplotype information.</li>
<li>HiFiasm is optimized for resolving complex regions and
distinguishing haplotypes in diploid or polyploid organisms.</li>
<li>The assembler generates primary contigs and haplotype-resolved
contigs, offering valuable information for downstream analyses.</li>
<li>By adjusting purging levels and using parental kmer profiles, users
can improve haplotype resolution and assembly quality.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div>
</section></section><section id="aio-5-oxford-nanopore-assembly"><p>Content from <a href="5-oxford-nanopore-assembly.html">Oxford Nanopore Assembly using Flye</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/5-oxford-nanopore-assembly.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What are the key features of ONT reads?</li>
<li>Why is Flye good for assembling ONT reads?</li>
<li>What are the main steps in the Flye assembly workflow?</li>
<li>How can you evaluate the quality of a Flye assembly?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the characteristics of ONT reads.</li>
<li>Learn about the Flye assembler and its advantages for ONT data.</li>
<li>Explore the key steps in the Flye assembly workflow.</li>
<li>Evaluate the quality of a Flye assembly using common metrics.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction-to-ont-reads-and-flye-assembler">Introduction to ONT reads and Flye Assembler<a class="anchor" aria-label="anchor" href="#introduction-to-ont-reads-and-flye-assembler"></a>
</h2>
<hr class="half-width">
<p>Oxford Nanopore Technologies (ONT) has revolutionized sequencing by
providing long-read data, enabling the resolution of complex genomic
structures that were previously intractable with short-read
technologies. However, ONT reads are error-prone, necessitating
specialized assembly algorithms that can handle high sequencing error
rates while maximizing contiguity and accuracy.</p>
<p>Traditional assemblers designed for short reads rely on de Bruijn
graph approaches, which break sequences into fixed k-mers and struggle
with error-rich long reads. In contrast, modern long-read assemblers
like <code>Flye</code> use alternative graph-based strategies to
overcome these limitations. Flye specifically constructs repeat graphs
to accurately reconstruct genomes while addressing challenges posed by
structural variations and repeats. This makes it particularly
well-suited for ONT data, producing high-quality, contiguous assemblies
for small microbial genomes to large eukaryotic genomes.</p>
<p>The latest ultra-long ONT reads, such as those generated by the
PromethION platform, have further improved assembly quality and
contiguity. Flye can leverage these ultra-long reads to generate even
more accurate and contiguous assemblies, making it a powerful tool for a
wide range of genomic analyses.</p>
<figure><img src="https://github.com/user-attachments/assets/7a2559a7-df81-4e52-9962-1f2e41bfd6e7" alt="ONT for T2T assemblies" class="figure mx-auto d-block"><div class="figcaption">ONT for T2T assemblies</div>
</figure></section><section><h2 class="section-heading" id="installation-and-setup">Installation and Setup<a class="anchor" aria-label="anchor" href="#installation-and-setup"></a>
</h2>
<hr class="half-width">
<p>Flye is available as module on RCAC clusters. You can load the module
using the following command:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">ml</span> flye</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="ex">flye</span> <span class="at">--version</span></span></code></pre>
</div>
<p>You can also use the Singularity container for HiFiasm, which
provides a consistent environment across different systems. The
container can be pulled from the BioContainers registry using the
following command:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">apptainer</span> pull docker://quay.io/biocontainers/flye:2.9.5--py311h2de2dd3_2</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">apptainer</span> exec hflye_2.9.5--py311h2de2dd3_2.sif flye <span class="at">--version</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="overview-of-flye-assembler">Overview of Flye Assembler<a class="anchor" aria-label="anchor" href="#overview-of-flye-assembler"></a>
</h2>
<hr class="half-width">
<p>Flye is a <em>de novo</em> assembler designed for high-error,
long-read sequencing data from Oxford Nanopore Technologies (ONT) and
PacBio. It is optimized to handle the inherent noise in single-molecule
sequencing (SMS) reads while producing highly contiguous assemblies.
Flye is particularly well-suited for assembling complex genomes,
resolving repetitive regions, and reconstructing structural variations
that short-read assemblers struggle with.</p>
<p>The Flye assembly workflow typically involves:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Read preprocessing</strong> – filtering and quality-checking
raw ONT reads<br>
</li>
<li>
<strong>Disjointig generation</strong> – constructing long,
error-prone sequences from overlapping reads<br>
</li>
<li>
<strong>Repeat graph construction</strong> – building a repeat-aware
assembly graph to represent genome structure<br>
</li>
<li>
<strong>Graph resolution</strong> – disentangling repeats and
structural variations to produce accurate contigs<br>
</li>
<li>
<strong>Polishing</strong> – refining assemblies to improve
base-level accuracy using read alignment<br>
</li>
<li>
<strong>Post-processing</strong> – assessing assembly quality and
generating final output</li>
</ol>
<p>Flye is optimized for this process, leveraging repeat graph-based
assembly to generate longer, more contiguous sequences than many
traditional long-read assemblers. Its ability to handle highly
repetitive regions, coupled with its fast runtime and efficient memory
usage, makes it a powerful choice for ONT genome assembly.</p>
<figure><img src="https://github.com/user-attachments/assets/2fef9e07-675d-4ee4-8b11-73e11afe9673" alt="Flye assembly overview" class="figure mx-auto d-block"><div class="figcaption">Flye assembly overview</div>
</figure></section><section><h2 class="section-heading" id="flye-basic-workflow">Flye: basic workflow<a class="anchor" aria-label="anchor" href="#flye-basic-workflow"></a>
</h2>
<hr class="half-width">
<p>To run <code>flye</code>, you need to provide the input long reads in
FASTA or FASTQ format, specifying the long read type, provide estimated
genome size, output directory and the threads to use. The basic command
structure is as follows:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="ex">ml</span> flye</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">flye</span> <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="at">--nano-raw</span> 9994.q20.CCS-filtered-60x.fastq <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="at">--genome-size</span> 135m <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  <span class="at">--out-dir</span> flye_ont <span class="dt">\</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>  <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span></code></pre>
</div>
<div id="options-used" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="options-used" class="callout-inner">
<h3 class="callout-title">Options used</h3>
<div class="callout-content">
<ul>
<li>
<code>--nano-raw</code> specifies the input ONT long reads in FASTQ
format</li>
<li>
<code>--genome-size</code> provides an estimate of the genome size
to guide assembly</li>
<li>
<code>--out-dir</code> specifies the output directory for Flye
results</li>
<li>
<code>--threads</code> specifies the number of CPU threads to use
for assembly</li>
</ul>
<p>The input can either be fastq or fasta, compressed or uncompressed.
The output will be stored in the directory provided.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="understanding-flye-output">Understanding Flye Output<a class="anchor" aria-label="anchor" href="#understanding-flye-output"></a>
</h2>
<hr class="half-width">
<p>The output of Flye includes several files and directories that
provide information about the assembly process and results. Key
components of the Flye output include:</p>
<table class="table">
<colgroup>
<col width="7%">
<col width="92%">
</colgroup>
<thead><tr class="header">
<th align="left"><strong>File/Folder</strong></th>
<th><strong>Description</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><strong>00-assembly/</strong></td>
<td>Initial draft assembly output.</td>
</tr>
<tr class="even">
<td align="left"><strong>10-consensus/</strong></td>
<td>Consensus refinement step.</td>
</tr>
<tr class="odd">
<td align="left"><strong>20-repeat/</strong></td>
<td>Repeat graph construction and analysis.</td>
</tr>
<tr class="even">
<td align="left"><strong>30-contigger/</strong></td>
<td>Final contig generation step.</td>
</tr>
<tr class="odd">
<td align="left"><strong>40-polishing/</strong></td>
<td>Final polishing step for improving assembly quality.</td>
</tr>
<tr class="even">
<td align="left">assembly.fasta</td>
<td><em>Final polished assembly sequence.</em></td>
</tr>
<tr class="odd">
<td align="left">assembly_graph.gfa</td>
<td>Final assembly graph in GFA format.</td>
</tr>
<tr class="even">
<td align="left">assembly_graph.gv</td>
<td>Visualization of final assembly graph.</td>
</tr>
<tr class="odd">
<td align="left">assembly_info.txt</td>
<td>Summary information about the assembly.</td>
</tr>
<tr class="even">
<td align="left">flye.log</td>
<td>Log file detailing the Flye run.</td>
</tr>
</tbody>
</table>
<div id="which-file-should-i-use-as-my-final-assembly" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="which-file-should-i-use-as-my-final-assembly" class="callout-inner">
<h3 class="callout-title">Which file should I use as my final assembly?</h3>
<div class="callout-content">
<p>The <code>assembly.fasta</code> file contains the final polished
assembly sequence and is typically used as the primary output for
downstream analyses. This file represents the best estimate of the
assembled genome based on the input data and the Flye assembly process.
You can use this file for further analyses, such as gene prediction,
variant calling, or comparative genomics studies.</p>
</div>
</div>
</div>
<p>Quick look at metrics for this assembly:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="ex">ml</span> quast</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="ex">quast.py</span> <span class="dt">\</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>    <span class="at">--fast</span> <span class="dt">\</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>    <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>    <span class="at">-o</span> quast_basic_stats <span class="dt">\</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>    assembly.fasta</span></code></pre>
</div>
<div id="which-of-these-assemblies-look-better" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="which-of-these-assemblies-look-better" class="callout-inner">
<h3 class="callout-title">Which of these assemblies look better?</h3>
<div class="callout-content">
<p>Check the <code>quast_basic_stats/report.txt</code> file to check
assembly statistics. Based on your previous assembly using
<code>hifiasm</code>, what assembly do you think is better? What metrics
are you using to make this decision? Discuss which assembly has better
contiguity and completeness based on these statistics.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="other-important-parameters">Other important parameters<a class="anchor" aria-label="anchor" href="#other-important-parameters"></a>
</h2>
<hr class="half-width">
<p><code>flye</code> provides several <a href="https://github.com/mikolmogorov/Flye/blob/flye/docs/USAGE.md" class="external-link">additional
parameters</a> that can be used to customize the assembly process and
improve results. Some key parameters include:</p>
<ul>
<li>Pick the right input type (<code>--nano-hq</code>,
<code>--pacbio-hifi</code>, etc.) → Incorrect selection affects
accuracy.</li>
<li>Always specify <code>--out-dir</code> and <code>--threads</code> for
faster and organized runs.</li>
<li>Use <code>--keep-haplotypes</code> if you don’t want a collapsed
assembly.</li>
<li>For metagenomes, use <code>--meta</code> to handle variable
coverage.</li>
<li>If the assembly fails, use <code>--resume</code> to avoid losing
progress.</li>
</ul>
<div id="interested-in-exploring-more-about-flye" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="interested-in-exploring-more-about-flye" class="callout-inner">
<h3 class="callout-title">Interested in exploring more about Flye?</h3>
<div class="callout-content">
<p>Check out the <a href="https://github.com/mikolmogorov/Flye/blob/flye/docs/FAQ.md" class="external-link">Flye
FAQ</a> for answers to common questions and troubleshooting tips. You
can also explore the <a href="https://github.com/mikolmogorov/Flye/tree/flye" class="external-link">Flye GitHub
repository</a> for the latest updates, documentation, and discussions
about the assembler.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="improving-assembly-quality-with-polishing-optional">Improving Assembly Quality with Polishing (Optional)<a class="anchor" aria-label="anchor" href="#improving-assembly-quality-with-polishing-optional"></a>
</h2>
<hr class="half-width">
<p>After generating the initial assembly, it is often beneficial to
polish the assembly to improve base-level accuracy. Polishing involves
aligning the raw reads back to the assembly and correcting errors to
produce a more accurate consensus sequence. This step can significantly
enhance the quality of the assembly, especially for error-prone
long-read data like ONT reads.</p>
<p>Flye provides built-in polishing capabilities. By default, Flye
performs one round of polishing to refine the assembly. However, you can
customize the polishing process by running polishing separately after
the initial assembly.</p>
<p>An example command to polish assembly with accurate PacBio HiFi
reads:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="ex">ml</span> flye</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="ex">flye</span> <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  <span class="at">--polish-target</span> SEQ_TO_POLISH <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>  <span class="at">--pacbio-raw</span> READS <span class="dt">\ </span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>  <span class="ex">--iterations</span> NUM_ITER <span class="dt">\</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>  <span class="at">--out-dir</span> OUTPUTDIR <span class="dt">\</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>  <span class="at">--threads</span> THREADS</span></code></pre>
</div>
<p>*You can also provide Bam file as input instead of reads</p>
<p>There are many other polishing tools available, such as
<code>Racon</code>, <code>Nanoploish</code>, and <code>medaka</code>,
which can be used to further refine the ONT assembly. Each tool has its
strengths and limitations, so it is recommended to try different
polishing strategies to achieve the best results for your specific
dataset. <code>Medaka</code> is a popular choice for polishing ONT
assemblies due to its accuracy and efficiency.</p>
</section><section><h2 class="section-heading" id="hifiasm-for-ont-data-optional">HiFiasm for ONT Data (Optional)<a class="anchor" aria-label="anchor" href="#hifiasm-for-ont-data-optional"></a>
</h2>
<hr class="half-width">
<p>Since HiFiasm also supports ONT reads for assembly, we can test it
out to access the quality of the assembly. The basic command structure
is as follows:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="ex">ml</span> hifiasm</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="ex">hifiasm</span> <span class="dt">\</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>    <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>    <span class="at">-o</span> athaliana_ont.asm <span class="dt">\</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>    <span class="at">--ont</span> <span class="dt">\</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>    basecalled_2025-02-12-filtered_60x.fastq</span></code></pre>
</div>
<div id="post-processing" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="post-processing" class="callout-inner">
<h3 class="callout-title">Post processing</h3>
<div class="callout-content">
<p>Once the run completes (~30 mins with 32 threads), you can convert
GFA to FASTA using the following command:</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="cf">for</span> ctg <span class="kw">in</span> <span class="pp">*</span>_ctg.gfa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>    <span class="fu">awk</span> <span class="st">'/^S/{print "&gt;"$2"\n"$3}'</span> <span class="va">${ctg}</span> <span class="op">&gt;</span> <span class="va">${ctg</span><span class="op">%</span>.gfa<span class="va">}</span>.fasta</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
<p>Get teh basic stats using <code>quast</code>:</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="ex">ml</span> quast</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="ex">quast.py</span> <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>    <span class="at">--fast</span> <span class="dt">\</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>    <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>    <span class="at">-o</span> quast_basic_stats <span class="dt">\</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>    <span class="pp">*</span>_ctg.fasta</span></code></pre>
</div>
<p>and run <code>compleasm</code> to get the assembly completeness:</p>
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="ex">ml</span> completeasm</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="cf">for</span> fasta <span class="kw">in</span> <span class="pp">*</span>_ctg.fasta<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>    <span class="ex">compleasm</span> run <span class="dt">\</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>       <span class="at">-a</span> <span class="va">${fasta}</span> <span class="dt">\</span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>       <span class="at">-o</span> <span class="va">${fasta</span><span class="op">%</span>.<span class="pp">*</span><span class="va">}</span> <span class="dt">\</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>       <span class="at">-l</span> brassicales_odb10 <span class="dt">\</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>       <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>ONT provides long-read sequencing data with high error rates.</li>
<li>Flye is a long-read assembler optimized for handling ONT data and
producing highly contiguous assemblies.</li>
<li>The Flye assembly workflow involves read preprocessing, repeat graph
construction, graph resolution, polishing, and post-processing.</li>
<li>Flye output includes the final assembly sequence, assembly graph,
and summary information for evaluation.</li>
<li>Polishing the assembly can improve base-level accuracy and overall
assembly quality.</li>
<li>Flye provides built-in polishing capabilities, and other tools like
Racon, Nanopolish, and Medaka can be used for further refinement.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-6-hybrid-assembly"><p>Content from <a href="6-hybrid-assembly.html">Hybrid Long Read Assembly (optional)</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/6-hybrid-assembly.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is hybrid assembly, and how does it combine different
sequencing technologies?</li>
<li>How can you perform hybrid assembly using both types of long-read
data?</li>
<li>What are the key steps in hybrid assembly, including polishing and
scaffolding?</li>
<li>How do you evaluate the quality of a hybrid assembly using
bioinformatics tools?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the concept of hybrid assembly and its advantages in
genome sequencing.</li>
<li>Learn how to perform hybrid assembly using long-read sequencing data
from PacBio and ONT platforms.</li>
<li>Explore the key steps involved in hybrid assembly, including
polishing and scaffolding.</li>
<li>Evaluate the quality of a hybrid assembly using bioinformatics tools
such as QUAST and Compleasm.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="flye-for-hybrid-assembly">Flye for Hybrid Assembly<a class="anchor" aria-label="anchor" href="#flye-for-hybrid-assembly"></a>
</h2>
<hr class="half-width">
<p>For hybrid assembly using <code>flye</code>, first, run the pipeline
with all your reads in the –pacbio-raw mode (you can specify multiple
files, no need to merge all you reads into one). Also add –iterations 0
to stop the pipeline before polishing. Once the assembly finishes, run
polishing using either PacBio or ONT reads only. Use the same assembly
options, but add –resume-from polishing. Here is an example of a script
that should do the job:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">ml</span> flye</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="va">PBREADS</span><span class="op">=</span><span class="st">"9994.q20.CCS-filtered-60x.fastq"</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="va">ONTREADS</span><span class="op">=</span><span class="st">"basecalled_2025-02-12-filtered_60x.fastq"</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="ex">flye</span> <span class="dt">\</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>    <span class="at">--pacbio-raw</span> <span class="va">$PBREADS</span> <span class="va">$ONTREADS</span> <span class="dt">\</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>    <span class="at">--iterations</span> 0 <span class="dt">\</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>    <span class="at">--out-dir</span> hybrid_flye_out <span class="dt">\</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>    <span class="at">--genome-size</span> 135m <span class="dt">\</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>    <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="ex">flye</span> <span class="dt">\</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>   <span class="at">--pacbio-raw</span> <span class="va">$PBREADS</span> <span class="dt">\</span></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>   <span class="at">--resume-from</span> polishing <span class="dt">\</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a>   <span class="at">--out-dir</span> hybrid_flye_out  <span class="dt">\</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a>   <span class="at">--genome-size</span> 135m <span class="dt">\</span></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="evaluating-assembly-quality">Evaluating Assembly Quality<a class="anchor" aria-label="anchor" href="#evaluating-assembly-quality"></a>
</h2>
<hr class="half-width">
<p>For quick evaluation, we will run <code>quast</code> and
<code>compleasm</code> on the hybrid assembly output.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="ex">ml</span> quast</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="ex">ml</span> compleasm</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="va">fasta</span><span class="op">=</span><span class="st">"hybrid_flye_out/assembly.fasta"</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="ex">quast.py</span> <span class="dt">\</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>  <span class="at">--fast</span> <span class="dt">\</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>  <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>  <span class="at">-o</span> quast_basic <span class="dt">\</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>    <span class="va">${fasta}</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="ex">compleasm</span> <span class="dt">\</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>compleasm run <span class="dt">\</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>   <span class="at">-a</span> <span class="va">${fasta}</span> <span class="dt">\</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>   <span class="at">-o</span> compleasm_out <span class="dt">\</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>   <span class="at">-l</span> brassicales_odb10  <span class="dt">\</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>   <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="scaffolding-with-bionano">Scaffolding with Bionano<a class="anchor" aria-label="anchor" href="#scaffolding-with-bionano"></a>
</h2>
<hr class="half-width">
<p>To scaffold the assembly using Bionano data, we will use the
<code>bionano solve</code>. We can run the following script to scaffold
the assembly:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span><span class="va">$PATH</span>:/apps/biocontainers/exported-wrappers/bionano/3.8.0</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="va">fasta</span><span class="op">=</span><span class="st">"hybrid_flye_out/assembly.fasta"</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">run_hybridscaffold.sh</span> <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="at">-c</span> /opt/Solve3.7_10192021_74_1/HybridScaffold/1.0/hybridScaffold_DLE1_config.xml<span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="at">-b</span> workshop_assembly/col-0_bionano/Evry.OpticalMap.Col-0.cmap <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  <span class="at">-n</span> <span class="va">${fasta}</span> <span class="dt">\</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>  <span class="at">-u</span> CTTAAG <span class="dt">\</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>  <span class="at">-z</span> results_bionano_hybrid_scaffolding.zip <span class="dt">\</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>  <span class="at">-w</span> log.txt <span class="dt">\</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>  <span class="at">-B</span> 2 <span class="dt">\</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>  <span class="at">-N</span> 2 <span class="dt">\</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>  <span class="at">-g</span> <span class="dt">\</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>  <span class="at">-f</span> <span class="dt">\</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>  <span class="at">-r</span> /opt/Solve3.7_10192021_74_1/RefAligner/1.0/sse/RefAligner <span class="dt">\</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a>  <span class="at">-p</span> /opt/Solve3.7_10192021_74_1/Pipeline/1.0 <span class="dt">\</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a>  <span class="at">-o</span> bionano_hybrid_scaffolding</span></code></pre>
</div>
<p>Once this completes, you can generate the final scaffold-level
assembly by merging placed and unplaced contigs in the
<code>bionano_hybrid_scaffolding/hybrid_scaffolds</code> directory.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="bu">cd</span> bionano_hybrid_scaffolding/hybrid_scaffolds</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="fu">cat</span> <span class="pp">*</span>HYBRID_SCAFFOLD.fasta <span class="pp">*</span>_HYBRID_SCAFFOLD_NOT_SCAFFOLDED.fasta <span class="dt">\</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>   <span class="op">&gt;</span> ../../assembly_scaffolds.fasta</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="bu">cd</span> ../..</span></code></pre>
</div>
<p>You can evaluate the final assembly using <code>quast</code> and
<code>compleasm</code> as before.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="va">fasta</span><span class="op">=</span><span class="st">"assembly_scaffolds.fasta"</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">quast.py</span> <span class="dt">\</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="at">--fast</span> <span class="dt">\</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  <span class="at">-o</span> quast_scaffolds <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>    <span class="va">${fasta}</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="ex">compleasm</span> run <span class="dt">\</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>   <span class="at">-a</span> <span class="va">${fasta}</span> <span class="dt">\</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>   <span class="at">-o</span> compleasm_scaffolds_out <span class="dt">\</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>   <span class="at">-l</span> brassicales_odb10  <span class="dt">\</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>   <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="hybrid-assembly-summary">Hybrid Assembly Summary<a class="anchor" aria-label="anchor" href="#hybrid-assembly-summary"></a>
</h2>
<hr class="half-width">
<p>In this section you have learned how to perform a hybrid assembly
using Flye, polish the assembly, scaffold it using Bionano, and evaluate
the final assembly quality. This workflow combines the advantages of ONT
and PacBio sequencing, improves structural accuracy with Bionano
scaffolding, and ensures a high-quality genome assembly. The steps
involved are:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Run Hybrid Assembly with Flye</strong>
<ul>
<li>Use <strong>Flye</strong> in <code>--pacbio-raw</code> mode to
assemble both PacBio and ONT reads.<br>
</li>
<li>Set <code>--iterations 0</code> to stop before polishing.</li>
</ul>
</li>
<li>
<strong>Polishing the Assembly</strong>
<ul>
<li>Polish the assembly using <strong>either PacBio or ONT
reads</strong> by resuming Flye with
<code>--resume-from polishing</code>.</li>
</ul>
</li>
<li>
<strong>Evaluate Assembly Quality</strong>
<ul>
<li>Run <strong>QUAST</strong> for basic assembly metrics (contig count,
N50, genome size, misassemblies).<br>
</li>
<li>Use <strong>Compleasm</strong> to assess genome completeness based
on conserved single-copy genes.</li>
</ul>
</li>
<li>
<strong>Scaffold Assembly with Bionano Optical Mapping</strong>
<ul>
<li>Use <strong>Bionano Solve</strong> to integrate optical maps and
scaffold the assembly.<br>
</li>
<li>Run <code>run_hybridscaffold.sh</code> with the reference
<code>.cmap</code> optical map file.</li>
</ul>
</li>
<li>
<strong>Generate Final Scaffold-Level Assembly</strong>
<ul>
<li>Merge placed and unplaced contigs from the Bionano scaffolding
output to create the final genome assembly.</li>
</ul>
</li>
<li>
<strong>Final Evaluation of Scaffolds</strong>
<ul>
<li>Re-run <strong>QUAST and Compleasm</strong> to validate improvements
and ensure genome completeness after scaffolding.</li>
</ul>
</li>
</ol>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Hybrid assembly with Flye combines ONT and PacBio reads to leverage
long-read continuity and high-accuracy sequencing, with separate
polishing steps to refine base-level errors.<br>
</li>
<li>Assembly quality assessment using QUAST and Compleasm provides
critical insights into contiguity, completeness, and potential
misassemblies, ensuring reliability before scaffolding.<br>
</li>
<li>Bionano Optical Genome Mapping (OGM) improves hybrid assemblies by
scaffolding contigs, resolving misassemblies, and enhancing genome
continuity, leading to chromosome-scale assemblies.</li>
<li>Final scaffolding validation and quality assessment ensure the
integrity of the genome assembly, with QUAST and Compleasm used to
confirm improvements after Bionano integration.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-7-scaffolding-assemblies"><p>Content from <a href="7-scaffolding-assemblies.html">Scaffolding using Optical Genome Mapping</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/7-scaffolding-assemblies.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is Bionano optical genome mapping (OGM) and how does it improve
genome assembly?</li>
<li>How does Bionano Solve hybrid scaffolding integrate optical maps
with sequence assemblies?</li>
<li>What are the key steps involved in running the Bionano Solve
pipeline for hybrid scaffolding?</li>
<li>How can you assess the quality of hybrid scaffolds generated by
Bionano Solve?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the principles of Bionano optical genome mapping (OGM)
and its role in genome assembly.</li>
<li>Learn how to run the Bionano Solve hybrid scaffolding pipeline to
improve genome assemblies.</li>
<li>Explore the key steps involved in scaffolding HiFiasm and Flye
assemblies using Bionano Solve.</li>
<li>Evaluate the quality of hybrid scaffolds generated by Bionano Solve
and interpret the results.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction-to-bionano-optical-genome-mapping-ogm">Introduction to Bionano optical genome mapping (OGM)<a class="anchor" aria-label="anchor" href="#introduction-to-bionano-optical-genome-mapping-ogm"></a>
</h2>
<hr class="half-width">
<p>Bionano optical mapping is a high-resolution genome analysis
technique that generates long-range structural information by labeling
and imaging ultra-long DNA molecules. It provides genome-wide maps that
can be used to scaffold contigs from sequencing-based assemblies,
significantly improving contiguity and structural accuracy. By
integrating Bionano maps with assemblies from PacBio HiFi and Oxford
Nanopore Technologies (ONT), misassemblies can be corrected, chimeric
contigs resolved, and scaffold N50s increased by orders of magnitude.
This approach is particularly valuable for complex genomes, where
repetitive sequences and structural variations pose challenges for
traditional sequencing methods. Bionano hybrid scaffolding has become a
standard for enhancing genome assemblies, enabling researchers to
achieve high-quality, chromosome-level assemblies efficiently.</p>
<figure><img src="https://github.com/user-attachments/assets/41c7c42b-c5e0-4d0d-b899-a479e62ffaee" alt="Overview of Bionano OGM" class="figure mx-auto d-block"><div class="figcaption">Overview of Bionano OGM</div>
</figure></section><section><h2 class="section-heading" id="bionano-solve-hybrid-scaffolding">Bionano Solve Hybrid Scaffolding<a class="anchor" aria-label="anchor" href="#bionano-solve-hybrid-scaffolding"></a>
</h2>
<hr class="half-width">
<p>Bionano Solve improves genome assembly by integrating optical genome
mapping data with sequence assemblies, generating ultra-long hybrid
scaffolds that enhance contiguity and accuracy. The pipeline identifies
and resolves assembly conflicts, orders and orients sequence contigs,
and estimates gap sizes between adjacent sequences.</p>
<p>The scaffolding workflow involves:</p>
<ol style="list-style-type: decimal">
<li>In Silico Map Generation – Converting sequence assembly into a map
format for alignment.</li>
<li>Conflict Resolution – Aligning in silico maps to Bionano genome maps
and identifying misassemblies.</li>
<li>Hybrid Scaffolding – Merging high-confidence sequence and optical
maps into a refined scaffold.</li>
<li>Final Alignment – Mapping sequence contigs back to the hybrid
scaffold for consistency validation.</li>
<li>Output Generation – Producing final AGP and FASTA files with
corrected genome structures.</li>
</ol>
<figure><img src="https://github.com/user-attachments/assets/e568386e-348d-4b18-82ca-e304369f0fed" alt="Scaffolding using Bionano" class="figure mx-auto d-block"><div class="figcaption">Scaffolding using Bionano</div>
</figure><p><strong>What is the source of this data?</strong></p>
<ul>
<li>The optical genome mapping data was obtained from the project <a href="https://www.ebi.ac.uk/ena/browser/view/PRJEB50694" class="external-link">PRJEB50694</a>.<br>
</li>
<li>Data corresponds to <em>Arabidopsis thaliana</em> ecotype Col-0,
generated using Bionano optical genome mapping technology.<br>
</li>
<li>The dataset is publicly available on the European Nucleotide Archive
(ENA) and can be downloaded using:<br>
</li>
<li>The <code>.cmap</code> file contains high-resolution optical maps
used for scaffolding and structural validation of genome
assemblies.</li>
</ul>
<p>To download:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">wget</span> ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR522/ERR5229977/Col-0_2021-09-30.cmap</span></code></pre>
</div>
</section><section><h2 class="section-heading" id="installation-and-setup">Installation and Setup<a class="anchor" aria-label="anchor" href="#installation-and-setup"></a>
</h2>
<hr class="half-width">
<p>Bionano Solve is available on Bionano.com and can be installed on
Linux-based systems. The software requires a valid license and access to
Bionano data files for processing.</p>
<p>Custom container with just the hybrid scaffolding tools can be used
to run the Bionano Solve pipeline. On Negishi, you can add it to your
PATH using the command below:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span><span class="va">$PATH</span>:/apps/biocontainers/exported-wrappers/bionano/3.8.0</span></code></pre>
</div>
</section><section><h2 class="section-heading" id="running-bionano-solve">Running Bionano Solve<a class="anchor" aria-label="anchor" href="#running-bionano-solve"></a>
</h2>
<hr class="half-width">
<p>To scaffold a genome using Bionano Solve, you need to provide the
following input files:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span><span class="va">$PATH</span>:/apps/biocontainers/exported-wrappers/bionano/3.8.0</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">run_hybridscaffold.sh</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  <span class="ex">-c</span> /opt/Solve3.7_10192021_74_1/HybridScaffold/1.0/hybridScaffold_DLE1_config.xml<span class="dt">\</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  <span class="at">-b</span> input.cmap <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="at">-n</span> genome.fasta <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="at">-u</span> CTTAAG <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  <span class="at">-z</span> results_output.zip <span class="dt">\</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>  <span class="at">-w</span> log.txt <span class="dt">\</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>  <span class="at">-B</span> 2 <span class="dt">\</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>  <span class="at">-N</span> 2 <span class="dt">\</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>  <span class="at">-g</span> <span class="dt">\</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>  <span class="at">-f</span> <span class="dt">\</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>  <span class="at">-r</span> /opt/Solve3.7_10192021_74_1/RefAligner/1.0/sse/RefAligner <span class="dt">\</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>  <span class="at">-p</span> /opt/Solve3.7_10192021_74_1/Pipeline/1.0 <span class="dt">\</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>  <span class="at">-o</span> output_dir</span></code></pre>
</div>
<div id="options-used" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="options-used" class="callout-inner">
<h3 class="callout-title">Options used</h3>
<div class="callout-content">
<table class="table">
<colgroup>
<col width="12%">
<col width="20%">
<col width="66%">
</colgroup>
<thead><tr class="header">
<th align="center"><strong>Option</strong></th>
<th align="left"><strong>Argument</strong></th>
<th><strong>Description</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center"><code>-c</code></td>
<td align="left"><code>/opt/Solve3.7_10192021_74_1/HybridScaffold/1.0/hybridScaffold_DLE1_config.xml</code></td>
<td>Specifies the hybrid scaffolding configuration file required for the
pipeline.</td>
</tr>
<tr class="even">
<td align="center"><code>-b</code></td>
<td align="left"><code>input.cmap</code></td>
<td>Input Bionano CMAP file, which contains the optical genome map
data.</td>
</tr>
<tr class="odd">
<td align="center"><code>-n</code></td>
<td align="left"><code>genome.fasta</code></td>
<td>Input genome sequence in FASTA format from NGS assembly.</td>
</tr>
<tr class="even">
<td align="center"><code>-u</code></td>
<td align="left"><code>CTTAAG</code></td>
<td>Specifies the sequence of the enzyme recognition site, overriding
the one in the config XML file.</td>
</tr>
<tr class="odd">
<td align="center"><code>-z</code></td>
<td align="left"><code>results_output.zip</code></td>
<td>Generates a ZIP archive containing essential output files.</td>
</tr>
<tr class="even">
<td align="center"><code>-w</code></td>
<td align="left"><code>log.txt</code></td>
<td>Defines the name of the status text file needed for IrysView.</td>
</tr>
<tr class="odd">
<td align="center"><code>-B</code></td>
<td align="left"><code>2</code></td>
<td>Conflict filter level: <code>2</code> means cut the contig at
conflict points (required if not using <code>-M</code>).</td>
</tr>
<tr class="even">
<td align="center"><code>-N</code></td>
<td align="left"><code>2</code></td>
<td>Conflict filter level: <code>2</code> means cut the contig at
conflict points (same as <code>-B</code>, applied to sequencing
contigs).</td>
</tr>
<tr class="odd">
<td align="center"><code>-g</code></td>
<td align="left">(No argument)</td>
<td>Enables trimming of overlapping NGS sequences during AGP and FASTA
export.</td>
</tr>
<tr class="even">
<td align="center"><code>-f</code></td>
<td align="left">(No argument)</td>
<td>Forces output generation and overwrites any existing files in the
output directory.</td>
</tr>
<tr class="odd">
<td align="center"><code>-r</code></td>
<td align="left"><code>/opt/Solve3.7_10192021_74_1/RefAligner/1.0/sse/RefAligner</code></td>
<td>Specifies the path to the RefAligner program, which is required for
scaffolding.</td>
</tr>
<tr class="even">
<td align="center"><code>-p</code></td>
<td align="left"><code>/opt/Solve3.7_10192021_74_1/Pipeline/1.0</code></td>
<td>Specifies the directory for the de novo assembly pipeline (optional,
required for <code>-x</code>).</td>
</tr>
<tr class="odd">
<td align="center"><code>-o</code></td>
<td align="left"><code>output_dir</code></td>
<td>Defines the output folder where scaffolded results will be
stored.</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="section level3">
<h3 id="scffolding-hifiasm-assembly-with-bionano-solve">Scffolding HiFiasm assembly with Bionano Solve<a class="anchor" aria-label="anchor" href="#scffolding-hifiasm-assembly-with-bionano-solve"></a>
</h3>
<p>For HiFiasm assembly, you need to provide the HiFiasm assembly FASTA
file as input to the Bionano Solve pipeline. The command structure
remains the same, with the only change being the input sequence
file.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span><span class="va">$PATH</span>:/apps/biocontainers/exported-wrappers/bionano/3.8.0</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">run_hybridscaffold.sh</span> <span class="dt">\</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="at">-c</span> /opt/Solve3.7_10192021_74_1/HybridScaffold/1.0/hybridScaffold_DLE1_config.xml<span class="dt">\</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="at">-b</span> Evry.OpticalMap.Col-0.cmap <span class="dt">\</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>  <span class="at">-n</span> hifiasm_60x/athaliana_hifi.asm.bp.p_ctg.fasta <span class="dt">\</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>  <span class="at">-u</span> CTTAAG <span class="dt">\</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>  <span class="at">-z</span> results_bionano_hifiasm_scaffolding.zip <span class="dt">\</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>  <span class="at">-w</span> log.txt <span class="dt">\</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>  <span class="at">-B</span> 2 <span class="dt">\</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>  <span class="at">-N</span> 2 <span class="dt">\</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  <span class="at">-g</span> <span class="dt">\</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>  <span class="at">-f</span> <span class="dt">\</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>  <span class="at">-r</span> /opt/Solve3.7_10192021_74_1/RefAligner/1.0/sse/RefAligner <span class="dt">\</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>  <span class="at">-p</span> /opt/Solve3.7_10192021_74_1/Pipeline/1.0 <span class="dt">\</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>  <span class="at">-o</span> bionano_hifiasm_scaffolding</span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="scffolding-flye-assembly-with-bionano-solve">Scffolding Flye assembly with Bionano Solve<a class="anchor" aria-label="anchor" href="#scffolding-flye-assembly-with-bionano-solve"></a>
</h3>
<p>For Flye assembly, the process is similar to HiFiasm, but you need to
provide the Flye assembly FASTA file instead of the HiFiasm assembly.
The command structure remains the same, with the only change being the
input sequence file.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">PATH</span><span class="op">=</span><span class="va">$PATH</span>:/apps/biocontainers/exported-wrappers/bionano/3.8.0</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">run_hybridscaffold.sh</span> <span class="dt">\</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="at">-c</span> /opt/Solve3.7_10192021_74_1/HybridScaffold/1.0/hybridScaffold_DLE1_config.xml<span class="dt">\</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="at">-b</span> workshop_assembly/col-0_bionano/Evry.OpticalMap.Col-0.cmap <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  <span class="at">-n</span> flye_ont_60x/assembly.fasta <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>  <span class="at">-u</span> CTTAAG <span class="dt">\</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>  <span class="at">-z</span> results_bionano_flye_scaffolding.zip <span class="dt">\</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>  <span class="at">-w</span> log.txt <span class="dt">\</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>  <span class="at">-B</span> 2 <span class="dt">\</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>  <span class="at">-N</span> 2 <span class="dt">\</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>  <span class="at">-g</span> <span class="dt">\</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>  <span class="at">-f</span> <span class="dt">\</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>  <span class="at">-r</span> /opt/Solve3.7_10192021_74_1/RefAligner/1.0/sse/RefAligner <span class="dt">\</span></span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a>  <span class="at">-p</span> /opt/Solve3.7_10192021_74_1/Pipeline/1.0 <span class="dt">\</span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a>  <span class="at">-o</span> bionano_flye_scaffolding</span></code></pre>
</div>
</div>
</section><section><h2 class="section-heading" id="understanding-hybrid-scaffolding-output">Understanding Hybrid Scaffolding Output<a class="anchor" aria-label="anchor" href="#understanding-hybrid-scaffolding-output"></a>
</h2>
<hr class="half-width">
<p>The output of the Bionano Solve pipeline includes scaffolded genome
assemblies in AGP and FASTA formats, along with alignment and conflict
resolution information. The hybrid scaffolds provide a more accurate
representation of the genome structure, with improved contiguity and
reduced misassemblies. The output files can be visualized using genome
browsers or alignment viewers to assess the quality and completeness of
the assembly.</p>
<table class="table">
<colgroup>
<col width="48%">
<col width="52%">
</colgroup>
<thead><tr class="header">
<th><strong>Folder</strong></th>
<th><strong>Contents</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>agp_fasta/</strong></td>
<td>Final scaffolded genome assembly in FASTA, AGP format, alignment
results, gap information, and logs.</td>
</tr>
<tr class="even">
<td><strong>align0/</strong></td>
<td>Initial alignment of optical maps to the sequence assembly,
including XMAP, CMAP, and error logs.</td>
</tr>
<tr class="odd">
<td><strong>align1/</strong></td>
<td>Secondary alignment refinement, similar to align0 but after
resolving initial inconsistencies.</td>
</tr>
<tr class="even">
<td><strong>align_final/</strong></td>
<td>Final alignment results of hybrid scaffolds to optical maps,
including mapping rates and statistics.</td>
</tr>
<tr class="odd">
<td><strong>assignAlignType/</strong></td>
<td>Tracks conflicts between NGS contigs and optical maps, includes
exclusion and trimming decisions.</td>
</tr>
<tr class="even">
<td><strong>cut_conflicts/</strong></td>
<td>Stores files related to contig trimming and conflict resolution
between NGS and optical maps.</td>
</tr>
<tr class="odd">
<td><strong>fa2cmap/</strong></td>
<td>Converts the sequence assembly into Bionano’s CMAP format before
integration with optical maps.</td>
</tr>
<tr class="even">
<td><strong>hybrid_scaffolds/</strong></td>
<td>Contains final scaffolded genome with CMAP, XMAP, AGP files, and a
scaffolding report.</td>
</tr>
<tr class="odd">
<td><strong>mergeNGS_BN/</strong></td>
<td>Stores intermediate files merging NGS contigs with Bionano maps,
including hybrid scaffold progress.</td>
</tr>
<tr class="even">
<td><strong>results_output.zip</strong></td>
<td>Compressed archive containing essential scaffolding results for easy
sharing.</td>
</tr>
</tbody>
</table>
<p>Within <code>hybrid_scaffolds</code>, the files ending with
<code>HYBRID_SCAFFOLD.fasta</code> and
<code>HYBRID_SCAFFOLD_NOT_SCAFFOLDED.fasta</code> represent the final
scaffolded genome and unplaced contigs, respectively. You will need to
merge these files to obtain your final scaffolded genome assembly.</p>
</section><section><h2 class="section-heading" id="quality-assessment-of-hybrid-scaffolds">Quality Assessment of Hybrid Scaffolds<a class="anchor" aria-label="anchor" href="#quality-assessment-of-hybrid-scaffolds"></a>
</h2>
<hr class="half-width">
<p>The hybird scaffolds report file will be in the
<code>hybrid_scaffolds</code> directory and will provide a summary of
the scaffolding process, including alignment statistics, conflict
resolution, and scaffold N50 values. This report is essential for
evaluating the quality and completeness of the hybrid scaffolds and
identifying any potential issues that need further investigation.</p>
<table class="table">
<colgroup>
<col width="61%">
<col width="19%">
<col width="19%">
</colgroup>
<thead><tr class="header">
<th align="left"><strong>Category</strong></th>
<th align="right"><strong>PacBio HiFi (hifiasm)</strong></th>
<th align="right"><strong>ONT (Flye)</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><strong>Original BioNano Genome Map</strong></td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="even">
<td align="left">Count</td>
<td align="right">18</td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td align="left">Min length (Mbp)</td>
<td align="right">0.342</td>
<td align="right">0.342</td>
</tr>
<tr class="even">
<td align="left">Median length (Mbp)</td>
<td align="right">3.956</td>
<td align="right">3.956</td>
</tr>
<tr class="odd">
<td align="left">Mean length (Mbp)</td>
<td align="right">7.396</td>
<td align="right">7.396</td>
</tr>
<tr class="even">
<td align="left">N50 length (Mbp)</td>
<td align="right">15.529</td>
<td align="right">15.529</td>
</tr>
<tr class="odd">
<td align="left">Max length (Mbp)</td>
<td align="right">17.518</td>
<td align="right">17.518</td>
</tr>
<tr class="even">
<td align="left">Total length (Mbp)</td>
<td align="right">133.124</td>
<td align="right">133.124</td>
</tr>
<tr class="odd">
<td align="left"><strong>Original NGS Sequences</strong></td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="even">
<td align="left">Count</td>
<td align="right">152</td>
<td align="right">43</td>
</tr>
<tr class="odd">
<td align="left">Min length (Mbp)</td>
<td align="right">0.027</td>
<td align="right">0.008</td>
</tr>
<tr class="even">
<td align="left">Median length (Mbp)</td>
<td align="right">0.050</td>
<td align="right">0.276</td>
</tr>
<tr class="odd">
<td align="left">Mean length (Mbp)</td>
<td align="right">0.896</td>
<td align="right">2.797</td>
</tr>
<tr class="even">
<td align="left">N50 length (Mbp)</td>
<td align="right">7.981</td>
<td align="right">9.261</td>
</tr>
<tr class="odd">
<td align="left">Max length (Mbp)</td>
<td align="right">13.758</td>
<td align="right">14.609</td>
</tr>
<tr class="even">
<td align="left">Total length (Mbp)</td>
<td align="right">136.156</td>
<td align="right">120.259</td>
</tr>
<tr class="odd">
<td align="left"><strong>Conflict Resolution (BNG-NGS
Alignment)</strong></td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="even">
<td align="left">Conflict cuts made to Bionano maps</td>
<td align="right">2</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">Conflict cuts made to NGS sequences</td>
<td align="right">30</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Bionano maps to be cut</td>
<td align="right">2</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">NGS sequences to be cut</td>
<td align="right">18</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><strong>NGS FASTA Sequence in Hybrid
Scaffold</strong></td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="odd">
<td align="left">Count</td>
<td align="right">40</td>
<td align="right">26</td>
</tr>
<tr class="even">
<td align="left">Min length (Mbp)</td>
<td align="right">0.033</td>
<td align="right">0.065</td>
</tr>
<tr class="odd">
<td align="left">Median length (Mbp)</td>
<td align="right">0.945</td>
<td align="right">1.681</td>
</tr>
<tr class="even">
<td align="left">Mean length (Mbp)</td>
<td align="right">2.689</td>
<td align="right">4.558</td>
</tr>
<tr class="odd">
<td align="left">N50 length (Mbp)</td>
<td align="right">8.437</td>
<td align="right">9.261</td>
</tr>
<tr class="even">
<td align="left">Max length (Mbp)</td>
<td align="right">13.484</td>
<td align="right">14.609</td>
</tr>
<tr class="odd">
<td align="left">Total length (Mbp)</td>
<td align="right">107.558</td>
<td align="right">118.508</td>
</tr>
<tr class="even">
<td align="left"><strong>Hybrid Scaffold FASTA</strong></td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="odd">
<td align="left">Count</td>
<td align="right">11</td>
<td align="right">12</td>
</tr>
<tr class="even">
<td align="left">Min length (Mbp)</td>
<td align="right">0.104</td>
<td align="right">0.524</td>
</tr>
<tr class="odd">
<td align="left">Median length (Mbp)</td>
<td align="right">11.824</td>
<td align="right">12.426</td>
</tr>
<tr class="even">
<td align="left">Mean length (Mbp)</td>
<td align="right">10.518</td>
<td align="right">9.891</td>
</tr>
<tr class="odd">
<td align="left">N50 length (Mbp)</td>
<td align="right">14.479</td>
<td align="right">14.886</td>
</tr>
<tr class="even">
<td align="left">Max length (Mbp)</td>
<td align="right">15.227</td>
<td align="right">16.188</td>
</tr>
<tr class="odd">
<td align="left">Total length (Mbp)</td>
<td align="right">115.698</td>
<td align="right">118.689</td>
</tr>
<tr class="even">
<td align="left"><strong>Hybrid Scaffold FASTA + Not Scaffolded
NGS</strong></td>
<td align="right"></td>
<td align="right"></td>
</tr>
<tr class="odd">
<td align="left">Count</td>
<td align="right">161</td>
<td align="right">33</td>
</tr>
<tr class="even">
<td align="left">Min length (Mbp)</td>
<td align="right">0.024</td>
<td align="right">0.006</td>
</tr>
<tr class="odd">
<td align="left">Median length (Mbp)</td>
<td align="right">0.051</td>
<td align="right">0.159</td>
</tr>
<tr class="even">
<td align="left">Mean length (Mbp)</td>
<td align="right">0.896</td>
<td align="right">3.650</td>
</tr>
<tr class="odd">
<td align="left">N50 length (Mbp)</td>
<td align="right">14.083</td>
<td align="right">14.886</td>
</tr>
<tr class="even">
<td align="left">Max length (Mbp)</td>
<td align="right">15.227</td>
<td align="right">16.188</td>
</tr>
<tr class="odd">
<td align="left">Total length (Mbp)</td>
<td align="right">144.295</td>
<td align="right">120.440</td>
</tr>
</tbody>
</table>
<div id="which-assembler-and-data-performed-better" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="which-assembler-and-data-performed-better" class="callout-inner">
<h3 class="callout-title">Which assembler and data performed better?</h3>
<div class="callout-content">
<ul>
<li>The HiFiasm assembly with PacBio HiFi data resulted in a higher N50
length and total length in the hybrid scaffold compared to the Flye
assembly with ONT data.</li>
<li>The conflict resolution process involved more cuts in the NGS
sequences for the HiFiasm assembly, indicating a higher level of
alignment discrepancies.</li>
<li>The final hybrid scaffold from the HiFiasm assembly had a higher N50
length and total length, suggesting better contiguity and completeness
compared to the Flye assembly.</li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Bionano optical genome mapping (OGM) provides long-range structural
information for scaffolding genome assemblies.</li>
<li>Bionano Solve hybrid scaffolding integrates optical maps with
sequence assemblies to improve contiguity and accuracy.</li>
<li>The Bionano Solve pipeline involves in silico map generation,
conflict resolution, hybrid scaffolding, and final alignment.</li>
<li>The output of Bionano Solve includes scaffolded genome assemblies in
AGP and FASTA formats, alignment results, and conflict resolution
information.</li>
<li>Quality assessment of hybrid scaffolds involves evaluating alignment
statistics, conflict resolution, scaffold N50 values, and completeness
of the assembly.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-8-assembly-evaluation"><p>Content from <a href="8-assembly-evaluation.html">Assembly Assessment</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/8-assembly-evaluation.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>Why is evaluating genome assembly quality important?</li>
<li>What tools can be used to assess assembly completeness, accuracy,
and structural integrity?</li>
<li>How do you interpret key metrics from assembly evaluation
tools?</li>
<li>What are the main steps in evaluating a genome assembly using
bioinformatics tools?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the importance of evaluating genome assembly
quality.</li>
<li>Learn about tools for assessing assembly completeness, accuracy, and
structural integrity.</li>
<li>Interpret key metrics from assembly evaluation tools to guide
further analysis.</li>
<li>Evaluate a genome assembly using bioinformatics tools such as QUAST,
Compleasm, Merqury, and Bandage.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="evaluating-assembly-quality">Evaluating Assembly Quality<a class="anchor" aria-label="anchor" href="#evaluating-assembly-quality"></a>
</h2>
<hr class="half-width">
<p>Assessing genome assembly quality is essential to ensure
completeness, accuracy, and structural integrity before downstream
analyses. Different tools provide complementary
insights—<strong>QUAST</strong> evaluates assembly contiguity,
<strong>Compleasm</strong> assesses gene-space completeness,
<strong>Merqury</strong> validates k-mer consistency, and
<strong>Bandage</strong> visualizes assembly graphs for structural
assessment. Together, these methods help identify errors, improve genome
reconstruction, and ensure high-quality results.</p>
<p><strong>Why is Assembly Evaluation Important?</strong></p>
<ul>
<li>
<strong>Detects misassemblies and structural errors</strong>:
Identifies fragmented, misjoined, or incorrectly placed contigs that can
impact genome interpretation.<br>
</li>
<li>
<strong>Measures completeness and accuracy</strong>: Ensures that
essential genes and expected genome regions are properly assembled and
not missing or duplicated.<br>
</li>
<li>
<strong>Validates sequencing data quality</strong>: Confirms whether
sequencing errors, biases, or artifacts affect the final assembly.<br>
</li>
<li>
<strong>Guides further refinement</strong>: Helps decide whether
additional polishing, scaffolding, or reassembly is needed for better
genome reconstruction.</li>
</ul></section><section><h2 class="section-heading" id="quast-for-quality-metrics">Quast for quality metrics<a class="anchor" aria-label="anchor" href="#quast-for-quality-metrics"></a>
</h2>
<hr class="half-width">
<p>You can run <code>quast</code> to evaluate the quality of your genome
assembly. It is also useful for comparing multiple assemblies to
identify the best one based on key metrics such as contig count, N50,
and misassemblies.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">ml</span> compleasm</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> quast_evaluation</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co"># ln -s ../assembly1.fasta all_assemblies/assembly1.fasta</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="co"># ln -s ../assembly2.fasta all_assemblies/assembly2.fasta</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="co"># ln -s ../assembly3.fasta all_assemblies/assembly3.fasta</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="co"># link any other assemblies you want to compare</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="co"># ln -s ../pacbio/9994.q20.CCS-filtered-60x.fastq</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="co"># donwload the reference genome</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="fu">wget</span> https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-60/fasta/arabidopsis_thaliana/dna/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa.gz</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="fu">gunzip</span> Arabidopsis_thaliana.TAIR10.dna.toplevel.fa.gz</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="ex">quast.py</span> <span class="dt">\</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>   <span class="at">--output-dir</span> quast_complete_stats <span class="dt">\</span></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>   <span class="at">--no-read-stats</span> <span class="dt">\</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a>   <span class="at">-r</span>  Arabidopsis_thaliana.TAIR10.dna.toplevel.fa <span class="dt">\</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a>   <span class="at">--eukaryote</span> <span class="dt">\</span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a>   <span class="at">--pacbio</span> 9994.q20.CCS_ge20Kb.fasta <span class="dt">\</span></span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a>   assembly1.fasta assembly2.fasta assembly3.fasta</span></code></pre>
</div>
<p>This will generate a detailed report in the
<code>quast_complete_stats</code> directory, including key metrics for
each assembly and a summary of their quality. You can use this
information to compare different assemblies and select the best one for
downstream analysis.</p>
</section><section><h2 class="section-heading" id="compleasm-for-genome-completeness-gene-space">Compleasm for genome completeness (gene-space)<a class="anchor" aria-label="anchor" href="#compleasm-for-genome-completeness-gene-space"></a>
</h2>
<hr class="half-width">
<p>Similarly, you can use <code>compleasm</code> to assess the
completeness of your genome assembly in terms of gene-space
representation. This tool compares the assembly against a set of
conserved genes to estimate the level of completeness and identify
missing or fragmented genes.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="ex">ml</span> compleasm</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> compleasm_evaluation</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co"># ln -s ../assembly1.fasta all_assemblies/assembly1.fasta</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co"># ln -s ../assembly2.fasta all_assemblies/assembly2.fasta</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co"># ln -s ../assembly3.fasta all_assemblies/assembly3.fasta</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co"># link any other assemblies you want to compare</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co"># ln -s ../quast_evaluation/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa # reference</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="cf">for</span> fasta <span class="kw">in</span> <span class="pp">*</span>.fasta<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>  <span class="ex">compleasm</span> run <span class="dt">\</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>    <span class="at">-a</span> <span class="va">${fasta}</span> <span class="dt">\</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>    <span class="at">-o</span> <span class="va">${fasta</span><span class="op">%</span>.<span class="pp">*</span><span class="va">}</span>_out <span class="dt">\</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>    <span class="at">-l</span> brassicales_odb10  <span class="dt">\</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>    <span class="at">-t</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
<p>This will generate a detailed report for each assembly in the
directory, highlighting the completeness of conserved genes and
potential gaps in the genome reconstruction. The assessment result by
compleasm is saved in the file <code>summary.txt</code> in the
<code>compleasm_evaluation/assemblyN_out</code> (specified in output
<code>-o</code> option) folder. These BUSCO genes are categorized into
the following classes:</p>
<ul>
<li>
<code>S</code> (Single Copy Complete Genes): The BUSCO genes that
can be entirely aligned in the assembly, with only one copy
present.</li>
<li>
<code>D</code> (Duplicated Complete Genes): The BUSCO genes that can
be completely aligned in the assembly, with more than one copy
present.</li>
<li>
<code>F</code> (Fragmented Genes, subclass 1): The BUSCO genes which
only a portion of the gene is present in the assembly, and the rest of
the gene cannot be aligned.</li>
<li>
<code>I</code> (Fragmented Genes, subclass 2): The BUSCO genes in
which a section of the gene aligns to one position in the assembly,
while the remaining part aligns to another position.</li>
<li>
<code>M</code> (Missing Genes): The BUSCO genes with no alignment
present in the assembly.</li>
</ul></section><section><h2 class="section-heading" id="merqury-for-evaluating-genome-assembly">Merqury for evaluating genome assembly<a class="anchor" aria-label="anchor" href="#merqury-for-evaluating-genome-assembly"></a>
</h2>
<hr class="half-width">
<p>Merqury is a tool for reference-free assembly evaluation based on
efficient k-mer set operations. It provides insights into various
aspects of genome assembly, offering a comprehensive view of genome
quality without relying on a reference sequence. Specifically, Merqury
can generate the following plots and metrics:</p>
<ul>
<li>
<strong>Copy Number Spectrum (Spectra-cn Plot):</strong>
<ul>
<li>A <strong>k-mer-based analysis</strong> that detects heterozygosity
levels and genome repeats by identifying peaks in k-mer coverage.<br>
</li>
<li>Helps estimate genome size, detect missing regions, and distinguish
between homozygous and heterozygous k-mers in an assembly.</li>
</ul>
</li>
<li>
<strong>Assembly Spectrum (Spectra-asm Plot):</strong>
<ul>
<li>Compares k-mers between different assemblies or between an assembly
and raw sequencing reads.<br>
</li>
<li>Useful for detecting missing sequences, shared regions, and
assembly-specific k-mers that may indicate errors or haplotype-specific
variations.</li>
</ul>
</li>
<li>
<strong>K-mer Completeness:</strong>
<ul>
<li>Measures how many <strong>reliable k-mers</strong> (those likely to
be real and not sequencing errors) are present in both the sequencing
reads and the assembly.<br>
</li>
<li>Helps identify missing regions, misassemblies, and sequencing biases
affecting genome reconstruction.</li>
</ul>
</li>
<li>
<strong>Consensus Quality (QV) Estimation:</strong>
<ul>
<li>Uses <strong>k-mer agreement between the assembly and the read
set</strong> to estimate base-level accuracy.<br>
</li>
<li>Higher QV scores indicate a more accurate consensus sequence, but
results depend on read quality and coverage depth.</li>
</ul>
</li>
<li>
<strong>Misassembly Detection with K-mer Positioning:</strong>
<ul>
<li>Identifies <strong>unexpected k-mers</strong> or <strong>false
duplications</strong> in assemblies, reporting their positions in
<code>.bed</code> and <code>.tdf</code> files for visualization in
genome browsers.<br>
</li>
<li>Helps pinpoint structural errors such as collapsed repeats, chimeric
joins, or large insertions/deletions.</li>
</ul>
</li>
</ul>
<p>This <strong>k-mer-based approach</strong> in Merqury provides
<strong>reference-free genome quality evaluation</strong>, making it
highly effective for <strong>de novo assemblies and structural
validation</strong>.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="ex">ml</span> merqury</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">ml</span> meryl</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> merqury_evaluation</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co"># ln -s ../assembly1.fasta all_assemblies/assembly1.fasta</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co"># ln -s ../assembly2.fasta all_assemblies/assembly2.fasta</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co"># ln -s ../assembly3.fasta all_assemblies/assembly3.fasta</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co"># link any other assemblies you want to compare</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co"># ln -s ../quast_evaluation/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa # reference</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="co"># ln -s ../pacbio/9994.q20.CCS-filtered-60x.fastq # pacbio reads</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="ex">meryl</span> <span class="dt">\</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>   count k=21 <span class="dt">\</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>   threads=<span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>   memory=8g <span class="dt">\</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a>   output 9994.q20.CCS-filtered.meryl<span class="dt">\</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a>   9994.q20.CCS-filtered.fastq</span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a><span class="ex">merqury</span> <span class="dt">\</span></span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a>   <span class="at">-a</span> assembly1.fasta <span class="dt">\</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a>   <span class="at">-r</span> 9994.q20.CCS-filtered.meryl <span class="dt">\</span></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a>   <span class="at">-o</span> merqury_evaluation/assembly1</span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a><span class="ex">merqury.sh</span> <span class="dt">\</span></span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a>   9994.q20.CCS-filtered.meryl </span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a>   <span class="ex">assembly1.fasta</span> assembly2.fasta assembly3.fasta</span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a>   <span class="ex">merqury_evaluation_output</span></span></code></pre>
</div>
<p>This will generate numberous files with
<code>merqury_evaluation_output</code> prefix, including k-mer spectra,
completeness metrics, and consensus quality estimates for each assembly.
You can use these results to evaluate the accuracy, completeness, and
structural integrity of your genome assemblies.</p>
</section><section><h2 class="section-heading" id="assembly-graph-visualization-using-bandage">Assembly graph visualization using Bandage<a class="anchor" aria-label="anchor" href="#assembly-graph-visualization-using-bandage"></a>
</h2>
<hr class="half-width">
<p>Bandage is a tool for visualizing assembly graphs, which represent
the connections between contigs or scaffolds in a genome assembly. By
visualizing the graph structure, you can identify complex regions,
repetitive elements, and potential misassemblies that may affect the
genome reconstruction.</p>
<p>To visualize the assembly graph using Bandage:</p>
<ol style="list-style-type: decimal">
<li>Open a web browser and navigate to <a href="">desktop.negishi.rcac.purdue.edu</a>.</li>
<li>Log in with your Purdue Career Account username and password, but
append “,push” to your password.</li>
<li>Lauch the terminal and run the following command:</li>
</ol>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="ex">ml</span> bandage</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="ex">Bandage</span></span></code></pre>
</div>
<ol start="4" style="list-style-type: decimal">
<li>In the Bandage interface, navigate to your assembly folder (hifiasm
or flye), and load your assembly graph (e.g.,
<code>assembly1.fasta</code>) .</li>
<li>Explore the graph structure, identify complex regions, and visualize
connections between contigs or scaffolds.</li>
</ol>
<figure><img src="https://github.com/user-attachments/assets/172513cc-d43b-401d-afbe-239d415d12bb" alt="Bandage interface" class="figure mx-auto d-block"><div class="figcaption">Bandage interface</div>
</figure><div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>
<strong>QUAST</strong> evaluates assembly contiguity and quality
metrics.</li>
<li>
<strong>Compleasm</strong> assesses gene-space completeness in
genome assemblies.</li>
<li>
<strong>Merqury</strong> provides reference-free evaluation based on
k-mer analysis.</li>
<li>
<strong>Bandage</strong> visualizes assembly graphs for structural
assessment.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-9-annotation-strategies"><p>Content from <a href="9-annotation-strategies.html">Annotation Strategies</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/9-annotation-strategies.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What are the different strategies used for genome annotation?<br>
</li>
<li>How do various methods predict genes, and what are their strengths
and limitations?<br>
</li>
<li>What role do evidence play in improving gene predictions?<br>
</li>
<li>How do deep learning and large models enhance genome annotation
accuracy?<br>
</li>
<li>What tools are used in this workshop, and how do they fit into
different strategies?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the principles behind different genome annotation
approaches.<br>
</li>
<li>Learn how intrinsic, extrinsic, and hybrid methods predict gene
structures.<br>
</li>
<li>Explore the role of multi-evidence approaches, including
transcriptomics, proteomics, and functional annotation.<br>
</li>
<li>Introduce modern machine learning and large model-based methods for
gene annotation.<br>
</li>
<li>Familiarize with annotation tools used in this workshop and their
specific applications.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<hr class="half-width">
<ul>
<li>Genome annotation identifies functional elements in a genome,
including protein-coding genes, non-coding rnas, and regulatory
regions<br>
</li>
<li>Annotation strategies vary between evidence-based, ab initio, and
hybrid approaches, each with strengths and limitations<br>
</li>
<li>Selecting the right method depends on genome complexity, available
transcriptomic and proteomic data, and the need for accuracy versus
sensitivity</li>
</ul></section><section><h2 class="section-heading" id="annotation-methods">Annotation Methods<a class="anchor" aria-label="anchor" href="#annotation-methods"></a>
</h2>
<hr class="half-width">
<p><strong>Intrinsic (Ab Initio) Methods</strong></p>
<ul>
<li>Predict genes based on sequence features without external
evidence.<br>
</li>
<li>Use statistical models such as hidden Markov models (HMMs) to
recognize coding regions, exon-intron boundaries, and gene
structures.<br>
</li>
<li>Rely on nucleotide composition patterns like codon usage, GC
content, and splice site signals.<br>
</li>
<li>Perform well in well-studied genomes but struggle with novel genes
and complex gene structures.</li>
</ul>
<p><strong>Extrinsic (Evidence-Based) Methods</strong></p>
<ul>
<li>Utilize external data such as RNA-seq, ESTs, and protein homology to
guide gene predictions.<br>
</li>
<li>Align transcriptomic and proteomic data to the genome to refine
exon-intron structures.<br>
</li>
<li>Improve annotation accuracy by incorporating evolutionary
conservation and known gene models.<br>
</li>
<li>Require high-quality reference datasets, making them less effective
in newly sequenced organisms with little prior data.</li>
</ul>
<p><strong>Hybrid Methods</strong></p>
<ul>
<li>Combine ab initio predictions with evidence-based approaches to
improve accuracy and completeness.<br>
</li>
<li>Use gene predictions from statistical models and refine them with
transcriptomic and proteomic data.<br>
</li>
<li>Balance sensitivity (detecting novel genes) and specificity
(avoiding false positives).<br>
</li>
<li>Often used in large-scale genome annotation projects where both
approaches complement each other.</li>
</ul>
<p><strong>Multi-Evidence based methods</strong></p>
<ul>
<li>Incorporate diverse biological data sources to improve gene
prediction accuracy.<br>
</li>
<li>Use <strong>ribosome profiling (Ribo-seq)</strong> and
<strong>proteomics</strong> to confirm protein-coding potential.<br>
</li>
<li>Leverage <strong>poly(A) signal presence</strong> to distinguish
mature transcripts from non-coding RNA.<br>
</li>
<li>Integrate <strong>RNA-seq, long-read sequencing, and EST
data</strong> to refine exon-intron structures.<br>
</li>
<li>Provide direct experimental validation, reducing reliance on purely
computational models.</li>
</ul>
<p><strong>Large Model-Based Methods</strong></p>
<ul>
<li>Utilize deep learning and large language models (LMMs) trained on
vast genomic datasets.<br>
</li>
<li>Predict gene structures by recognizing sequence patterns learned
from diverse organisms.<br>
</li>
<li>Adapt to various genome complexities without requiring manually
curated training data.<br>
</li>
<li>Identify non-canonical gene structures, alternative splicing events,
and novel genes.<br>
</li>
<li>Example: <strong>Helixer</strong>, a GPU-based model, predicts genes
using high-dimensional sequence patterns, improving over traditional
HMM-based methods.</li>
</ul>
<div id="what-do-we-annotate-in-the-genome" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="what-do-we-annotate-in-the-genome" class="callout-inner">
<h3 class="callout-title">What do we annotate in the genome?</h3>
<div class="callout-content">
<ul>
<li>
<strong>Protein-coding genes</strong>: Genes that encode proteins,
identified based on open reading frames (ORFs) and coding sequence
features.<br>
</li>
<li>
<strong>Pseudogenes</strong>: Non-functional remnants of once-active
genes that have accumulated mutations.<br>
</li>
<li>
<strong>Non-coding RNAs (ncRNAs)</strong>: Functional RNAs that do
not code for proteins, including tRNAs, rRNAs, and lncRNAs.<br>
</li>
<li>
<strong>Upstream Open Reading Frames (uORFs)</strong>: Small ORFs in
the 5’ untranslated region (UTR) that can regulate translation.<br>
</li>
<li>
<strong>Alternative Splicing Variants</strong>: Different transcript
isoforms from the same gene, generated by exon-intron
rearrangements.<br>
</li>
<li>
<strong>Repeat Elements and Transposons</strong>: Mobile DNA
sequences that can impact gene function and genome structure.<br>
</li>
<li>
<strong>Regulatory Elements</strong>: Promoters, enhancers, and
other sequences controlling gene expression.</li>
</ul>
<p>For this workshop, we focus only on <strong>protein-coding
genes</strong> and their annotation.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="annotation-workflows-used">Annotation Workflows Used<a class="anchor" aria-label="anchor" href="#annotation-workflows-used"></a>
</h2>
<hr class="half-width">
<p><strong>1. BRAKER3</strong>: Combines GeneMark-ET and AUGUSTUS to
predict genes in eukaryotic genomes.</p>
<ul>
<li>Integrates GeneMark and AUGUSTUS with RNA-Seq and/or protein
evidence.</li>
<li>Supports various data types or runs without them.</li>
<li>Fully automated training and gene prediction.</li>
<li>Requires evidence data for high accuracy; performs poorly without
it.</li>
<li>Struggles with scalability on larger genomes.</li>
</ul>
<div class="tabs">
<nav><div id="nav-tab-1" class="nav nav-tabs" role="tablist">
<button class="nav-link active" id="nav-tab-1-1" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-1" type="button" role="tab" aria-controls="nav-tabpanel-1-1" aria-selected="true">
  <h3 class="tab-header" id="nav-tab-heading-1-1">
  w/ RNA-seq
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-2" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-2" type="button" role="tab" aria-controls="nav-tabpanel-1-2" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-2">
  w/ Proteins
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-3" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-3" type="button" role="tab" aria-controls="nav-tabpanel-1-3" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-3">
  w/ RNA-seq + Proteins
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-4" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-4" type="button" role="tab" aria-controls="nav-tabpanel-1-4" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-4">
  ab initio
  </h3>
</button>
<button class="nav-link" id="nav-tab-1-5" data-bs-toggle="tab" data-bs-target="#nav-tabpanel-1-5" type="button" role="tab" aria-controls="nav-tabpanel-1-5" aria-selected="false">
  <h3 class="tab-header" id="nav-tab-heading-1-5">
  Pretrained model
  </h3>
</button>
</div>
</nav><div id="nav-tabContent-1" class="tab-content">
<div id="nav-tabpanel-1-1" class="tab-pane show active" aria-labelledby="nav-tab-1-1" role="tabpanel">
<figure><img src="https://github.com/user-attachments/assets/201d283e-83cc-4d2a-bb2b-68104dc72cc3" alt="with RNA-Seq" height="500" class="figure mx-auto d-block"></figure>
</div>
<div id="nav-tabpanel-1-2" class="tab-pane" aria-labelledby="nav-tab-1-2" role="tabpanel">
<figure><img src="https://github.com/user-attachments/assets/6a0cc45f-12f4-4363-81de-93c929166334" alt="with Proteins" height="550" class="figure mx-auto d-block"></figure>
</div>
<div id="nav-tabpanel-1-3" class="tab-pane" aria-labelledby="nav-tab-1-3" role="tabpanel">
<figure><img src="https://github.com/user-attachments/assets/b48adfaa-c357-411d-a252-8450cf720e5f" alt="with RNAseq and proteins" class="figure mx-auto d-block"><div class="figcaption">with RNAseq and proteins</div>
</figure>
</div>
<div id="nav-tabpanel-1-4" class="tab-pane" aria-labelledby="nav-tab-1-4" role="tabpanel">
<figure><img src="https://github.com/user-attachments/assets/5049a336-398a-42b6-834c-3adf75d8be07" alt="_ab initio_" height="500" class="figure mx-auto d-block"></figure>
</div>
<div id="nav-tabpanel-1-5" class="tab-pane" aria-labelledby="nav-tab-1-5" role="tabpanel">
<figure><img src="https://github.com/user-attachments/assets/81dcb3b6-25a1-4c22-a550-f6906a44af4e" alt="_ab initio_" height="500" class="figure mx-auto d-block"></figure>
</div>
</div>
</div>
<p><strong>2. Helixer</strong>: A deep learning-based gene prediction
tool that uses a convolutional neural network (CNN) to predict genes in
eukaryotic genomes.</p>
<ul>
<li>Utilizes a deep learning model trained on diverse animal and plant
genomes, reducing the need for retraining on closely related
species.</li>
<li>Provides probabilities for every base, enabling detailed predictions
for intergenic, untranslated, coding, and intronic regions.</li>
<li>Outperforms traditional tools like AUGUSTUS in both base-wise
metrics and RNA-Seq consistency.</li>
<li>Handles large and complex eukaryotic genomes effectively, even for
species with longer genes.</li>
<li>Requires no additional data except genome – no repeatmasking!</li>
<li>Cannot predict alternative splicing events or spliced start
codons.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/fd826e9d-30f9-48fe-9e82-e93cc8d779ce" alt="Helixer" class="figure mx-auto d-block"><div class="figcaption">Helixer</div>
</figure><p><strong>3. EASEL</strong>: Efficient, Accurate, Scalable Eukaryotic
modeLs: eukaryotic genome annotation tool</p>
<ul>
<li>Combines machine learning, RNA folding, and functional annotations
to enhance prediction accuracy.</li>
<li>Leverages RNA-Seq, protein alignments, and additional evidence for
gene prediction.</li>
<li>Built on Nextflow, ensuring portability across HPC systems and
compatibility with Docker and Singularity.</li>
<li>Features EnTAP integration for streamlined functional
annotation.</li>
<li>Provides multiple formats (GFF, GTF, CDS, protein sequences) for
downstream analysis.</li>
<li>Employs BUSCO for quality assessment and automates parameter
optimization.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/12e320eb-5ea6-43a9-bedb-2ae64097e44b" alt="EASEL" class="figure mx-auto d-block"><div class="figcaption">EASEL</div>
</figure><p><strong>4. EnTAP</strong>: eukaryotic non-model annotation pipeline
(functional annotation)</p>
<ul>
<li>Optimized for functional annotation of transcriptomes without
reference genomes, addressing challenges like fragmentation and assembly
artifacts.</li>
<li>Much faster than traditional annotation pipelines.</li>
<li>Integrates taxonomic filtering, contaminant detection, and
informativeness metrics for selecting optimal alignments.</li>
<li>Enables annotation across diverse repositories (e.g., RefSeq,
Swiss-Prot) with customizable database integration.</li>
<li>Includes gene family assignments, protein domains, and GO terms for
enriched annotation.</li>
<li>User-friendly, Unix-based pipeline suitable for HPC environments,
combining simplicity with high throughput.</li>
</ul>
<figure><img src="https://github.com/user-attachments/assets/11c2a5e7-2545-45a8-9d59-0b8d233f4de9" alt="EnTAP" class="figure mx-auto d-block"><div class="figcaption">EnTAP</div>
</figure><div id="how-to-we-access-gene-predictions" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="how-to-we-access-gene-predictions" class="callout-inner">
<h3 class="callout-title">How to we access gene predictions?</h3>
<div class="callout-content">
<ul>
<li>
<strong>Busco and Omark</strong> assess gene set completeness by
checking the presence of conserved orthologs across species, helping
identify missing or fragmented genes<br>
</li>
<li>
<strong>Gff3 metrics</strong> provide structural statistics on gene
models, including exon-intron distribution, gene lengths, and coding
sequence properties, highlighting inconsistencies in annotation<br>
</li>
<li>
<strong>Feature assignment</strong> using RNA-seq read mapping
quantifies how well predicted genes capture transcriptomic evidence,
indicating potential missing or incorrect annotations<br>
</li>
<li>
<strong>Reference comparison</strong> with known annotations
measures sensitivity and precision, identifying correctly predicted,
missed, or falsely annotated genes<br>
</li>
<li>
<strong>Multiple validation methods</strong> combine structural,
functional, and comparative assessments to improve confidence in gene
predictions and refine annotation quality</li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Genome annotation strategies include <strong>ab initio,
evidence-based, hybrid, multi-evidence, and large model-based
methods</strong>, each with different strengths and data
requirements.<br>
</li>
<li>
<strong>Ab initio methods</strong> predict genes using sequence
patterns, while <strong>evidence-based methods</strong> use RNA-seq,
proteins, and evolutionary conservation for refinement.<br>
</li>
<li>
<strong>Multi-evidence approaches</strong> integrate diverse
biological data such as ribosome profiling, proteomics, and long-read
RNA-seq to improve accuracy.<br>
</li>
<li>
<strong>Large model-based methods</strong>, like Helixer, use deep
learning to predict genes without requiring manual training,
outperforming traditional tools in complex genomes.<br>
</li>
<li>The workshop focuses on <strong>protein-coding gene
annotation</strong> using tools like <strong>BRAKER, Helixer, EASEL, and
EnTAP</strong>, followed by assessment with BUSCO, OMARK, and structural
metrics.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-10-gene-annotation-with-braker"><p>Content from <a href="10-gene-annotation-with-braker.html">Annotation using BRAKER</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/10-gene-annotation-with-braker.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is BRAKER3?</li>
<li>What are the different scenarios in which BRAKER3 can be used?</li>
<li>How to run BRAKER3 with different input requirements?</li>
<li>What are the different output files generated by BRAKER3?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand the different scenarios in which BRAKER3 can be used</li>
<li>Learn how to run BRAKER3 with different input requirements</li>
<li>Learn how to interpret the output files generated by BRAKER3</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>BRAKER3 is a pipeline that combines GeneMark-ET and AUGUSTUS to
predict genes in eukaryotic genomes. This pipeline is particularly
useful for annotating newly sequenced genomes. The flexibility of
BRAKER3 allows users to provide various input datasets for improving
gene prediction accuracy. In this example, we will use various scenarios
to predict genes in a Maize genome using BRAKER3. Following are the
scenarios we will cover</p>
<section><h2 class="section-heading" id="organizing-the-data">Organizing the Data<a class="anchor" aria-label="anchor" href="#organizing-the-data"></a>
</h2>
<hr class="half-width">
<p>Before running BRAKER3, we need to organize the data in the following
format:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case{1..5}</span></code></pre>
</div>
<figure><img src="https://github.com/user-attachments/assets/d08a88c4-e847-4e5d-b3cc-c60db10453d6" alt="Folder organization" class="figure mx-auto d-block"><div class="figcaption">Folder organization</div>
</figure></section><section><h2 class="section-heading" id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<hr class="half-width">
<p>Before running BRAKER3, we need to set up:</p>
<ol style="list-style-type: decimal">
<li>
<code>GeneMark-ES/ET/EP/ETP</code> license key</li>
<li>The <code>AUGUSTUS_CONFIG_PATH</code> configuration path</li>
</ol>
<p>The license key for <code>GeneMark-ES/ET/EP/ETP</code> can be
obtained from the <a href="http://exon.gatech.edu/GeneMark/license_download.cgi" class="external-link">GeneMark
website</a>. Once downloaded, you need to place it in your home
directory:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">tar</span> xf gm_key_64.gz</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">cp</span> gm_key_64 ~/.gm_key</span></code></pre>
</div>
<p>For the <code>AUGUSTUS_CONFIG_PATH</code>, we need to copy the
<code>config</code> directory from the Singularity container to the
scratch directory. This is required because BRAKER3 needs to write to
the <code>config</code> directory, and the Singularity container is
read-only. To copy the <code>config</code> directory, run the following
command:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="ex">ml</span> braker3/v3.0.7.5</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${WORKSHOP_DIR}</span>/04_braker/augustus</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="ex">copy_augustus_config</span> <span class="va">${WORKSHOP_DIR}</span>/04_braker/augustus</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="bu">export</span> <span class="va">AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/augustus/config"</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="case-1-using-rnaseq-reads">Case 1: Using RNAseq Reads<a class="anchor" aria-label="anchor" href="#case-1-using-rnaseq-reads"></a>
</h2>
<hr class="half-width">
<p>To run BRAKER3 with RNAseq reads, we need to provide the RNAseq reads
in BAM format. We will use the following command to run BRAKER3 with
RNAseq reads:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks=32</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#SBATCH --account=workshop</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#SBATCH --time=04-00:00:00</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#SBATCH --job-name=braker_case1</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#SBATCH --output=%x.o%j</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#SBATCH --error=%x.e%j</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co"># Load the required modules</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="ex">ml</span> braker3/v3.0.7.5</span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="co"># directory/file paths</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="va">BAM_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/bamfiles"</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="va">BAM_FILES</span><span class="op">=</span><span class="va">$(</span><span class="fu">find</span> <span class="st">"</span><span class="va">${BAM_DIR}</span><span class="st">"</span> <span class="at">-name</span> <span class="st">"*.bam"</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">'\n'</span> <span class="st">','</span><span class="va">)</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/athaliana_softmasked.fasta"</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a><span class="co"># runtime parameters</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a><span class="va">AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/augustus/config"</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a><span class="va">GENEMARK_PATH</span><span class="op">=</span><span class="st">"/opt/ETP/bin/gmes"</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a><span class="co"># Run BRAKER3</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case1</span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a><span class="ex">braker.pl</span> <span class="dt">\</span></span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a>   <span class="at">--AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="va">${AUGUSTUS_CONFIG_PATH}</span> <span class="dt">\</span></span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a>   <span class="at">--GENEMARK_PATH</span><span class="op">=</span><span class="va">${GENEMARK_PATH}</span> <span class="dt">\</span></span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a>   <span class="at">--bam</span><span class="op">=</span><span class="va">${BAM_FILES}</span> <span class="dt">\</span></span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a>   <span class="at">--genome</span><span class="op">=</span><span class="va">${genome}</span> <span class="dt">\</span></span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a>   <span class="at">--species</span><span class="op">=</span>At_<span class="va">$(</span><span class="fu">date</span> +<span class="st">"%Y%m%d"</span><span class="va">)</span>.c1b <span class="dt">\</span></span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a>   <span class="at">--workingdir</span><span class="op">=</span><span class="va">${workdir}</span> <span class="dt">\</span></span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a>   <span class="at">--gff3</span> <span class="dt">\</span></span>
<span id="cb4-32"><a href="#cb4-32" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>The job takes ~55 mins to finish. The results of the BRAKER3 run will
be stored in the respective <code>braker_case1</code> directory, as
<code>braker.gff3</code> file.</p>
</section><section><h2 class="section-heading" id="case-2-using-conserved-proteins-only">Case 2: Using Conserved proteins only<a class="anchor" aria-label="anchor" href="#case-2-using-conserved-proteins-only"></a>
</h2>
<hr class="half-width">
<p>Using the <a href="https://github.com/tomasbruna/orthodb-clades" class="external-link">orthodb-clades</a>
tool, we can download protein sequences for a specific clade. In this
scenario, since we are using the Arabidopsis genome, we can download the
<code>clades</code> specific <code>Viridiplantae.fa</code> <a href="https://www.orthodb.org/" class="external-link">OrthoDB v12</a> protein sets.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case2</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${workdir}</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="fu">git</span> clone git@github.com:tomasbruna/orthodb-clades.git</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="bu">cd</span> orthodb-clades</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="ex">ml</span> snakemake</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="ex">snakemake</span> <span class="at">--cores</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> selectViridiplantae</span></code></pre>
</div>
<p>When this is done, you should see a folder named <code>clade</code>
with <code>Viridiplantae.fa</code> in the <code>orthodb-clades</code>
directory. We will use this as one of the input datasets for
BRAKER3.<br>
The following command will run BRAKER3 with the input genome and protein
sequences:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks=32</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co">#SBATCH --time=04-00:00:00</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#SBATCH --job-name=braker_case2</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#SBATCH --output=%x.o%j</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#SBATCH --error=%x.e%j</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co"># Load the required modules</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="ex">ml</span> braker3/v3.0.7.5</span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="co"># directory/file paths</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="va">proteins</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/braker_case2/orthodb-clades/clades/Viridiplantae.fa"</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/athaliana_softmasked.fasta"</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="co"># runtime parameters</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a><span class="va">AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/augustus/config"</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a><span class="va">GENEMARK_PATH</span><span class="op">=</span><span class="st">"/opt/ETP/bin/gmes"</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a><span class="co"># Run BRAKER3</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case2</span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${workdir}</span></span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb6-23"><a href="#cb6-23" tabindex="-1"></a><span class="ex">braker.pl</span> <span class="dt">\</span></span>
<span id="cb6-24"><a href="#cb6-24" tabindex="-1"></a>   <span class="at">--AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="va">${AUGUSTUS_CONFIG_PATH}</span> <span class="dt">\</span></span>
<span id="cb6-25"><a href="#cb6-25" tabindex="-1"></a>   <span class="at">--GENEMARK_PATH</span><span class="op">=</span><span class="va">${GENEMARK_PATH}</span> <span class="dt">\</span></span>
<span id="cb6-26"><a href="#cb6-26" tabindex="-1"></a>   <span class="at">--genome</span><span class="op">=</span><span class="va">${genome}</span> <span class="dt">\</span></span>
<span id="cb6-27"><a href="#cb6-27" tabindex="-1"></a>   <span class="at">--prot_seq</span><span class="op">=</span><span class="va">${proteins}</span> <span class="dt">\</span></span>
<span id="cb6-28"><a href="#cb6-28" tabindex="-1"></a>   <span class="at">--species</span><span class="op">=</span>At_<span class="va">$(</span><span class="fu">date</span> +<span class="st">"%Y%m%d"</span><span class="va">)</span>.c2 <span class="dt">\</span></span>
<span id="cb6-29"><a href="#cb6-29" tabindex="-1"></a>   <span class="at">--workingdir</span><span class="op">=</span><span class="va">${workdir}</span> <span class="dt">\</span></span>
<span id="cb6-30"><a href="#cb6-30" tabindex="-1"></a>   <span class="at">--gff3</span> <span class="dt">\</span></span>
<span id="cb6-31"><a href="#cb6-31" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>The results of the BRAKER3 run will be stored in the respective
<code>braker_case2</code> directory, as <code>braker.gff3</code> file.
The run time for this job is ~170 mins.</p>
</section><section><h2 class="section-heading" id="case-3-using-rnaseq-and-proteins-together">Case 3: Using RNAseq and Proteins Together<a class="anchor" aria-label="anchor" href="#case-3-using-rnaseq-and-proteins-together"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks=32</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co">#SBATCH --time=04-00:00:00</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co">#SBATCH --job-name=braker_case3</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co">#SBATCH --output=%x.o%j</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="co">#SBATCH --error=%x.e%j</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="co"># Load the required modules</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a><span class="ex">ml</span> braker3/v3.0.7.5</span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a><span class="co"># directory/file paths</span></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a><span class="va">BAM_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/bamfiles"</span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a><span class="va">BAM_FILES</span><span class="op">=</span><span class="va">$(</span><span class="fu">find</span> <span class="st">"</span><span class="va">${BAM_DIR}</span><span class="st">"</span> <span class="at">-name</span> <span class="st">"*.bam"</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">'\n'</span> <span class="st">','</span><span class="va">)</span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a><span class="va">proteins</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/braker_case2/orthodb-clades/clade/Viridiplantae.fa"</span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/athaliana_softmasked.fasta"</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a><span class="co"># runtime parameters</span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a><span class="va">AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/augustus/config"</span></span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a><span class="va">GENEMARK_PATH</span><span class="op">=</span><span class="st">"/opt/ETP/bin/gmes"</span></span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a><span class="co"># Run BRAKER3</span></span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case3</span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${workdir}</span></span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a><span class="ex">braker.pl</span> <span class="dt">\</span></span>
<span id="cb7-26"><a href="#cb7-26" tabindex="-1"></a>   <span class="at">--AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="va">${AUGUSTUS_CONFIG_PATH}</span> <span class="dt">\</span></span>
<span id="cb7-27"><a href="#cb7-27" tabindex="-1"></a>   <span class="at">--GENEMARK_PATH</span><span class="op">=</span><span class="va">${GENEMARK_PATH}</span> <span class="dt">\</span></span>
<span id="cb7-28"><a href="#cb7-28" tabindex="-1"></a>   <span class="at">--genome</span><span class="op">=</span><span class="va">${genome}</span> <span class="dt">\</span></span>
<span id="cb7-29"><a href="#cb7-29" tabindex="-1"></a>   <span class="at">--prot_seq</span><span class="op">=</span><span class="va">${proteins}</span> <span class="dt">\</span></span>
<span id="cb7-30"><a href="#cb7-30" tabindex="-1"></a>   <span class="at">--bam</span><span class="op">=</span><span class="va">${BAM_FILES}</span> <span class="dt">\</span></span>
<span id="cb7-31"><a href="#cb7-31" tabindex="-1"></a>   <span class="at">--species</span><span class="op">=</span>At_<span class="va">$(</span><span class="fu">date</span> +<span class="st">"%Y%m%d"</span><span class="va">)</span>.c3 <span class="dt">\</span></span>
<span id="cb7-32"><a href="#cb7-32" tabindex="-1"></a>   <span class="at">--workingdir</span><span class="op">=</span><span class="va">${workdir}</span> <span class="dt">\</span></span>
<span id="cb7-33"><a href="#cb7-33" tabindex="-1"></a>   <span class="at">--gff3</span> <span class="dt">\</span></span>
<span id="cb7-34"><a href="#cb7-34" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>The results of the BRAKER3 run will be stored in the respective
<code>braker_case3</code> directory, as <code>braker.gff3</code> file.
The run time for this job is ~120 mins.</p>
</section><section><h2 class="section-heading" id="case-4-ab-initio-mode">Case 4: <em>ab Initio</em> mode<a class="anchor" aria-label="anchor" href="#case-4-ab-initio-mode"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks=32</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="co">#SBATCH --time=04-00:00:00</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="co">#SBATCH --job-name=braker_case4</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co">#SBATCH --output=%x.o%j</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="co">#SBATCH --error=%x.e%j</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="co"># Load the required modules</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a><span class="ex">ml</span> braker3/v3.0.7.5</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a><span class="co"># directory/file paths</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/athaliana_softmasked.fasta"</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a><span class="co"># runtime parameters</span></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="va">AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/augustus/config"</span></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a><span class="va">GENEMARK_PATH</span><span class="op">=</span><span class="st">"/opt/ETP/bin/gmes"</span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a><span class="co"># Run BRAKER3</span></span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case4</span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${workdir}</span></span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a><span class="ex">braker.pl</span> <span class="dt">\</span></span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a>   <span class="at">--AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="va">${AUGUSTUS_CONFIG_PATH}</span> <span class="dt">\</span></span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a>   <span class="at">--GENEMARK_PATH</span><span class="op">=</span><span class="va">${GENEMARK_PATH}</span> <span class="dt">\</span></span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a>   <span class="at">--esmode</span> <span class="dt">\</span></span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a>   <span class="at">--genome</span><span class="op">=</span><span class="va">${genome}</span> <span class="dt">\</span></span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a>   <span class="at">--species</span><span class="op">=</span>Zm_<span class="va">$(</span><span class="fu">date</span> +<span class="st">"%Y%m%d"</span><span class="va">)</span>.c4 <span class="dt">\</span></span>
<span id="cb8-28"><a href="#cb8-28" tabindex="-1"></a>   <span class="at">--workingdir</span><span class="op">=</span><span class="va">${workdir}</span> <span class="dt">\</span></span>
<span id="cb8-29"><a href="#cb8-29" tabindex="-1"></a>   <span class="at">--gff3</span> <span class="dt">\</span></span>
<span id="cb8-30"><a href="#cb8-30" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>The results of the BRAKER3 run will be stored in the respective
<code>braker_case4</code> directory, as <code>braker.gff3</code> file.
The run time for this job is ~60 mins.</p>
</section><section><h2 class="section-heading" id="case-5-using-pre-trained-model">Case 5: Using Pre-trained Model<a class="anchor" aria-label="anchor" href="#case-5-using-pre-trained-model"></a>
</h2>
<hr class="half-width">
<div class="codewrapper sourceCode" id="cb9">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks=32</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="co">#SBATCH --time=04-00:00:00</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="co">#SBATCH --job-name=braker_case5</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a><span class="co">#SBATCH --output=%x.o%j</span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a><span class="co">#SBATCH --error=%x.e%j</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a><span class="co"># Load the required modules</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a><span class="ex">ml</span> braker3/v3.0.7.5</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a><span class="co"># directory/file paths</span></span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/00_datasets/genome/athaliana_softmasked.fasta"</span></span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a><span class="co"># runtime parameters</span></span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a><span class="va">AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="st">"</span><span class="va">${WORKSHOP_DIR}</span><span class="st">/04_braker/augustus/config"</span></span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a><span class="va">GENEMARK_PATH</span><span class="op">=</span><span class="st">"/opt/ETP/bin/gmes"</span></span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a><span class="co"># Run BRAKER3</span></span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/04_braker/braker_case5</span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${workdir}</span></span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a><span class="ex">braker.pl</span> <span class="dt">\</span></span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a>   <span class="at">--AUGUSTUS_CONFIG_PATH</span><span class="op">=</span><span class="va">${AUGUSTUS_CONFIG_PATH}</span> <span class="dt">\</span></span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a>   <span class="at">--GENEMARK_PATH</span><span class="op">=</span><span class="va">${GENEMARK_PATH}</span> <span class="dt">\</span></span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a>   <span class="at">--skipAllTraining</span> <span class="dt">\</span></span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a>   <span class="at">--genome</span><span class="op">=</span><span class="va">${genome}</span> <span class="dt">\</span></span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a>   <span class="at">--species</span><span class="op">=</span>arabidopsis <span class="dt">\</span></span>
<span id="cb9-28"><a href="#cb9-28" tabindex="-1"></a>   <span class="at">--workingdir</span><span class="op">=</span><span class="va">${workdir}</span> <span class="dt">\</span></span>
<span id="cb9-29"><a href="#cb9-29" tabindex="-1"></a>   <span class="at">--gff3</span> <span class="dt">\</span></span>
<span id="cb9-30"><a href="#cb9-30" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>The results of the BRAKER3 run will be stored in the respective
<code>braker_case5/Augustus</code> directory. The main result file
<code>augustus.ab_initio.gff3</code> will be used for downstream
analysis. Runtime for this job is ~7 mins.</p>
</section><section><h2 class="section-heading" id="results-and-outputs">Results and Outputs<a class="anchor" aria-label="anchor" href="#results-and-outputs"></a>
</h2>
<hr class="half-width">
<p>coming soon!</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>BRAKER3 is a pipeline that combines GeneMark-ET and AUGUSTUS to
predict genes in eukaryotic genomes</li>
<li>BRAKER3 can be used with different input datasets to improve gene
prediction accuracy</li>
<li>BRAKER3 can be run in different scenarios to predict genes in a
genome</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-11-gene-annotation-with-helixer"><p>Content from <a href="11-gene-annotation-with-helixer.html">Annotation using Helixer</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/11-gene-annotation-with-helixer.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How to predict genes using Helixer?</li>
<li>How to download trained models for Helixer?</li>
<li>How to run Helixer on the HPC cluster (Gilbreth)?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Learn how to predict genes using Helixer.</li>
<li>Learn how to download trained models for Helixer.</li>
<li>Learn how to run Helixer on the HPC cluster (Gilbreth).</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>Helixer is a deep learning-based gene prediction tool that uses a
convolutional neural network (CNN) to predict genes in eukaryotic
genomes. Helixer is trained on a wide range of eukaryotic genomes and
can predict genes in both plant and animal genomes. Helixer can predict
genes wihtout any extrinisic information such as RNA-seq data or
homology information, purely based on the sequence of the genome.</p>
<div id="this-section-should-be-run-on-gilbreth-hpc-cluster." class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="this-section-should-be-run-on-gilbreth-hpc-cluster." class="callout-inner">
<h3 class="callout-title">This section should be run on Gilbreth HPC cluster.</h3>
<div class="callout-content">
<p>Due to the GPU requirement for Helixer, you need to run this section
on the Gilbreth HPC cluster. You don’t have to copy fastq, or bamfiles,
but only need <code>athaliana.fasta</code> file in the
<code>00_datasets/genome</code> directory.</p>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${WORKSHOP_DIR}</span>/05_helixer</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">cp</span> <span class="op">&lt;</span>depot_location<span class="op">&gt;</span>/genome/athaliana.fasta <span class="va">${WORKSHOP_DIR}</span>/05_helixer/athaliana.fasta</span></code></pre>
</div>
<figure><img src="https://github.com/user-attachments/assets/a1767236-304b-4912-8200-e2ea97ec63b8" alt="Folder organization" class="figure mx-auto d-block"><div class="figcaption">Folder organization</div>
</figure><section><h2 class="section-heading" id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<hr class="half-width">
<p>Helixer is available as a Singularity/apptainer container. You can
pull the container using the following <code>apptainer pull</code>
command. See the <a href="https://github.com/gglyptodon/helixer-docker" class="external-link">Helixer Docker</a>
repository for more information.</p>
<p>Helixer is installed as a module on the Gilbreth cluster, and can be
loaded using the following commands:</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="ex">ml</span> helixer</span></code></pre>
</div>
</section><section><h2 class="section-heading" id="downloading-trained-models">Downloading trained models<a class="anchor" aria-label="anchor" href="#downloading-trained-models"></a>
</h2>
<hr class="half-width">
<p>Helixer requires a trained model to predict genes. With the included
script <code>fetch_helixer_models.py</code> you can download models for
specific lineages. Currently, models are available for the following
lineages:</p>
<ul>
<li><code>land_plant</code></li>
<li><code>vertebrate</code></li>
<li><code>invertibrate</code></li>
<li><code>fungi</code></li>
</ul>
<p>You can download the models using the following command:</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># all models</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co"># fetch_helixer_models.py --all</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co"># or for a specific lineage</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">fetch_helixer_models.py</span> <span class="at">--lineage</span> land_plant</span></code></pre>
</div>
<p>This will download all lineage models in the models directory. You
can also download models for specific lineages using the
<code>--lineage</code> option as shown above.</p>
<p>By default, files will be downloaded to
<code>~/.local/share/Helixer</code> directory. You should see the
follwing files:</p>
<pre><code>.
└── models
    └── land_plant
        ├── land_plant_v0.3_a_0080.h5
        ├── land_plant_v0.3_a_0090.h5
        ├── land_plant_v0.3_a_0100.h5
        ├── land_plant_v0.3_a_0200.h5
        ├── land_plant_v0.3_a_0300.h5
        ├── land_plant_v0.3_a_0400.h5
        ├── land_plant_v0.3_m_0100.h5
        └── land_plant_v0.3_m_0200.h5</code></pre>
<p><code>land_plant_v0.3_a_0080.h5</code> is the smallest model and
<code>land_plant_v0.3_m_0200.h5</code> is the largest model. The model
size is determined by the number of parameters in the model. The larger
models are more accurate but require more memory and time to run.</p>
</section><section><h2 class="section-heading" id="running-helixer">Running Helixer<a class="anchor" aria-label="anchor" href="#running-helixer"></a>
</h2>
<hr class="half-width">
<p>Helixer requires GPU for prediction. For running Helixer, you need to
request a GPU node. You will also need the genome sequence in fasta
format. For this tutorial, we will use Maize genome (Zea mays subsp.
mays), and use the <code>land_plant</code> model to predict genes.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks-per-node=16</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#SBATCH --gpus-per-node=1</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#SBATCH --time=04:00:00</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#SBATCH --account=standby</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#SBATCH --job-name=helixer</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#SBATCH --output=%x.o%j</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co">#SBATCH --error=%x.e%j</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="ex">ml</span> helixer</span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span>athaliana.fasta</span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="va">species</span><span class="op">=</span><span class="st">"Arabidopsis thaliana"</span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a><span class="va">output</span><span class="op">=</span><span class="st">"athaliana_helixer.gff"</span></span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a><span class="ex">Helixer.py</span> <span class="dt">\</span></span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a>    <span class="at">--lineage</span> land_plant <span class="dt">\</span></span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a>    <span class="at">--fasta-path</span> <span class="va">${genome}</span> <span class="dt">\</span></span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a>    <span class="at">--species</span> <span class="va">${species}</span> <span class="dt">\</span></span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a>    <span class="at">--gff-output-path</span> <span class="va">${output}</span></span></code></pre>
</div>
<p>A typical job will take ~40 mins to finish depedning on the GPU the
job gets allocated.</p>
<p>You can count the nubmer of predictions in your gff3 file using the
following command:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'$3=="gene"'</span> athaliana_helixer.gff <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code></pre>
</div>
<p>The GFF format output had 27,201 genes predicted using Helixer. You
can view the various features in the gff file using the following
command:</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-v</span> <span class="st">"^#"</span> athaliana_helixer.gff <span class="kw">|</span><span class="dt">\</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>   <span class="fu">cut</span> <span class="at">-f</span> 3 <span class="kw">|</span><span class="dt">\</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>   <span class="fu">sort</span> <span class="kw">|</span><span class="dt">\</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>   <span class="fu">uniq</span> <span class="at">-c</span></span></code></pre>
</div>
<p>To get <code>cds</code> and <code>pep</code> files, you can use the
following command:</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="ex">ml</span> cufflinks</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="ex">gffread</span> Arabidopsis_thaliana.gff3 <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>   <span class="at">-g</span> athaliana.fasta <span class="dt">\</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>   <span class="at">-y</span> helixer_pep.fa <span class="dt">\</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>   <span class="at">-x</span> helixer_cds.fa</span></code></pre>
</div>
<p>As you may have noticed, the number of mRNA and gene features are the
same. This is because isoforms aren’t predicted by Helixer and you only
have one transcript per gene. Exons are indetified with high confidence
and alternative isoforms are usually collapsed into a single gene model.
This is one of the known limitations of Helixer.</p>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Helixer is a deep learning-based gene prediction tool that uses a
convolutional neural network (CNN) to predict genes in eukaryotic
genomes.</li>
<li>Helixer can predict genes wihtout any extrinisic information such as
RNA-seq data or homology information, purely based on the sequence of
the genome.</li>
<li>Helixer requires a trained model and GPU for prediction.</li>
<li>Helixer predicts genes in the GFF3 format, but will not predict
isoforms.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-12-gene-annotation-with-gemoma"><p>Content from <a href="12-gene-annotation-with-gemoma.html">Homology-based gene annotation with GeMoMa</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/12-gene-annotation-with-gemoma.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>todo</li>
<li>todo</li>
<li>todo</li>
<li>todo</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>todo</li>
<li>todo</li>
<li>todo</li>
<li>todo</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div class="section level1">
<h1 id="entap-overview">EnTAP Overview<a class="anchor" aria-label="anchor" href="#entap-overview"></a>
</h1>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>todo</li>
<li>todo</li>
<li>todo</li>
<li>todo</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</div></section><section id="aio-13-functional-annotation-with-entap"><p>Content from <a href="13-functional-annotation-with-entap.html">Functional annotation using EnTAP</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/13-functional-annotation-with-entap.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is EnTAP, and how does it improve functional annotation for
non-model transcriptomes?</li>
<li>How does EnTAP filter, annotate, and assign functional roles to
predicted transcripts?</li>
<li>What databases and evidence sources does EnTAP integrate for
annotation?</li>
<li>What are the key steps required to set up and execute EnTAP on an
HPC system?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand how EnTAP improves functional annotation for non-model
eukaryotes.</li>
<li>Learn how EnTAP processes transcript data through filtering,
alignment, and functional assignment.</li>
<li>Set up and modify EnTAP configuration files for correct
execution.</li>
<li>Run EnTAP on an HPC system and interpret the generated
annotations.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="entap-overview">EnTAP Overview<a class="anchor" aria-label="anchor" href="#entap-overview"></a>
</h2>
<hr class="half-width">
<p>EnTAP is a bioinformatics pipeline designed to enhance the accuracy,
speed, and flexibility of functional annotation for de novo assembled
transcriptomes in non-model eukaryotic organisms. It mitigates assembly
fragmentation, improves annotation rates, and provides extensive
functional insights into transcriptomes. You can provide predicted
transcripts in FASTA format and a GFF file containing gene models to run
EnTAP.</p>
<div id="key-features" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="key-features" class="callout-inner">
<h3 class="callout-title">Key Features</h3>
<div class="callout-content">
<ul>
<li>
<strong>Optimized for non-model eukaryotes</strong> – Overcomes
challenges of fragmented transcriptome assemblies.</li>
<li>
<strong>Fast &amp; efficient</strong> – Runs significantly faster
than comparable annotation tools.</li>
<li>
<strong>Customizable</strong> – Supports optional filtering and
analysis steps for user-specific needs.</li>
<li>
<strong>Comprehensive functional insights</strong> – Integrates
multiple annotation sources for high-confidence gene assignments.</li>
<li>
<strong>Contaminant detection</strong> – Helps remove misleading
sequences for cleaner datasets.</li>
</ul>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="how-entap-works">How EnTAP Works<a class="anchor" aria-label="anchor" href="#how-entap-works"></a>
</h2>
<hr class="half-width">
<ol style="list-style-type: decimal">
<li>
<strong>Transcriptome Filtering</strong> – Identifies true coding
sequences (CDS) and removes assembly artifacts:
<ul>
<li>
<strong>Expression Filtering (optional)</strong> – Filters
transcripts based on gene expression levels using RSEM.</li>
<li>
<strong>Frame Selection (optional)</strong> – Further refines CDS
predictions using TransDecoder.</li>
</ul>
</li>
<li>
<strong>Transcriptome Annotation</strong> – Assigns functional
information to sequences:
<ul>
<li>
<strong>Similarity Search</strong> – Rapid alignment against
user-selected databases using DIAMOND.</li>
<li>
<strong>Contaminant Filtering &amp; Best Hit Selection</strong> –
Identifies optimal annotations and flags potential contaminants.</li>
<li>
<strong>Orthologous Group Assignment</strong> – Assigns translated
proteins to gene families using <strong>eggNOG/eggnog-mapper</strong>,
including:
<ul>
<li>
<strong>Protein Domains</strong> (SMART/Pfam)</li>
<li><strong>Gene Ontology (GO) Terms</strong></li>
<li><strong>KEGG Pathway Annotation</strong></li>
</ul>
</li>
<li>
<strong>InterProScan (optional)</strong> – Searches InterPro
databases for additional domain, GO, and pathway annotations.</li>
<li>
<strong>Horizontal Gene Transfer Analysis (optional)</strong> –
Detects potential horizontal gene transfer (HGT) events via
DIAMOND.</li>
</ul>
</li>
</ol></section><section><h2 class="section-heading" id="running-entap">Running EnTAP<a class="anchor" aria-label="anchor" href="#running-entap"></a>
</h2>
<hr class="half-width">
<p>EnTAP is available as a module on the HPC cluster. You can load the
module using the following commands:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">ml</span> entap</span></code></pre>
</div>
<div id="first-time-setup" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="first-time-setup" class="callout-inner">
<h3 class="callout-title">First-Time Setup</h3>
<div class="callout-content">
<p>When running for the first time, you will have to set up the
databases for EnTAP. This includes downloading files from various
databases, and can be time consuming. This section is already performed
so you can skip this step and is included for reference.</p>
</div>
</div>
</div>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  Setting up databases (optional)
  </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" aria-labelledby="headingSpoiler1" data-bs-parent="#accordionSpoiler1">
<div class="accordion-body">
<p><code>entap_run.params</code> file should be setup as follows (be
sure to select the correct <code>databases</code> for your
organism):</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">INI<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode ini" tabindex="0"><code class="sourceCode ini"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="dt">out-dir</span><span class="ot">=</span><span class="st">entap_dbfiles</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="dt">overwrite</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="dt">resume</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="dt">input</span><span class="ot">=</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="dt">database</span><span class="ot">=</span><span class="st">uniprot_sprot,refseq_plant</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="dt">no-trim</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="dt">threads</span><span class="ot">=</span><span class="dv">1</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="dt">output-format</span><span class="ot">=</span><span class="st">1,3,4,7,</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="dt">fpkm</span><span class="ot">=</span><span class="fl">0.5</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="dt">align</span><span class="ot">=</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="dt">single-end</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="dt">frame-selection</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="dt">transdecoder-m</span><span class="ot">=</span><span class="dv">100</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="dt">transdecoder-no-refine-starts</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="dt">taxon</span><span class="ot">=</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="dt">qcoverage</span><span class="ot">=</span><span class="dv">50</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="dt">tcoverage</span><span class="ot">=</span><span class="dv">50</span></span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="dt">contam</span><span class="ot">=</span></span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="dt">e-value</span><span class="ot">=</span><span class="st">1e-05</span></span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="dt">uninformative</span><span class="ot">=</span><span class="st">conserved,predicted,unknown,unnamed,hypothetical,putative,unidentified,uncharacterized,uncultured,uninformative,</span></span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="dt">diamond-sensitivity</span><span class="ot">=</span><span class="st">very-sensitive</span></span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a><span class="dt">ontology_source</span><span class="ot">=</span><span class="st">0,</span></span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a><span class="dt">eggnog-contaminant</span><span class="ot">=</span><span class="kw">true</span></span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a><span class="dt">eggnog-dbmem</span><span class="ot">=</span><span class="kw">true</span></span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a><span class="dt">eggnog-sensitivity</span><span class="ot">=</span><span class="st">more-sensitive</span></span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a><span class="dt">interproscan-db</span><span class="ot">=</span></span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a><span class="dt">hgt-donor</span><span class="ot">=</span></span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a><span class="dt">hgt-recipient</span><span class="ot">=</span></span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a><span class="dt">hgt-gff</span><span class="ot">=</span></span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a><span class="dt">ncbi-api-key</span><span class="ot">=</span></span>
<span id="cb2-31"><a href="#cb2-31" tabindex="-1"></a><span class="dt">ncbi-api-enable</span><span class="ot">=</span><span class="kw">true</span></span></code></pre>
</div>
<p><code>entap_config.ini</code> file should be setup as follows (be
sure to modify the paths <code>&lt;custom_location&gt;</code> to your
desired location):</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">INI<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode ini" tabindex="0"><code class="sourceCode ini"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="dt">data-generate</span><span class="ot">=</span><span class="kw">false</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="dt">data-type</span><span class="ot">=</span><span class="st">0,</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="dt">entap-db-bin</span><span class="ot">=</span><span class="st">&lt;custom_location&gt;/entap_db/bin/entap_database.bin</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="dt">entap-db-sql</span><span class="ot">=</span><span class="st">entap_database.db</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="dt">entap-graph</span><span class="ot">=</span><span class="st">entap_graphing.py</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="dt">rsem-calculate-expression</span><span class="ot">=</span><span class="st">rsem-calculate-expression</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="dt">rsem-sam-validator</span><span class="ot">=</span><span class="st">rsem-sam-validator</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="dt">rsem-prepare-reference</span><span class="ot">=</span><span class="st">rsem-prepare-reference</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="dt">convert-sam-for-rsem</span><span class="ot">=</span><span class="st">convert-sam-for-rsem</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="dt">transdecoder-long-exe</span><span class="ot">=</span><span class="st">TransDecoder.LongOrfs</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="dt">transdecoder-predict-exe</span><span class="ot">=</span><span class="st">TransDecoder.Predict</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="dt">diamond-exe</span><span class="ot">=</span><span class="st">diamond</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="dt">eggnog-map-exe</span><span class="ot">=</span><span class="st">emapper.py</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="dt">eggnog-map-data</span><span class="ot">=</span><span class="st">&lt;custom_location&gt;/entap_db/databases</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="dt">eggnog-map-dmnd</span><span class="ot">=</span><span class="st">&lt;custom_location&gt;/entap_db/bin/eggnog_proteins.dmnd</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="dt">interproscan-exe</span><span class="ot">=</span><span class="st">interproscan.sh</span></span></code></pre>
</div>
<p>Once done, for the first time setup, you can run the following
command:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="ex">ml</span> entap</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="ex">EnTAP</span> <span class="dt">\</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>   <span class="at">--config</span> <span class="dt">\</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>   <span class="at">--run-ini</span> ./entap_run.params <span class="dt">\</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>   <span class="at">--entap-ini</span> ./entap_config.ini <span class="dt">\</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
<p>This will download the databases and set up the configuration files
for EnTAP.</p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="step-1-prepare-files">
<strong>Step 1:</strong> Prepare files<a class="anchor" aria-label="anchor" href="#step-1-prepare-files"></a>
</h2>
<hr class="half-width">
<p>Your input files should be in the following format:</p>
<ul>
<li>
<strong>Transcript FASTA file</strong> – Contains predicted
transcripts in FASTA format.</li>
<li>
<strong>Configuration file</strong> – Specifies parameters for EnTAP
execution.
<ul>
<li>
<code>entap_run.params</code> – Contains runtime parameters for
EnTAP.</li>
<li>
<code>entap_config.ini</code> – Specifies paths to EnTAP binaries
and databases. (you can copy the files from
<code>/depot/itap/datasets/entap_db/entap_{config.ini,run.params}</code>)</li>
</ul>
</li>
</ul>
<p>Edit the <code>entap_run.params</code> file to specify the output
directory for EnTAP results and the correct input file</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">INI<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode ini" tabindex="0"><code class="sourceCode ini"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="dt">out-dir</span><span class="ot">=</span><span class="st">entap_out</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="dt">input</span><span class="ot">=</span><span class="st">input_cds.fasta</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="step-2-run-entap">
<strong>Step 2:</strong> Run EnTAP<a class="anchor" aria-label="anchor" href="#step-2-run-entap"></a>
</h2>
<hr class="half-width">
<p>Run EnTAP using the following command:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="ex">ml</span> entap</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="ex">EnTAP</span> <span class="at">-</span><span class="dt">\</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>   <span class="at">-run</span> <span class="dt">\</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>   <span class="at">--run-ini</span> ./entap_run.params <span class="dt">\</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>   <span class="at">--entap-ini</span> ./entap_config.ini <span class="dt">\</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>   <span class="at">--threads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="interpreting-results">Interpreting Results<a class="anchor" aria-label="anchor" href="#interpreting-results"></a>
</h2>
<hr class="half-width">
<p>EnTAP generates several output files, but the key results will be in
the <code>entap_out/final_results</code> directory.</p>
<p><strong>EnTAP Output Files Summary</strong></p>
<table class="table">
<colgroup>
<col width="42%">
<col width="57%">
</colgroup>
<thead><tr class="header">
<th align="left"><strong>File/Directory</strong></th>
<th><strong>Description</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><strong>Final Results
(<code>final_results/</code>)</strong></td>
<td></td>
</tr>
<tr class="even">
<td align="left"><code>annotated.fnn</code></td>
<td>FASTA file of <strong>annotated</strong> transcripts.</td>
</tr>
<tr class="odd">
<td align="left"><code>annotated.tsv</code></td>
<td>Tab-separated file with <strong>functional
annotations</strong>.</td>
</tr>
<tr class="even">
<td align="left"><code>annotated_gene_ontology_terms.tsv</code></td>
<td>GO terms assigned to annotated transcripts.</td>
</tr>
<tr class="odd">
<td align="left"><code>entap_results.tsv</code></td>
<td>
<strong>Master summary</strong> of all results, including
annotations.</td>
</tr>
<tr class="even">
<td align="left"><code>unannotated.fnn</code></td>
<td>FASTA file of <strong>unannotated</strong> transcripts.</td>
</tr>
<tr class="odd">
<td align="left"><code>unannotated.tsv</code></td>
<td>List of transcripts that <strong>failed</strong> annotation.</td>
</tr>
<tr class="even">
<td align="left"><strong><code>gene_family/</code></strong></td>
<td>Stores <strong>eggNOG</strong> gene family assignments, including
orthologs and functional annotations.</td>
</tr>
<tr class="odd">
<td align="left"><strong><code>similarity_search/</code></strong></td>
<td>Contains results from <strong>DIAMOND BLASTX</strong> searches
against selected databases.</td>
</tr>
<tr class="even">
<td align="left"><strong><code>transcriptomes/</code></strong></td>
<td>Holds the <strong>input transcriptome (CDS)</strong> and the
<strong>processed</strong> version after filtering.</td>
</tr>
</tbody>
</table>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>EnTAP enhances functional annotation by integrating multiple
evidence sources, including homology, protein domains, and gene
ontology.</li>
<li>Proper setup of configuration files and databases is essential for
accurate and efficient EnTAP execution.</li>
<li>Running EnTAP involves transcript filtering, similarity searches,
and functional annotation through automated workflows.</li>
<li>The pipeline provides extensive insights into transcript function,
improving downstream biological interpretations.</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section><section id="aio-14-annotation-assesment"><p>Content from <a href="14-annotation-assesment.html">Annotation Assesment</a></p>
<hr>
<p>Last updated on 2025-03-01 |

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/episodes/14-annotation-assesment.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 12 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How to assess the quality of a genome annotation?</li>
<li>What are the different tools available for assessing the quality of
a genome annotation?</li>
<li>How to compare the predicted annotation with the reference
annotation?</li>
<li>How to measure the number of raw reads assigned to the features
predicted by the annotation?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Assess the quality of a genome annotation using different
tools.</li>
<li>Compare the predicted annotation with the reference annotation.</li>
<li>Measure the number of raw reads assigned to the features predicted
by the annotation.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<hr class="half-width">
<p>The folder organization is as follows:</p>
<p>The annotation files (<code>gff3</code>) are from the previous steps.
But are collected in this folder for convenience.</p>
<p>We will extract CDS and Protein sequences from the <code>gff3</code>
file before we begin the assesment.</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="ex">ml</span> cufflinks</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="pp">*</span>.gff3<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${i</span><span class="op">%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>    <span class="ex">gffread</span> <span class="dt">\</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>       <span class="at">-g</span> <span class="va">${WORKSHOP_DIR}</span>/00_datasets/genome/athaliana.fa <span class="dt">\</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>       <span class="at">-x</span> <span class="va">${base}</span>_cds.fa <span class="dt">\</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>       <span class="at">-y</span> <span class="va">${base}</span>_pep.fa <span class="va">$i</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="gff3-metrics">GFF3 metrics<a class="anchor" aria-label="anchor" href="#gff3-metrics"></a>
</h2>
<hr class="half-width">
<p>We will use <code>agat_sp_statistics.pl</code> to calculate the
statistics of the GFF3 file.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="ex">ml</span> agat</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> gff3_stats</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="cf">for</span> gff3 <span class="kw">in</span> <span class="pp">*</span>.gff3<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$gff3</span> .gff3<span class="va">)</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>    <span class="ex">agat_sp_statistics.pl</span> <span class="dt">\</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>       <span class="at">--gff</span> <span class="va">${gff3}</span> <span class="dt">\</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>       <span class="at">-o</span> gff3_stats/<span class="va">${base}</span>_stats.txt</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="busco-assesment">BUSCO assesment<a class="anchor" aria-label="anchor" href="#busco-assesment"></a>
</h2>
<hr class="half-width">
<p>We will use BUSCO to asses the quality of the annotation.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">ml</span> busco</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="cf">for</span> pep <span class="kw">in</span> <span class="pp">*</span>_pep.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${pep</span><span class="op">%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>     <span class="ex">busco</span> <span class="dt">\</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>        <span class="at">--in</span> <span class="va">${pep}</span> <span class="dt">\</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>        <span class="at">--cpu</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>        <span class="at">--out</span> <span class="va">${base}</span>_busco <span class="dt">\</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>        <span class="at">--mode</span> prot <span class="dt">\</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>        <span class="at">--lineage_dataset</span> brassicales_odb10 <span class="dt">\</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a>        <span class="at">--force</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
<p>Once the BUSCO assesment is complete, we can view the results using
the following command:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="ex">ml</span> busco</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> busco_results</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="cf">for</span> f <span class="kw">in</span> <span class="pp">*</span>/<span class="pp">*</span>_busco/short_summary<span class="pp">*</span>busco.txt<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>   <span class="fu">cp</span> <span class="va">$f</span> busco_results/<span class="kw">;</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="ex">generate_plot.py</span> –wd busco_results</span></code></pre>
</div>
</section><section><h2 class="section-heading" id="omark-assesment">Omark assesment<a class="anchor" aria-label="anchor" href="#omark-assesment"></a>
</h2>
<hr class="half-width">
<p>OMArk is a software for proteome (protein-coding gene repertoire)
quality assessment. It provides measures of proteome completeness,
characterizes the consistency of all protein coding genes with regard to
their homologs, and identifies the presence of contamination from other
species.</p>
<p>The proteomes should be filtered to take only primary isoforms before
running OMark.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="ex">ml</span> seqkit</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="cf">for</span> pep <span class="kw">in</span> braker<span class="pp">*</span>_pep.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${pep</span><span class="op">%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>    <span class="fu">grep</span> <span class="st">"\.t1$"</span> <span class="va">$pep</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/&gt;//g'</span> <span class="op">&gt;</span> <span class="va">${base}</span>.primary.ids</span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>    <span class="ex">seqkit</span> grep <span class="at">-f</span> <span class="va">${base}</span>.primary.ids <span class="va">$pep</span> <span class="op">&gt;</span> <span class="va">${base}</span>.primary.pep.fa</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>    <span class="fu">rm</span> <span class="va">${base}</span>.primary.ids</span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="fu">cp</span> helixer_pep.fa helixer.primary.pep.fa</span></code></pre>
</div>
<p>Running OMark:</p>
<div class="codewrapper sourceCode" id="cb6">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="ex">ml</span> omark</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="va">database</span><span class="op">=</span><span class="st">"/depot/itap/datasets/omark/LUCA.h5"</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="cf">for</span> pep <span class="kw">in</span> <span class="pp">*</span>.primary.pep.fa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${pep</span><span class="op">%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>    <span class="ex">omamer</span> search <span class="dt">\</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>       <span class="at">--db</span> <span class="va">${database}</span> <span class="dt">\</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a>       <span class="at">--query</span> <span class="va">${pep}</span> <span class="dt">\</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a>       <span class="at">--out</span> <span class="va">${pep</span><span class="op">%</span>.<span class="pp">*</span><span class="va">}</span>.omamer <span class="dt">\</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a>       <span class="at">--nthreads</span> <span class="va">${SLURM_CPUS_ON_NODE}</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a>    <span class="ex">omark</span> <span class="dt">\</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a>       <span class="at">--file</span> <span class="va">${pep</span><span class="op">%</span>.<span class="pp">*</span><span class="va">}</span>.omamer <span class="dt">\</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a>       <span class="at">--database</span> <span class="va">${database}</span> <span class="dt">\</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a>       <span class="at">--outputFolder</span> <span class="va">${pep</span><span class="op">%</span>.<span class="pp">*</span><span class="va">}</span> <span class="dt">\</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a>       <span class="at">--og_fasta</span> <span class="va">${pep}</span></span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
<div id="accordionSpoiler1" class="accordion spoiler-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button spoiler-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSpoiler1" aria-expanded="false" aria-controls="collapseSpoiler1">
  <h3 class="accordion-header" id="headingSpoiler1">
  <div class="note-square"><i aria-hidden="true" class="callout-icon" data-feather="eye"></i></div>
  OMark Results Summary
  </h3>
</button>
<div id="collapseSpoiler1" class="accordion-collapse collapse" aria-labelledby="headingSpoiler1" data-bs-parent="#accordionSpoiler1">
<div class="accordion-body">
<p><strong>Conserved HOGs in Brassicaceae</strong></p>
<table class="table">
<thead><tr class="header">
<th align="left"><strong>Category</strong></th>
<th align="right"><strong>Count</strong></th>
<th align="center"><strong>Percentage</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><strong>Total Conserved HOGs</strong></td>
<td align="right">17,996</td>
<td align="center">100%</td>
</tr>
<tr class="even">
<td align="left"><strong>Single-Copy HOGs (S)</strong></td>
<td align="right">16,237</td>
<td align="center">90.23%</td>
</tr>
<tr class="odd">
<td align="left"><strong>Duplicated HOGs (D)</strong></td>
<td align="right">1,315</td>
<td align="center">7.31%</td>
</tr>
<tr class="even">
<td align="left"><em>Unexpected Duplications (U)</em></td>
<td align="right">351</td>
<td align="center">1.95%</td>
</tr>
<tr class="odd">
<td align="left"><em>Expected Duplications (E)</em></td>
<td align="right">964</td>
<td align="center">5.36%</td>
</tr>
<tr class="even">
<td align="left"><strong>Missing HOGs (M)</strong></td>
<td align="right">444</td>
<td align="center">2.47%</td>
</tr>
</tbody>
</table>
<p><strong>Proteome Composition</strong></p>
<table class="table">
<thead><tr class="header">
<th align="left"><strong>Category</strong></th>
<th align="right"><strong>Count</strong></th>
<th align="right"><strong>Percentage</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><strong>Total Proteins</strong></td>
<td align="right">26,994</td>
<td align="right">100%</td>
</tr>
<tr class="even">
<td align="left"><strong>Consistent (A)</strong></td>
<td align="right">25,480</td>
<td align="right">94.39%</td>
</tr>
<tr class="odd">
<td align="left"><em>Partial Hits (P)</em></td>
<td align="right">1,306</td>
<td align="right">4.84%</td>
</tr>
<tr class="even">
<td align="left"><em>Fragmented (F)</em></td>
<td align="right">630</td>
<td align="right">2.33%</td>
</tr>
<tr class="odd">
<td align="left"><strong>Inconsistent (I)</strong></td>
<td align="right">167</td>
<td align="right">0.62%</td>
</tr>
<tr class="even">
<td align="left"><em>Partial Hits (P)</em></td>
<td align="right">106</td>
<td align="right">0.39%</td>
</tr>
<tr class="odd">
<td align="left"><em>Fragmented (F)</em></td>
<td align="right">29</td>
<td align="right">0.11%</td>
</tr>
<tr class="even">
<td align="left"><strong>Likely Contamination (C)</strong></td>
<td align="right">0</td>
<td align="right">0.00%</td>
</tr>
<tr class="odd">
<td align="left"><em>Partial Hits (P)</em></td>
<td align="right">0</td>
<td align="right">0.00%</td>
</tr>
<tr class="even">
<td align="left"><em>Fragmented (F)</em></td>
<td align="right">0</td>
<td align="right">0.00%</td>
</tr>
<tr class="odd">
<td align="left"><strong>Unknown (U)</strong></td>
<td align="right">1,347</td>
<td align="right">4.99%</td>
</tr>
</tbody>
</table>
<p><strong>HOG Placement - Detected Species</strong></p>
<table class="table">
<colgroup>
<col width="53%">
<col width="15%">
<col width="15%">
<col width="15%">
</colgroup>
<thead><tr class="header">
<th><strong>Species</strong></th>
<th align="right"><strong>NCBI TaxID</strong></th>
<th align="right"><strong>Associated Proteins</strong></th>
<th align="right"><strong>% of Total Proteome</strong></th>
</tr></thead>
<tbody><tr class="odd">
<td><em>Arabidopsis thaliana</em></td>
<td align="right">3702</td>
<td align="right">25,647</td>
<td align="right">95.01%</td>
</tr></tbody>
</table>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="feature-assignment">Feature assignment<a class="anchor" aria-label="anchor" href="#feature-assignment"></a>
</h2>
<hr class="half-width">
<p>To measure the number of raw reads assigned to the features predicted
by the annotation, we will use <code>featureCounts</code> from the
<code>subread</code> package.</p>
<div class="codewrapper sourceCode" id="cb7">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="ex">ml</span> subread</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> featureCounts</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="cf">for</span> gff3 <span class="kw">in</span> <span class="pp">*</span>.gff3<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${gff3</span><span class="op">%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>    <span class="ex">featureCounts</span> <span class="dt">\</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>      <span class="at">-T</span> <span class="va">${SLURM_CPUS_ON_NODE}</span> <span class="dt">\</span></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>      <span class="at">-a</span> <span class="va">${gff3}</span> <span class="dt">\</span></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a>      <span class="at">-t</span> exon <span class="dt">\</span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>      <span class="at">-g</span> ID <span class="dt">\</span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a>      <span class="at">-p</span> <span class="dt">\</span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a>      <span class="at">-B</span> <span class="dt">\</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a>      <span class="at">-o</span> <span class="va">${base}</span>_merged_counts.txt <span class="dt">\</span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a>      <span class="at">--tmpDir</span> /tmp <span class="va">${RCAC_SCRATCH}</span>/annotation_workshop/00_datasets/bamfiles/<span class="pp">*</span>.bam</span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="reference-annotation-comparison">Reference annotation comparison<a class="anchor" aria-label="anchor" href="#reference-annotation-comparison"></a>
</h2>
<hr class="half-width">
<p>We will use <code>mikado compare</code> the reference annotation with
the predicted annotation.</p>
<div class="codewrapper sourceCode" id="cb8">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="ex">ml</span> <span class="at">--force</span> purge</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="ex">ml</span> mikado</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="va">WORKSHOP_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">${RCAC_SCRATCH}</span><span class="st">/annotation_workshop"</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="va">workdir</span><span class="op">=</span><span class="va">${WORKSHOP_DIR}</span>/08_assessment</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="bu">cd</span> <span class="va">${workdir}</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="cf">for</span> gff3 <span class="kw">in</span> <span class="pp">*</span>.gff3<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${gff3</span><span class="op">%</span>.<span class="pp">*</span><span class="va">})</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>    <span class="ex">mikado</span> compare <span class="dt">\</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>      <span class="at">--protein-coding</span> <span class="dt">\</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>      <span class="at">-r</span> <span class="va">${RCAC_SCRATCH}</span>/annotation_workshop/00_datasets/genome/athaliana_TAIR10.gff3 <span class="dt">\</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>      <span class="at">-p</span> <span class="va">${gff3}</span> <span class="dt">\</span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>      <span class="at">-o</span> ref-TAIR10_vs_prediction_<span class="va">${base}</span>_compared <span class="dt">\</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a>      <span class="at">--log</span> <span class="va">${base}</span>_compare.log</span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="cf">done</span></span></code></pre>
</div>
</section><section><h2 class="section-heading" id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<hr class="half-width">
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>
<code>busco</code> and <code>omark</code> assess how well conserved
genes are represented in the predicted gene set<br>
</li>
<li>
<code>gff3</code> metrics provide structural insights and highlight
discrepancies compared to known annotations<br>
</li>
<li>
<code>featureCounts</code> assignment quantifies the number of
RNA-seq reads aligning to predicted features<br>
</li>
<li>Reference annotation comparison evaluates how closely the predicted
genes match an established reference<br>
</li>
<li>Multiple assessment methods ensure a comprehensive evaluation of
annotation quality</li>
</ul>
</div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://rcac-bioinformatics.github.io/maize-genomics/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://rcac-bioinformatics.github.io/maize-genomics/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://rcac-bioinformatics.github.io/maize-genomics/" class="external-link">Source</a></p>
				<p><a href="https://rcac-bioinformatics.github.io/maize-genomics/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:rcac-help@purdue.edu">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.11" class="external-link">sandpaper (0.16.11)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.9" class="external-link">pegboard (0.7.9)</a>, and <a href="https://github.com/aseetharam/varnish/tree/db4f5c6f3c4799dfe8247fd057642c8099d5244b" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://maize-genomics.github.io/NA/instructor/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "gene-annotation, braker, helixer, gemoma, entap, omark, busco, hifiasm",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://maize-genomics.github.io/NA/instructor/aio.html",
  "identifier": "https://maize-genomics.github.io/NA/instructor/aio.html",
  "dateCreated": "2025-02-05",
  "dateModified": "2025-03-01",
  "datePublished": "2025-03-01"
}

  </script><script>
		feather.replace();
	</script>
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

